{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0b87e3b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: limma\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "\n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(edgeR)\n",
    "library(magrittr)\n",
    "library(ggplot2)\n",
    "library(DESeq2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b9116a8",
   "metadata": {},
   "source": [
    "# Automatise"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d154690",
   "metadata": {},
   "source": [
    "# On cell-state level"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5ee47b2",
   "metadata": {},
   "source": [
    "## For LV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5bd3755f",
   "metadata": {},
   "outputs": [],
   "source": [
    "x=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLSTATE_PSEUDOBULK_LV.csv\")\n",
    "genes_tofilter=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLSTATE_PSEUDOBULK_FILTERING_LV.csv\")\n",
    "colnames(genes_tofilter) <- gsub(\"mutation.negative\", 'PVneg', colnames(genes_tofilter))\n",
    "\n",
    "# Only needed for the column cell_state\n",
    "CELLTYPE_STATE <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/CELLSTATE_TRANSLATION_TABLE.csv\")\n",
    "colnames(CELLTYPE_STATE) <- c(\"cell_states\", 'cell_type')\n",
    "\n",
    "# Removes patients with only few (<10) nuclei numbers (as they might bias the analysis)\n",
    "CELLTYPE_FILTER <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ABSOLUTE_CELLSTATES_NUMBER_LV.csv\")\n",
    "\n",
    "\n",
    "rownames(x) <- x$X\n",
    "x <- x[,-1]\n",
    "l <- strsplit(colnames(x), \"__\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7bfc55f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This steps are needed to make names equal - space and + was converted to .\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\+\", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\" \", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\-\", \"\\\\.\", CELLTYPE_FILTER$cell_states)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ec241d65",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.data <- t(as.data.frame(l))\n",
    "rownames(meta.data) <- colnames(x)\n",
    "\n",
    "meta.data <- as.data.frame(meta.data)\n",
    "colnames(meta.data)  <-  c(\"Primary_Genetic_Phenotype\", \"cell_state\", 'Patient', 'X10X_version', 'Sex')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c474221a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'AD1.0'</li><li>'AD1.1'</li><li>'AD2'</li><li>'AD3'</li><li>'B'</li><li>'CD4T_act'</li><li>'CD4T_naive'</li><li>'CD4T_reg'</li><li>'CD8T_cytox'</li><li>'CD8T_em'</li><li>'CD8T_te'</li><li>'CD8T_te_IFNGhi'</li><li>'CD8T_trans'</li><li>'cDC1'</li><li>'cDC2'</li><li>'EC1.0'</li><li>'EC2.0'</li><li>'EC5.0'</li><li>'EC6.0'</li><li>'EC7.0'</li><li>'EC8.0'</li><li>'ILC'</li><li>'MAIT.like'</li><li>'Mast'</li><li>'Meso'</li><li>'MO_CD16.'</li><li>'MO_VCAN.'</li><li>'MP_FOLR2'</li><li>'MP_ISG'</li><li>'MP_LYVE1hi.MHCII_intermediate'</li><li>'MP_LYVE1hi.MHCII_lo'</li><li>'MP_LYVE1lo.MHCII_hi'</li><li>'MP_NFkB'</li><li>'MP_OSM'</li><li>'MP_transitional'</li><li>'MP_TREM2.'</li><li>'NC1.0'</li><li>'NC1.1'</li><li>'NC1.2'</li><li>'NC1.3'</li><li>'NC2'</li><li>'NC3'</li><li>'NC4'</li><li>'NC5'</li><li>'NK_CD16hi'</li><li>'NK_CD16hiIFNGhi'</li><li>'NK_CD56hi'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'Plasma'</li><li>'prolif_Lymphoids'</li><li>'prolif_Myeloids'</li><li>'SMC1.1'</li><li>'SMC1.2'</li><li>'SMC2'</li><li>'vCM1.0'</li><li>'vCM1.1'</li><li>'vCM1.2'</li><li>'vCM1.3'</li><li>'vCM2'</li><li>'vCM3.0'</li><li>'vCM3.1'</li><li>'vCM4'</li><li>'vCM5'</li><li>'vFB1.0'</li><li>'vFB1.1'</li><li>'vFB1.2'</li><li>'vFB2'</li><li>'vFB3'</li><li>'vFB4'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'AD1.0'\n",
       "\\item 'AD1.1'\n",
       "\\item 'AD2'\n",
       "\\item 'AD3'\n",
       "\\item 'B'\n",
       "\\item 'CD4T\\_act'\n",
       "\\item 'CD4T\\_naive'\n",
       "\\item 'CD4T\\_reg'\n",
       "\\item 'CD8T\\_cytox'\n",
       "\\item 'CD8T\\_em'\n",
       "\\item 'CD8T\\_te'\n",
       "\\item 'CD8T\\_te\\_IFNGhi'\n",
       "\\item 'CD8T\\_trans'\n",
       "\\item 'cDC1'\n",
       "\\item 'cDC2'\n",
       "\\item 'EC1.0'\n",
       "\\item 'EC2.0'\n",
       "\\item 'EC5.0'\n",
       "\\item 'EC6.0'\n",
       "\\item 'EC7.0'\n",
       "\\item 'EC8.0'\n",
       "\\item 'ILC'\n",
       "\\item 'MAIT.like'\n",
       "\\item 'Mast'\n",
       "\\item 'Meso'\n",
       "\\item 'MO\\_CD16.'\n",
       "\\item 'MO\\_VCAN.'\n",
       "\\item 'MP\\_FOLR2'\n",
       "\\item 'MP\\_ISG'\n",
       "\\item 'MP\\_LYVE1hi.MHCII\\_intermediate'\n",
       "\\item 'MP\\_LYVE1hi.MHCII\\_lo'\n",
       "\\item 'MP\\_LYVE1lo.MHCII\\_hi'\n",
       "\\item 'MP\\_NFkB'\n",
       "\\item 'MP\\_OSM'\n",
       "\\item 'MP\\_transitional'\n",
       "\\item 'MP\\_TREM2.'\n",
       "\\item 'NC1.0'\n",
       "\\item 'NC1.1'\n",
       "\\item 'NC1.2'\n",
       "\\item 'NC1.3'\n",
       "\\item 'NC2'\n",
       "\\item 'NC3'\n",
       "\\item 'NC4'\n",
       "\\item 'NC5'\n",
       "\\item 'NK\\_CD16hi'\n",
       "\\item 'NK\\_CD16hiIFNGhi'\n",
       "\\item 'NK\\_CD56hi'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'Plasma'\n",
       "\\item 'prolif\\_Lymphoids'\n",
       "\\item 'prolif\\_Myeloids'\n",
       "\\item 'SMC1.1'\n",
       "\\item 'SMC1.2'\n",
       "\\item 'SMC2'\n",
       "\\item 'vCM1.0'\n",
       "\\item 'vCM1.1'\n",
       "\\item 'vCM1.2'\n",
       "\\item 'vCM1.3'\n",
       "\\item 'vCM2'\n",
       "\\item 'vCM3.0'\n",
       "\\item 'vCM3.1'\n",
       "\\item 'vCM4'\n",
       "\\item 'vCM5'\n",
       "\\item 'vFB1.0'\n",
       "\\item 'vFB1.1'\n",
       "\\item 'vFB1.2'\n",
       "\\item 'vFB2'\n",
       "\\item 'vFB3'\n",
       "\\item 'vFB4'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "\n",
       "**Levels**: 1. 'AD1.0'\n",
       "2. 'AD1.1'\n",
       "3. 'AD2'\n",
       "4. 'AD3'\n",
       "5. 'B'\n",
       "6. 'CD4T_act'\n",
       "7. 'CD4T_naive'\n",
       "8. 'CD4T_reg'\n",
       "9. 'CD8T_cytox'\n",
       "10. 'CD8T_em'\n",
       "11. 'CD8T_te'\n",
       "12. 'CD8T_te_IFNGhi'\n",
       "13. 'CD8T_trans'\n",
       "14. 'cDC1'\n",
       "15. 'cDC2'\n",
       "16. 'EC1.0'\n",
       "17. 'EC2.0'\n",
       "18. 'EC5.0'\n",
       "19. 'EC6.0'\n",
       "20. 'EC7.0'\n",
       "21. 'EC8.0'\n",
       "22. 'ILC'\n",
       "23. 'MAIT.like'\n",
       "24. 'Mast'\n",
       "25. 'Meso'\n",
       "26. 'MO_CD16.'\n",
       "27. 'MO_VCAN.'\n",
       "28. 'MP_FOLR2'\n",
       "29. 'MP_ISG'\n",
       "30. 'MP_LYVE1hi.MHCII_intermediate'\n",
       "31. 'MP_LYVE1hi.MHCII_lo'\n",
       "32. 'MP_LYVE1lo.MHCII_hi'\n",
       "33. 'MP_NFkB'\n",
       "34. 'MP_OSM'\n",
       "35. 'MP_transitional'\n",
       "36. 'MP_TREM2.'\n",
       "37. 'NC1.0'\n",
       "38. 'NC1.1'\n",
       "39. 'NC1.2'\n",
       "40. 'NC1.3'\n",
       "41. 'NC2'\n",
       "42. 'NC3'\n",
       "43. 'NC4'\n",
       "44. 'NC5'\n",
       "45. 'NK_CD16hi'\n",
       "46. 'NK_CD16hiIFNGhi'\n",
       "47. 'NK_CD56hi'\n",
       "48. 'PC1'\n",
       "49. 'PC2'\n",
       "50. 'PC3'\n",
       "51. 'Plasma'\n",
       "52. 'prolif_Lymphoids'\n",
       "53. 'prolif_Myeloids'\n",
       "54. 'SMC1.1'\n",
       "55. 'SMC1.2'\n",
       "56. 'SMC2'\n",
       "57. 'vCM1.0'\n",
       "58. 'vCM1.1'\n",
       "59. 'vCM1.2'\n",
       "60. 'vCM1.3'\n",
       "61. 'vCM2'\n",
       "62. 'vCM3.0'\n",
       "63. 'vCM3.1'\n",
       "64. 'vCM4'\n",
       "65. 'vCM5'\n",
       "66. 'vFB1.0'\n",
       "67. 'vFB1.1'\n",
       "68. 'vFB1.2'\n",
       "69. 'vFB2'\n",
       "70. 'vFB3'\n",
       "71. 'vFB4'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "factor(0)\n",
       "71 Levels: AD1.0 AD1.1 AD2 AD3 B CD4T_act CD4T_naive CD4T_reg ... vFB4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(meta.data$cell_state[which(!(meta.data$cell_state %in% CELLTYPE_FILTER[,1]))])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6913f6a0",
   "metadata": {},
   "source": [
    "Check that read counts are natural numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2880ad49",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DT4</th><th scope=col>H14</th><th scope=col>H16</th><th scope=col>H36</th><th scope=col>H40</th><th scope=col>IC_H02</th><th scope=col>H37</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th><th scope=col>H39</th><th scope=col>IC_H01</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>7</td><td>1</td><td>2</td><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & DT4 & H14 & H16 & H36 & H40 & IC\\_H02 & H37 & H41 & H34 & H35 & H39 & IC\\_H01\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 7 & 1 & 2 & 1 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 12\n",
       "\n",
       "| <!--/--> | DT4 &lt;int&gt; | H14 &lt;int&gt; | H16 &lt;int&gt; | H36 &lt;int&gt; | H40 &lt;int&gt; | IC_H02 &lt;int&gt; | H37 &lt;int&gt; | H41 &lt;int&gt; | H34 &lt;int&gt; | H35 &lt;int&gt; | H39 &lt;int&gt; | IC_H01 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 7 | 1 | 2 | 1 | 1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  DT4 H14 H16 H36 H40 IC_H02 H37 H41 H34 H35 H39 IC_H01\n",
       "1 7   1   2   1   1   1      0   0   0   0   0   0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, LMNA in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bcf27107",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>D1</th><th scope=col>D2</th><th scope=col>D4</th><th scope=col>D5</th><th scope=col>D6</th><th scope=col>D7</th><th scope=col>ED_H15</th><th scope=col>ED_H20</th><th scope=col>ED_H25</th><th scope=col>ED_H26</th><th scope=col>ED_H35</th><th scope=col>H49</th><th scope=col>H51</th><th scope=col>H53</th><th scope=col>ED_H37</th><th scope=col>H46</th><th scope=col>H55</th><th scope=col>H67</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>54</td><td>71</td><td>109</td><td>66</td><td>25</td><td>3</td><td>21</td><td>29</td><td>6</td><td>52</td><td>22</td><td>12</td><td>7</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & D1 & D2 & D4 & D5 & D6 & D7 & ED\\_H15 & ED\\_H20 & ED\\_H25 & ED\\_H26 & ED\\_H35 & H49 & H51 & H53 & ED\\_H37 & H46 & H55 & H67\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 54 & 71 & 109 & 66 & 25 & 3 & 21 & 29 & 6 & 52 & 22 & 12 & 7 & 1 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 18\n",
       "\n",
       "| <!--/--> | D1 &lt;int&gt; | D2 &lt;int&gt; | D4 &lt;int&gt; | D5 &lt;int&gt; | D6 &lt;int&gt; | D7 &lt;int&gt; | ED_H15 &lt;int&gt; | ED_H20 &lt;int&gt; | ED_H25 &lt;int&gt; | ED_H26 &lt;int&gt; | ED_H35 &lt;int&gt; | H49 &lt;int&gt; | H51 &lt;int&gt; | H53 &lt;int&gt; | ED_H37 &lt;int&gt; | H46 &lt;int&gt; | H55 &lt;int&gt; | H67 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 54 | 71 | 109 | 66 | 25 | 3 | 21 | 29 | 6 | 52 | 22 | 12 | 7 | 1 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  D1 D2 D4  D5 D6 D7 ED_H15 ED_H20 ED_H25 ED_H26 ED_H35 H49 H51 H53 ED_H37 H46\n",
       "1 54 71 109 66 25 3  21     29     6      52     22     12  7   1   0      0  \n",
       "  H55 H67\n",
       "1 0   0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, control in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"control\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "91fb54ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Primary_Genetic_Phenotype</th><th scope=col>cell_state</th><th scope=col>Patient</th><th scope=col>X10X_version</th><th scope=col>Sex</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__DL2__V3__m</th><td>LMNA</td><td>AD1.0</td><td>DL2</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H05__V3__m</th><td>LMNA</td><td>AD1.0</td><td>H05</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H26__V3__f</th><td>LMNA</td><td>AD1.0</td><td>H26</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H28__V3__m</th><td>LMNA</td><td>AD1.0</td><td>H28</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H31__V3__m</th><td>LMNA</td><td>AD1.0</td><td>H31</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.1__H05__V3__m</th><td>LMNA</td><td>AD1.1</td><td>H05</td><td>V3</td><td>m</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Primary\\_Genetic\\_Phenotype & cell\\_state & Patient & X10X\\_version & Sex\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tLMNA\\_\\_AD1.0\\_\\_DL2\\_\\_V3\\_\\_m & LMNA & AD1.0 & DL2 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H05\\_\\_V3\\_\\_m & LMNA & AD1.0 & H05 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H26\\_\\_V3\\_\\_f & LMNA & AD1.0 & H26 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H28\\_\\_V3\\_\\_m & LMNA & AD1.0 & H28 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H31\\_\\_V3\\_\\_m & LMNA & AD1.0 & H31 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD1.1\\_\\_H05\\_\\_V3\\_\\_m & LMNA & AD1.1 & H05 & V3 & m\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | Primary_Genetic_Phenotype &lt;fct&gt; | cell_state &lt;fct&gt; | Patient &lt;fct&gt; | X10X_version &lt;fct&gt; | Sex &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| LMNA__AD1.0__DL2__V3__m | LMNA | AD1.0 | DL2 | V3 | m |\n",
       "| LMNA__AD1.0__H05__V3__m | LMNA | AD1.0 | H05 | V3 | m |\n",
       "| LMNA__AD1.0__H26__V3__f | LMNA | AD1.0 | H26 | V3 | f |\n",
       "| LMNA__AD1.0__H28__V3__m | LMNA | AD1.0 | H28 | V3 | m |\n",
       "| LMNA__AD1.0__H31__V3__m | LMNA | AD1.0 | H31 | V3 | m |\n",
       "| LMNA__AD1.1__H05__V3__m | LMNA | AD1.1 | H05 | V3 | m |\n",
       "\n"
      ],
      "text/plain": [
       "                        Primary_Genetic_Phenotype cell_state Patient\n",
       "LMNA__AD1.0__DL2__V3__m LMNA                      AD1.0      DL2    \n",
       "LMNA__AD1.0__H05__V3__m LMNA                      AD1.0      H05    \n",
       "LMNA__AD1.0__H26__V3__f LMNA                      AD1.0      H26    \n",
       "LMNA__AD1.0__H28__V3__m LMNA                      AD1.0      H28    \n",
       "LMNA__AD1.0__H31__V3__m LMNA                      AD1.0      H31    \n",
       "LMNA__AD1.1__H05__V3__m LMNA                      AD1.1      H05    \n",
       "                        X10X_version Sex\n",
       "LMNA__AD1.0__DL2__V3__m V3           m  \n",
       "LMNA__AD1.0__H05__V3__m V3           m  \n",
       "LMNA__AD1.0__H26__V3__f V3           f  \n",
       "LMNA__AD1.0__H28__V3__m V3           m  \n",
       "LMNA__AD1.0__H31__V3__m V3           m  \n",
       "LMNA__AD1.1__H05__V3__m V3           m  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meta.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a15fee6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>LMNA</li><li>PKP2</li><li>PVneg</li><li>RBM20</li><li>TTN</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'control'</li><li>'LMNA'</li><li>'PKP2'</li><li>'PVneg'</li><li>'RBM20'</li><li>'TTN'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item LMNA\n",
       "\\item PKP2\n",
       "\\item PVneg\n",
       "\\item RBM20\n",
       "\\item TTN\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'control'\n",
       "\\item 'LMNA'\n",
       "\\item 'PKP2'\n",
       "\\item 'PVneg'\n",
       "\\item 'RBM20'\n",
       "\\item 'TTN'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. LMNA\n",
       "2. PKP2\n",
       "3. PVneg\n",
       "4. RBM20\n",
       "5. TTN\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'control'\n",
       "2. 'LMNA'\n",
       "3. 'PKP2'\n",
       "4. 'PVneg'\n",
       "5. 'RBM20'\n",
       "6. 'TTN'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] LMNA  PKP2  PVneg RBM20 TTN  \n",
       "Levels: control LMNA PKP2 PVneg RBM20 TTN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GENOTYPES <- unique(meta.data[,\"Primary_Genetic_Phenotype\"][meta.data[,\"Primary_Genetic_Phenotype\"] != \"control\"])\n",
    "GENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a340179d",
   "metadata": {},
   "outputs": [],
   "source": [
    "CELL_LEVEL <- \"CELLSTATES\"\n",
    "REGION <- \"LV\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7c3cbada",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########\n",
    "# Convert to bool\n",
    "CELLTYPE_FILTER <- CELLTYPE_FILTER\n",
    "CELLTYPE_FILTER[,2:ncol(CELLTYPE_FILTER)] <- apply(CELLTYPE_FILTER[,-1], 2, function(i) i>=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d618aa87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>DT4</li><li>H14</li><li>H16</li><li>H36</li><li>H40</li><li>IC_H02</li><li>H37</li><li>H41</li><li>H34</li><li>H35</li><li>H39</li><li>IC_H01</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'D1'</li><li>'D2'</li><li>'D4'</li><li>'D5'</li><li>'D6'</li><li>'D7'</li><li>'DL2'</li><li>'DL3'</li><li>'DT4'</li><li>'ED_H15'</li><li>'ED_H20'</li><li>'ED_H25'</li><li>'ED_H26'</li><li>'ED_H35'</li><li>'ED_H37'</li><li>'H01'</li><li>'H02'</li><li>'H03'</li><li>'H04'</li><li>'H05'</li><li>'H06'</li><li>'H07'</li><li>'H08'</li><li>'H09'</li><li>'H10'</li><li>'H11'</li><li>'H14'</li><li>'H15'</li><li>'H16'</li><li>'H17'</li><li>'H18'</li><li>'H19'</li><li>'H20'</li><li>'H21'</li><li>'H22'</li><li>'H24'</li><li>'H25'</li><li>'H26'</li><li>'H27'</li><li>'H28'</li><li>'H29'</li><li>'H31'</li><li>'H33'</li><li>'H34'</li><li>'H35'</li><li>'H36'</li><li>'H37'</li><li>'H39'</li><li>'H40'</li><li>'H41'</li><li>'H42'</li><li>'H46'</li><li>'H49'</li><li>'H51'</li><li>'H53'</li><li>'H55'</li><li>'H56'</li><li>'H57'</li><li>'H58'</li><li>'H59'</li><li>'H67'</li><li>'IC_H01'</li><li>'IC_H02'</li><li>'IC_H04'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item DT4\n",
       "\\item H14\n",
       "\\item H16\n",
       "\\item H36\n",
       "\\item H40\n",
       "\\item IC\\_H02\n",
       "\\item H37\n",
       "\\item H41\n",
       "\\item H34\n",
       "\\item H35\n",
       "\\item H39\n",
       "\\item IC\\_H01\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'D1'\n",
       "\\item 'D2'\n",
       "\\item 'D4'\n",
       "\\item 'D5'\n",
       "\\item 'D6'\n",
       "\\item 'D7'\n",
       "\\item 'DL2'\n",
       "\\item 'DL3'\n",
       "\\item 'DT4'\n",
       "\\item 'ED\\_H15'\n",
       "\\item 'ED\\_H20'\n",
       "\\item 'ED\\_H25'\n",
       "\\item 'ED\\_H26'\n",
       "\\item 'ED\\_H35'\n",
       "\\item 'ED\\_H37'\n",
       "\\item 'H01'\n",
       "\\item 'H02'\n",
       "\\item 'H03'\n",
       "\\item 'H04'\n",
       "\\item 'H05'\n",
       "\\item 'H06'\n",
       "\\item 'H07'\n",
       "\\item 'H08'\n",
       "\\item 'H09'\n",
       "\\item 'H10'\n",
       "\\item 'H11'\n",
       "\\item 'H14'\n",
       "\\item 'H15'\n",
       "\\item 'H16'\n",
       "\\item 'H17'\n",
       "\\item 'H18'\n",
       "\\item 'H19'\n",
       "\\item 'H20'\n",
       "\\item 'H21'\n",
       "\\item 'H22'\n",
       "\\item 'H24'\n",
       "\\item 'H25'\n",
       "\\item 'H26'\n",
       "\\item 'H27'\n",
       "\\item 'H28'\n",
       "\\item 'H29'\n",
       "\\item 'H31'\n",
       "\\item 'H33'\n",
       "\\item 'H34'\n",
       "\\item 'H35'\n",
       "\\item 'H36'\n",
       "\\item 'H37'\n",
       "\\item 'H39'\n",
       "\\item 'H40'\n",
       "\\item 'H41'\n",
       "\\item 'H42'\n",
       "\\item 'H46'\n",
       "\\item 'H49'\n",
       "\\item 'H51'\n",
       "\\item 'H53'\n",
       "\\item 'H55'\n",
       "\\item 'H56'\n",
       "\\item 'H57'\n",
       "\\item 'H58'\n",
       "\\item 'H59'\n",
       "\\item 'H67'\n",
       "\\item 'IC\\_H01'\n",
       "\\item 'IC\\_H02'\n",
       "\\item 'IC\\_H04'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. DT4\n",
       "2. H14\n",
       "3. H16\n",
       "4. H36\n",
       "5. H40\n",
       "6. IC_H02\n",
       "7. H37\n",
       "8. H41\n",
       "9. H34\n",
       "10. H35\n",
       "11. H39\n",
       "12. IC_H01\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'D1'\n",
       "2. 'D2'\n",
       "3. 'D4'\n",
       "4. 'D5'\n",
       "5. 'D6'\n",
       "6. 'D7'\n",
       "7. 'DL2'\n",
       "8. 'DL3'\n",
       "9. 'DT4'\n",
       "10. 'ED_H15'\n",
       "11. 'ED_H20'\n",
       "12. 'ED_H25'\n",
       "13. 'ED_H26'\n",
       "14. 'ED_H35'\n",
       "15. 'ED_H37'\n",
       "16. 'H01'\n",
       "17. 'H02'\n",
       "18. 'H03'\n",
       "19. 'H04'\n",
       "20. 'H05'\n",
       "21. 'H06'\n",
       "22. 'H07'\n",
       "23. 'H08'\n",
       "24. 'H09'\n",
       "25. 'H10'\n",
       "26. 'H11'\n",
       "27. 'H14'\n",
       "28. 'H15'\n",
       "29. 'H16'\n",
       "30. 'H17'\n",
       "31. 'H18'\n",
       "32. 'H19'\n",
       "33. 'H20'\n",
       "34. 'H21'\n",
       "35. 'H22'\n",
       "36. 'H24'\n",
       "37. 'H25'\n",
       "38. 'H26'\n",
       "39. 'H27'\n",
       "40. 'H28'\n",
       "41. 'H29'\n",
       "42. 'H31'\n",
       "43. 'H33'\n",
       "44. 'H34'\n",
       "45. 'H35'\n",
       "46. 'H36'\n",
       "47. 'H37'\n",
       "48. 'H39'\n",
       "49. 'H40'\n",
       "50. 'H41'\n",
       "51. 'H42'\n",
       "52. 'H46'\n",
       "53. 'H49'\n",
       "54. 'H51'\n",
       "55. 'H53'\n",
       "56. 'H55'\n",
       "57. 'H56'\n",
       "58. 'H57'\n",
       "59. 'H58'\n",
       "60. 'H59'\n",
       "61. 'H67'\n",
       "62. 'IC_H01'\n",
       "63. 'IC_H02'\n",
       "64. 'IC_H04'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] DT4    H14    H16    H36    H40    IC_H02 H37    H41    H34    H35   \n",
       "[11] H39    IC_H01\n",
       "64 Levels: D1 D2 D4 D5 D6 D7 DL2 DL3 DT4 ED_H15 ED_H20 ED_H25 ED_H26 ... IC_H04"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "59012c7d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>DT4</th><th scope=col>H14</th><th scope=col>H16</th><th scope=col>H36</th><th scope=col>H40</th><th scope=col>IC_H02</th><th scope=col>H37</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th><th scope=col>H39</th><th scope=col>IC_H01</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TRUE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & DT4 & H14 & H16 & H36 & H40 & IC\\_H02 & H37 & H41 & H34 & H35 & H39 & IC\\_H01\\\\\n",
       "  & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & TRUE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 12\n",
       "\n",
       "| <!--/--> | DT4 &lt;lgl&gt; | H14 &lt;lgl&gt; | H16 &lt;lgl&gt; | H36 &lt;lgl&gt; | H40 &lt;lgl&gt; | IC_H02 &lt;lgl&gt; | H37 &lt;lgl&gt; | H41 &lt;lgl&gt; | H34 &lt;lgl&gt; | H35 &lt;lgl&gt; | H39 &lt;lgl&gt; | IC_H01 &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TRUE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE |\n",
       "\n"
      ],
      "text/plain": [
       "  DT4  H14   H16   H36   H40   IC_H02 H37   H41   H34   H35   H39   IC_H01\n",
       "1 TRUE FALSE FALSE FALSE FALSE FALSE  FALSE FALSE FALSE FALSE FALSE FALSE "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# This is the example from above, they will be excluded in downstream analysis\n",
    "# In this case, the comparison will be excluded, as only one sample has enough observations\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19c9911f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d71007ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "###START: LMNA #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n",
      "\n",
      "###START: PKP2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "i <- 1\n",
    "\n",
    "for(GENOTYPE in as.character(GENOTYPES)){\n",
    "    message(\"\\n###START: \", GENOTYPE,\" #####\\n\")\n",
    "    \n",
    "    for(CELL_STATE in as.character(unique(meta.data$cell_state))){\n",
    "        \n",
    "        CONTROL_COLUMN <- paste(\"control_\",CELL_STATE, sep=\"\")\n",
    "        GENOTYPE_COLUMN <- paste(GENOTYPE,\"_\", CELL_STATE, sep=\"\")\n",
    "        CELLTYPE <- as.character(CELLTYPE_STATE[which(CELLTYPE_STATE$cell_states==CELL_STATE),\"cell_type\"])\n",
    "        \n",
    "        \n",
    "        x_sub <- x[,which(meta.data[,\"Primary_Genetic_Phenotype\"] %in% c(\"control\", GENOTYPE))]\n",
    "        meta.data_sub <- meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"] %in% c(\"control\", GENOTYPE)),]\n",
    "        \n",
    "        x_sub <- x_sub[,which(meta.data_sub$cell_state %in% c(CELL_STATE))]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$cell_state %in% c(CELL_STATE)),]\n",
    "        \n",
    "        ##### remove first line (cell_type column)\n",
    "        CELLTYPE_FILTER_SUB <- CELLTYPE_FILTER\n",
    "        PATIENTS_TOKEEP <- CELLTYPE_FILTER_SUB[which(CELLTYPE_FILTER_SUB$cell_states==CELL_STATE),-1]\n",
    "        PATIENTS_TOKEEP <- colnames(PATIENTS_TOKEEP)[which(as.logical(PATIENTS_TOKEEP))]\n",
    "        #####\n",
    "        \n",
    "        # Subset only for patients which fulfill condition\n",
    "        x_sub <- x_sub[,which(meta.data_sub$Patient %in% PATIENTS_TOKEEP)]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$Patient %in% PATIENTS_TOKEEP),]\n",
    "        meta.data_sub[,\"Primary_Genetic_Phenotype\"] = droplevels(meta.data_sub[,\"Primary_Genetic_Phenotype\"]) ## Drop factor levels\n",
    "   \n",
    "        \n",
    "        # The test is only run, if 2 conditions have at least 2 patients\n",
    "        if (length(unique(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))==2 & \n",
    "            all(table(as.character(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))>=3))\n",
    "           {\n",
    "             # Prepare DGEList object\n",
    "            meta.data_sub[,\"Primary_Genetic_Phenotype\"] <- as.factor(as.character(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))\n",
    "            dge <- DGEList(counts=x_sub, group=meta.data_sub[,\"Primary_Genetic_Phenotype\"])\n",
    "            \n",
    "            # Filter genes based on expression, the qlf Object will be filtered to exclude them from FDR calculation\n",
    "            genes_tofilter_sub_GENOTYPE <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% GENOTYPE &\n",
    "                                                                 gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                 gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "\n",
    "            genes_tofilter_sub_CONTROL <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% 'control' &\n",
    "                                                                gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "            \n",
    "            genes_tofilter_sub <- data.frame(\"X\"=genes_tofilter[,\"X\"],\n",
    "                                             CONTROL_COLUMN=apply(genes_tofilter_sub_CONTROL, 1, mean),\n",
    "                                             GENOTYPE_COLUMN=apply(genes_tofilter_sub_GENOTYPE, 1, mean))\n",
    "            colnames(genes_tofilter_sub) <- c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)\n",
    "            \n",
    "            keep <- which(genes_tofilter_sub[,CONTROL_COLUMN] > 0.0125 |\n",
    "                          genes_tofilter_sub[,GENOTYPE_COLUMN] > 0.0125)\n",
    "            \n",
    "            #keep <- genes_tofilter[,CONTROL_COLUMN]>0.0125 | ##### change & to | ##### \n",
    "            #        genes_tofilter[,GENOTYPE_COLUMN]>0.0125\n",
    "            #dge <- dge[keep, , keep.lib.sizes=FALSE]\n",
    "    \n",
    "            # PP, model matrix, https://www.nature.com/articles/nmeth.4612 (edgeRQLFDetRate)\n",
    "            dge <- calcNormFactors(dge)\n",
    "            cdr <- scale(colMeans(x_sub > 0))\n",
    "            design <- model.matrix(~ cdr + meta.data_sub[,\"Primary_Genetic_Phenotype\"])\n",
    "    \n",
    "            # \n",
    "            dge <- estimateDisp(dge, design = design)\n",
    "            fit <- glmQLFit(dge, design = design)\n",
    "            qlf <- glmQLFTest(fit)\n",
    "    \n",
    "            # For all\n",
    "            tt <- topTags(qlf, n = Inf)\n",
    "            \n",
    "            # Only for \"expressed/detected\" genes\n",
    "            tt_filtered <- topTags(qlf[keep,], n = Inf)\n",
    "            \n",
    "            tt$table[,\"Gene\"] <- as.character(rownames(tt$table))\n",
    "            tt_filtered$table[,\"Gene\"] <- as.character(rownames(tt_filtered$table))\n",
    "            tt_filtered$table[,\"low_expression\"] <- \"F\" #####  Low-expression column ##### \n",
    "            tt_filtered$table[,\"FDR_plot\"] <- -log10(tt_filtered$table$FDR) #####  For plotting Volcano, here we use FDR, not pValue  ##### \n",
    "            \n",
    "            \n",
    "            tt_merged <- merge(tt$table[,c(\"Gene\", \"logFC\", \"logCPM\", 'F', 'PValue')], \n",
    "                               tt_filtered$table[,c(\"Gene\", \"FDR\", \"low_expression\", \"FDR_plot\")], \n",
    "                               by=\"Gene\", all=T)   \n",
    "            \n",
    "            # Replace NA FDRs with NA                   #####  ORDER CHANGED #####  \n",
    "            tt_merged[which(is.na(tt_merged$FDR)),\"FDR\"] <- 1 \n",
    "            \n",
    "            EXPRESSION_MEAN <- genes_tofilter_sub[,c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)]\n",
    "            colnames(EXPRESSION_MEAN) <- c(\"Gene\", \"mean_exp_control\", 'mean_exp_genotype')\n",
    "            tt_merged <- merge(tt_merged, EXPRESSION_MEAN, by=\"Gene\")\n",
    "    \n",
    "            \n",
    "            tt_merged[,\"Region\"] <- REGION\n",
    "            tt_merged[,\"annotation_level\"] <- CELL_LEVEL\n",
    "            tt_merged[,\"cell_state\"] <- CELL_STATE\n",
    "            tt_merged[,\"cell_type\"] <- CELLTYPE\n",
    "            tt_merged[,\"comparison\"] <- paste(\"control_\", GENOTYPE, sep=\"\")\n",
    "            \n",
    "            ###### Suggestion to add those columns\n",
    "            tt_merged[,\"Observations_genotype\"] <- as.numeric(table(meta.data_sub[,\"Primary_Genetic_Phenotype\"])[GENOTYPE])\n",
    "            tt_merged[,\"Observations_reference\"] <- as.numeric(table(meta.data_sub[,\"Primary_Genetic_Phenotype\"])['control'])\n",
    "            \n",
    "            # low expression add T\n",
    "            tt_merged[which(is.na(tt_merged$low_expression)),\"low_expression\"] <- \"T\"\n",
    "            \n",
    "            if(i==1){\n",
    "                 final_df <- tt_merged   \n",
    "            } else {\n",
    "                final_df <- rbind(final_df, tt_merged)\n",
    "            }\n",
    "            \n",
    "            i <- i + 1\n",
    "            message(\"\\n###FINISHED: \", CELL_STATE,\" #####\\n\")\n",
    "        } else {\n",
    "            message(\"\\n###Skip: \", CELL_STATE,\" #####\\n\")\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "d4630ad5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>9256483</th><td>ZXDB  </td><td>-0.8793595</td><td>2.646900</td><td>2.3759966</td><td>0.12967598</td><td>1.0000000</td><td>T</td><td>       NA</td><td>0.004488071</td><td>0.002088314</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "\t<tr><th scope=row>9256484</th><td>ZXDC  </td><td> 0.1884164</td><td>6.614096</td><td>1.5478392</td><td>0.21940012</td><td>0.4574654</td><td>F</td><td>0.3396418</td><td>0.181390447</td><td>0.269253461</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "\t<tr><th scope=row>9256485</th><td>ZYG11A</td><td>-0.1996221</td><td>2.986955</td><td>0.1418047</td><td>0.70812656</td><td>1.0000000</td><td>T</td><td>       NA</td><td>0.010091202</td><td>0.005229333</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "\t<tr><th scope=row>9256486</th><td>ZYG11B</td><td> 0.2527605</td><td>6.283061</td><td>2.7807413</td><td>0.10181252</td><td>0.2921935</td><td>F</td><td>0.5343295</td><td>0.141435339</td><td>0.214863723</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "\t<tr><th scope=row>9256487</th><td>ZYX   </td><td>-0.7261541</td><td>4.436926</td><td>5.3063500</td><td>0.02555078</td><td>0.1227531</td><td>F</td><td>0.9109676</td><td>0.041143998</td><td>0.037142365</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "\t<tr><th scope=row>9256488</th><td>ZZEF1 </td><td> 0.1289970</td><td>6.690831</td><td>0.7087126</td><td>0.40398010</td><td>0.6458588</td><td>F</td><td>0.1898624</td><td>0.199194105</td><td>0.280689411</td><td>LV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>12</td><td>18</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t9256483 & ZXDB   & -0.8793595 & 2.646900 & 2.3759966 & 0.12967598 & 1.0000000 & T &        NA & 0.004488071 & 0.002088314 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\t9256484 & ZXDC   &  0.1884164 & 6.614096 & 1.5478392 & 0.21940012 & 0.4574654 & F & 0.3396418 & 0.181390447 & 0.269253461 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\t9256485 & ZYG11A & -0.1996221 & 2.986955 & 0.1418047 & 0.70812656 & 1.0000000 & T &        NA & 0.010091202 & 0.005229333 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\t9256486 & ZYG11B &  0.2527605 & 6.283061 & 2.7807413 & 0.10181252 & 0.2921935 & F & 0.5343295 & 0.141435339 & 0.214863723 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\t9256487 & ZYX    & -0.7261541 & 4.436926 & 5.3063500 & 0.02555078 & 0.1227531 & F & 0.9109676 & 0.041143998 & 0.037142365 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\t9256488 & ZZEF1  &  0.1289970 & 6.690831 & 0.7087126 & 0.40398010 & 0.6458588 & F & 0.1898624 & 0.199194105 & 0.280689411 & LV & CELLSTATES & vFB4 & FB & control\\_TTN & 12 & 18\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;chr&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;chr&gt; | annotation_level &lt;chr&gt; | cell_state &lt;chr&gt; | cell_type &lt;chr&gt; | comparison &lt;chr&gt; | Observations_genotype &lt;dbl&gt; | Observations_reference &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 9256483 | ZXDB   | -0.8793595 | 2.646900 | 2.3759966 | 0.12967598 | 1.0000000 | T |        NA | 0.004488071 | 0.002088314 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "| 9256484 | ZXDC   |  0.1884164 | 6.614096 | 1.5478392 | 0.21940012 | 0.4574654 | F | 0.3396418 | 0.181390447 | 0.269253461 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "| 9256485 | ZYG11A | -0.1996221 | 2.986955 | 0.1418047 | 0.70812656 | 1.0000000 | T |        NA | 0.010091202 | 0.005229333 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "| 9256486 | ZYG11B |  0.2527605 | 6.283061 | 2.7807413 | 0.10181252 | 0.2921935 | F | 0.5343295 | 0.141435339 | 0.214863723 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "| 9256487 | ZYX    | -0.7261541 | 4.436926 | 5.3063500 | 0.02555078 | 0.1227531 | F | 0.9109676 | 0.041143998 | 0.037142365 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "| 9256488 | ZZEF1  |  0.1289970 | 6.690831 | 0.7087126 | 0.40398010 | 0.6458588 | F | 0.1898624 | 0.199194105 | 0.280689411 | LV | CELLSTATES | vFB4 | FB | control_TTN | 12 | 18 |\n",
       "\n"
      ],
      "text/plain": [
       "        Gene   logFC      logCPM   F         PValue     FDR      \n",
       "9256483 ZXDB   -0.8793595 2.646900 2.3759966 0.12967598 1.0000000\n",
       "9256484 ZXDC    0.1884164 6.614096 1.5478392 0.21940012 0.4574654\n",
       "9256485 ZYG11A -0.1996221 2.986955 0.1418047 0.70812656 1.0000000\n",
       "9256486 ZYG11B  0.2527605 6.283061 2.7807413 0.10181252 0.2921935\n",
       "9256487 ZYX    -0.7261541 4.436926 5.3063500 0.02555078 0.1227531\n",
       "9256488 ZZEF1   0.1289970 6.690831 0.7087126 0.40398010 0.6458588\n",
       "        low_expression FDR_plot  mean_exp_control mean_exp_genotype Region\n",
       "9256483 T                     NA 0.004488071      0.002088314       LV    \n",
       "9256484 F              0.3396418 0.181390447      0.269253461       LV    \n",
       "9256485 T                     NA 0.010091202      0.005229333       LV    \n",
       "9256486 F              0.5343295 0.141435339      0.214863723       LV    \n",
       "9256487 F              0.9109676 0.041143998      0.037142365       LV    \n",
       "9256488 F              0.1898624 0.199194105      0.280689411       LV    \n",
       "        annotation_level cell_state cell_type comparison  Observations_genotype\n",
       "9256483 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "9256484 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "9256485 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "9256486 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "9256487 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "9256488 CELLSTATES       vFB4       FB        control_TTN 12                   \n",
       "        Observations_reference\n",
       "9256483 18                    \n",
       "9256484 18                    \n",
       "9256485 18                    \n",
       "9256486 18                    \n",
       "9256487 18                    \n",
       "9256488 18                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(final_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "d64a5af2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "                        AD1.1                           AD2 \n",
       "                        67076                         33538 \n",
       "                     CD4T_act                    CD4T_naive \n",
       "                       167690                        167690 \n",
       "                   CD8T_cytox                       CD8T_em \n",
       "                       167690                        167690 \n",
       "                      CD8T_te                    CD8T_trans \n",
       "                       167690                        167690 \n",
       "                         cDC2                         EC1.0 \n",
       "                       167690                        167690 \n",
       "                        EC2.0                         EC5.0 \n",
       "                       167690                        167690 \n",
       "                        EC6.0                         EC7.0 \n",
       "                       167690                        167690 \n",
       "                        EC8.0                           ILC \n",
       "                       167690                         33538 \n",
       "                    MAIT.like                          Mast \n",
       "                       134152                        134152 \n",
       "                     MO_CD16.                      MO_VCAN. \n",
       "                       167690                        167690 \n",
       "                     MP_FOLR2                        MP_ISG \n",
       "                       167690                        167690 \n",
       "MP_LYVE1hi.MHCII_intermediate           MP_LYVE1hi.MHCII_lo \n",
       "                       167690                        167690 \n",
       "          MP_LYVE1lo.MHCII_hi                       MP_NFkB \n",
       "                       167690                        167690 \n",
       "                       MP_OSM               MP_transitional \n",
       "                       167690                        167690 \n",
       "                    MP_TREM2.                         NC1.0 \n",
       "                       167690                        167690 \n",
       "                        NC1.1                         NC1.3 \n",
       "                       167690                         33538 \n",
       "                          NC2                           NC3 \n",
       "                       167690                        167690 \n",
       "                          NC5                     NK_CD16hi \n",
       "                       134152                        167690 \n",
       "                    NK_CD56hi                           PC1 \n",
       "                       167690                        167690 \n",
       "                          PC2                           PC3 \n",
       "                       167690                        167690 \n",
       "              prolif_Myeloids                        SMC1.1 \n",
       "                       134152                        167690 \n",
       "                       SMC1.2                          SMC2 \n",
       "                       167690                        167690 \n",
       "                       vCM1.0                        vCM1.1 \n",
       "                       167690                        167690 \n",
       "                       vCM1.2                        vCM1.3 \n",
       "                       167690                        167690 \n",
       "                         vCM2                        vCM3.0 \n",
       "                       167690                        167690 \n",
       "                       vCM3.1                          vCM4 \n",
       "                       167690                        167690 \n",
       "                         vCM5                        vFB1.0 \n",
       "                       167690                        167690 \n",
       "                       vFB1.1                        vFB1.2 \n",
       "                       167690                        167690 \n",
       "                         vFB2                          vFB3 \n",
       "                       167690                        167690 \n",
       "                         vFB4 \n",
       "                       167690 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(final_df[,'cell_state'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "9b0d1d1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>AD1.1</dt><dd>5.13391677991923e-14</dd><dt>AD2</dt><dd>5.28377313036318e-10</dd><dt>CD4T_act</dt><dd>6.20932775481292e-19</dd><dt>CD4T_naive</dt><dd>1.23722241209532e-14</dd><dt>CD8T_cytox</dt><dd>5.49298318202517e-10</dd><dt>CD8T_em</dt><dd>3.09021459962442e-08</dd><dt>CD8T_te</dt><dd>1.44795758765911e-22</dd><dt>CD8T_trans</dt><dd>2.89890838397887e-08</dd><dt>cDC2</dt><dd>5.12141016889942e-11</dd><dt>EC1.0</dt><dd>5.25309634411355e-10</dd><dt>EC2.0</dt><dd>9.39943160624751e-07</dd><dt>EC5.0</dt><dd>1.40289787718249e-09</dd><dt>EC6.0</dt><dd>2.62127084703116e-09</dd><dt>EC7.0</dt><dd>2.9633168684968e-19</dd><dt>EC8.0</dt><dd>4.07769172015091e-13</dd><dt>ILC</dt><dd>0.476485160229111</dd><dt>MAIT.like</dt><dd>4.08549460733142e-07</dd><dt>Mast</dt><dd>3.54800610026028e-19</dd><dt>MO_CD16.</dt><dd>2.59658950677604e-05</dd><dt>MO_VCAN.</dt><dd>1.54428936696515e-20</dd><dt>MP_FOLR2</dt><dd>1.47985809258384e-28</dd><dt>MP_ISG</dt><dd>3.17191405511018e-27</dd><dt>MP_LYVE1hi.MHCII_intermediate</dt><dd>4.89223850032154e-23</dd><dt>MP_LYVE1hi.MHCII_lo</dt><dd>1.31762993093471e-10</dd><dt>MP_LYVE1lo.MHCII_hi</dt><dd>4.41181855419828e-26</dd><dt>MP_NFkB</dt><dd>3.18016133664751e-09</dd><dt>MP_OSM</dt><dd>4.88704745240877e-11</dd><dt>MP_transitional</dt><dd>8.20232594379719e-24</dd><dt>MP_TREM2.</dt><dd>3.85698663794425e-16</dd><dt>NC1.0</dt><dd>1.99413202539219e-10</dd><dt>NC1.1</dt><dd>1.71954906451022e-21</dd><dt>NC1.3</dt><dd>0.00366763320355653</dd><dt>NC2</dt><dd>4.49343306423317e-14</dd><dt>NC3</dt><dd>5.88264113667196e-07</dd><dt>NC5</dt><dd>0.00503125215405884</dd><dt>NK_CD16hi</dt><dd>1.42753466258319e-31</dd><dt>NK_CD56hi</dt><dd>6.45623477882972e-13</dd><dt>PC1</dt><dd>2.34401435064438e-13</dd><dt>PC2</dt><dd>3.31937510526983e-31</dd><dt>PC3</dt><dd>1.00346028619164e-09</dd><dt>prolif_Myeloids</dt><dd>1.64736950146457e-20</dd><dt>SMC1.1</dt><dd>1.54001902778886e-15</dd><dt>SMC1.2</dt><dd>8.12180089893022e-21</dd><dt>SMC2</dt><dd>1.23422406405558e-27</dd><dt>vCM1.0</dt><dd>3.0580723574966e-07</dd><dt>vCM1.1</dt><dd>3.37615482437138e-09</dd><dt>vCM1.2</dt><dd>1.6702891392075e-10</dd><dt>vCM1.3</dt><dd>3.63698532721944e-09</dd><dt>vCM2</dt><dd>2.38633461394236e-07</dd><dt>vCM3.0</dt><dd>2.48721638695784e-07</dd><dt>vCM3.1</dt><dd>4.59515551366735e-09</dd><dt>vCM4</dt><dd>1.40485982499589e-08</dd><dt>vCM5</dt><dd>6.10480689516982e-20</dd><dt>vFB1.0</dt><dd>7.66346936548079e-10</dd><dt>vFB1.1</dt><dd>5.60880300415269e-09</dd><dt>vFB1.2</dt><dd>4.85732741257966e-14</dd><dt>vFB2</dt><dd>2.64222232469298e-11</dd><dt>vFB3</dt><dd>3.07094656726536e-08</dd><dt>vFB4</dt><dd>7.31629031820754e-12</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[AD1.1] 5.13391677991923e-14\n",
       "\\item[AD2] 5.28377313036318e-10\n",
       "\\item[CD4T\\textbackslash{}\\_act] 6.20932775481292e-19\n",
       "\\item[CD4T\\textbackslash{}\\_naive] 1.23722241209532e-14\n",
       "\\item[CD8T\\textbackslash{}\\_cytox] 5.49298318202517e-10\n",
       "\\item[CD8T\\textbackslash{}\\_em] 3.09021459962442e-08\n",
       "\\item[CD8T\\textbackslash{}\\_te] 1.44795758765911e-22\n",
       "\\item[CD8T\\textbackslash{}\\_trans] 2.89890838397887e-08\n",
       "\\item[cDC2] 5.12141016889942e-11\n",
       "\\item[EC1.0] 5.25309634411355e-10\n",
       "\\item[EC2.0] 9.39943160624751e-07\n",
       "\\item[EC5.0] 1.40289787718249e-09\n",
       "\\item[EC6.0] 2.62127084703116e-09\n",
       "\\item[EC7.0] 2.9633168684968e-19\n",
       "\\item[EC8.0] 4.07769172015091e-13\n",
       "\\item[ILC] 0.476485160229111\n",
       "\\item[MAIT.like] 4.08549460733142e-07\n",
       "\\item[Mast] 3.54800610026028e-19\n",
       "\\item[MO\\textbackslash{}\\_CD16.] 2.59658950677604e-05\n",
       "\\item[MO\\textbackslash{}\\_VCAN.] 1.54428936696515e-20\n",
       "\\item[MP\\textbackslash{}\\_FOLR2] 1.47985809258384e-28\n",
       "\\item[MP\\textbackslash{}\\_ISG] 3.17191405511018e-27\n",
       "\\item[MP\\textbackslash{}\\_LYVE1hi.MHCII\\textbackslash{}\\_intermediate] 4.89223850032154e-23\n",
       "\\item[MP\\textbackslash{}\\_LYVE1hi.MHCII\\textbackslash{}\\_lo] 1.31762993093471e-10\n",
       "\\item[MP\\textbackslash{}\\_LYVE1lo.MHCII\\textbackslash{}\\_hi] 4.41181855419828e-26\n",
       "\\item[MP\\textbackslash{}\\_NFkB] 3.18016133664751e-09\n",
       "\\item[MP\\textbackslash{}\\_OSM] 4.88704745240877e-11\n",
       "\\item[MP\\textbackslash{}\\_transitional] 8.20232594379719e-24\n",
       "\\item[MP\\textbackslash{}\\_TREM2.] 3.85698663794425e-16\n",
       "\\item[NC1.0] 1.99413202539219e-10\n",
       "\\item[NC1.1] 1.71954906451022e-21\n",
       "\\item[NC1.3] 0.00366763320355653\n",
       "\\item[NC2] 4.49343306423317e-14\n",
       "\\item[NC3] 5.88264113667196e-07\n",
       "\\item[NC5] 0.00503125215405884\n",
       "\\item[NK\\textbackslash{}\\_CD16hi] 1.42753466258319e-31\n",
       "\\item[NK\\textbackslash{}\\_CD56hi] 6.45623477882972e-13\n",
       "\\item[PC1] 2.34401435064438e-13\n",
       "\\item[PC2] 3.31937510526983e-31\n",
       "\\item[PC3] 1.00346028619164e-09\n",
       "\\item[prolif\\textbackslash{}\\_Myeloids] 1.64736950146457e-20\n",
       "\\item[SMC1.1] 1.54001902778886e-15\n",
       "\\item[SMC1.2] 8.12180089893022e-21\n",
       "\\item[SMC2] 1.23422406405558e-27\n",
       "\\item[vCM1.0] 3.0580723574966e-07\n",
       "\\item[vCM1.1] 3.37615482437138e-09\n",
       "\\item[vCM1.2] 1.6702891392075e-10\n",
       "\\item[vCM1.3] 3.63698532721944e-09\n",
       "\\item[vCM2] 2.38633461394236e-07\n",
       "\\item[vCM3.0] 2.48721638695784e-07\n",
       "\\item[vCM3.1] 4.59515551366735e-09\n",
       "\\item[vCM4] 1.40485982499589e-08\n",
       "\\item[vCM5] 6.10480689516982e-20\n",
       "\\item[vFB1.0] 7.66346936548079e-10\n",
       "\\item[vFB1.1] 5.60880300415269e-09\n",
       "\\item[vFB1.2] 4.85732741257966e-14\n",
       "\\item[vFB2] 2.64222232469298e-11\n",
       "\\item[vFB3] 3.07094656726536e-08\n",
       "\\item[vFB4] 7.31629031820754e-12\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "AD1.1\n",
       ":   5.13391677991923e-14AD2\n",
       ":   5.28377313036318e-10CD4T_act\n",
       ":   6.20932775481292e-19CD4T_naive\n",
       ":   1.23722241209532e-14CD8T_cytox\n",
       ":   5.49298318202517e-10CD8T_em\n",
       ":   3.09021459962442e-08CD8T_te\n",
       ":   1.44795758765911e-22CD8T_trans\n",
       ":   2.89890838397887e-08cDC2\n",
       ":   5.12141016889942e-11EC1.0\n",
       ":   5.25309634411355e-10EC2.0\n",
       ":   9.39943160624751e-07EC5.0\n",
       ":   1.40289787718249e-09EC6.0\n",
       ":   2.62127084703116e-09EC7.0\n",
       ":   2.9633168684968e-19EC8.0\n",
       ":   4.07769172015091e-13ILC\n",
       ":   0.476485160229111MAIT.like\n",
       ":   4.08549460733142e-07Mast\n",
       ":   3.54800610026028e-19MO_CD16.\n",
       ":   2.59658950677604e-05MO_VCAN.\n",
       ":   1.54428936696515e-20MP_FOLR2\n",
       ":   1.47985809258384e-28MP_ISG\n",
       ":   3.17191405511018e-27MP_LYVE1hi.MHCII_intermediate\n",
       ":   4.89223850032154e-23MP_LYVE1hi.MHCII_lo\n",
       ":   1.31762993093471e-10MP_LYVE1lo.MHCII_hi\n",
       ":   4.41181855419828e-26MP_NFkB\n",
       ":   3.18016133664751e-09MP_OSM\n",
       ":   4.88704745240877e-11MP_transitional\n",
       ":   8.20232594379719e-24MP_TREM2.\n",
       ":   3.85698663794425e-16NC1.0\n",
       ":   1.99413202539219e-10NC1.1\n",
       ":   1.71954906451022e-21NC1.3\n",
       ":   0.00366763320355653NC2\n",
       ":   4.49343306423317e-14NC3\n",
       ":   5.88264113667196e-07NC5\n",
       ":   0.00503125215405884NK_CD16hi\n",
       ":   1.42753466258319e-31NK_CD56hi\n",
       ":   6.45623477882972e-13PC1\n",
       ":   2.34401435064438e-13PC2\n",
       ":   3.31937510526983e-31PC3\n",
       ":   1.00346028619164e-09prolif_Myeloids\n",
       ":   1.64736950146457e-20SMC1.1\n",
       ":   1.54001902778886e-15SMC1.2\n",
       ":   8.12180089893022e-21SMC2\n",
       ":   1.23422406405558e-27vCM1.0\n",
       ":   3.0580723574966e-07vCM1.1\n",
       ":   3.37615482437138e-09vCM1.2\n",
       ":   1.6702891392075e-10vCM1.3\n",
       ":   3.63698532721944e-09vCM2\n",
       ":   2.38633461394236e-07vCM3.0\n",
       ":   2.48721638695784e-07vCM3.1\n",
       ":   4.59515551366735e-09vCM4\n",
       ":   1.40485982499589e-08vCM5\n",
       ":   6.10480689516982e-20vFB1.0\n",
       ":   7.66346936548079e-10vFB1.1\n",
       ":   5.60880300415269e-09vFB1.2\n",
       ":   4.85732741257966e-14vFB2\n",
       ":   2.64222232469298e-11vFB3\n",
       ":   3.07094656726536e-08vFB4\n",
       ":   7.31629031820754e-12\n",
       "\n"
      ],
      "text/plain": [
       "                        AD1.1                           AD2 \n",
       "                 5.133917e-14                  5.283773e-10 \n",
       "                     CD4T_act                    CD4T_naive \n",
       "                 6.209328e-19                  1.237222e-14 \n",
       "                   CD8T_cytox                       CD8T_em \n",
       "                 5.492983e-10                  3.090215e-08 \n",
       "                      CD8T_te                    CD8T_trans \n",
       "                 1.447958e-22                  2.898908e-08 \n",
       "                         cDC2                         EC1.0 \n",
       "                 5.121410e-11                  5.253096e-10 \n",
       "                        EC2.0                         EC5.0 \n",
       "                 9.399432e-07                  1.402898e-09 \n",
       "                        EC6.0                         EC7.0 \n",
       "                 2.621271e-09                  2.963317e-19 \n",
       "                        EC8.0                           ILC \n",
       "                 4.077692e-13                  4.764852e-01 \n",
       "                    MAIT.like                          Mast \n",
       "                 4.085495e-07                  3.548006e-19 \n",
       "                     MO_CD16.                      MO_VCAN. \n",
       "                 2.596590e-05                  1.544289e-20 \n",
       "                     MP_FOLR2                        MP_ISG \n",
       "                 1.479858e-28                  3.171914e-27 \n",
       "MP_LYVE1hi.MHCII_intermediate           MP_LYVE1hi.MHCII_lo \n",
       "                 4.892239e-23                  1.317630e-10 \n",
       "          MP_LYVE1lo.MHCII_hi                       MP_NFkB \n",
       "                 4.411819e-26                  3.180161e-09 \n",
       "                       MP_OSM               MP_transitional \n",
       "                 4.887047e-11                  8.202326e-24 \n",
       "                    MP_TREM2.                         NC1.0 \n",
       "                 3.856987e-16                  1.994132e-10 \n",
       "                        NC1.1                         NC1.3 \n",
       "                 1.719549e-21                  3.667633e-03 \n",
       "                          NC2                           NC3 \n",
       "                 4.493433e-14                  5.882641e-07 \n",
       "                          NC5                     NK_CD16hi \n",
       "                 5.031252e-03                  1.427535e-31 \n",
       "                    NK_CD56hi                           PC1 \n",
       "                 6.456235e-13                  2.344014e-13 \n",
       "                          PC2                           PC3 \n",
       "                 3.319375e-31                  1.003460e-09 \n",
       "              prolif_Myeloids                        SMC1.1 \n",
       "                 1.647370e-20                  1.540019e-15 \n",
       "                       SMC1.2                          SMC2 \n",
       "                 8.121801e-21                  1.234224e-27 \n",
       "                       vCM1.0                        vCM1.1 \n",
       "                 3.058072e-07                  3.376155e-09 \n",
       "                       vCM1.2                        vCM1.3 \n",
       "                 1.670289e-10                  3.636985e-09 \n",
       "                         vCM2                        vCM3.0 \n",
       "                 2.386335e-07                  2.487216e-07 \n",
       "                       vCM3.1                          vCM4 \n",
       "                 4.595156e-09                  1.404860e-08 \n",
       "                         vCM5                        vFB1.0 \n",
       "                 6.104807e-20                  7.663469e-10 \n",
       "                       vFB1.1                        vFB1.2 \n",
       "                 5.608803e-09                  4.857327e-14 \n",
       "                         vFB2                          vFB3 \n",
       "                 2.642222e-11                  3.070947e-08 \n",
       "                         vFB4 \n",
       "                 7.316290e-12 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tapply(final_df[,\"FDR\"], \n",
    "       final_df[,'cell_state'], \n",
    "       min)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f13abc61",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(final_df, \"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALLCELLSTATES_ALLGENOTYPES_EDGER_LV_V8.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "7f461ac6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#final_df <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALLCELLSTATES_ALLGENOTYPES_EDGER_LV_V3.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24f82fbe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "594741fe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "d612d55d",
   "metadata": {},
   "source": [
    "## For RV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "fc76b700",
   "metadata": {},
   "outputs": [],
   "source": [
    "x=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLSTATE_PSEUDOBULK_RV.csv\")\n",
    "genes_tofilter=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLSTATE_PSEUDOBULK_FILTERING_RV.csv\")\n",
    "colnames(genes_tofilter) <- gsub(\"mutation.negative\", 'PVneg', colnames(genes_tofilter))\n",
    "\n",
    "# Only needed for the column cell_state\n",
    "CELLTYPE_STATE <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/CELLSTATE_TRANSLATION_TABLE.csv\")\n",
    "colnames(CELLTYPE_STATE) <- c(\"cell_states\", 'cell_type')\n",
    "\n",
    "# Removes patients with only few (<10) nuclei numbers (as they might bias the analysis)\n",
    "CELLTYPE_FILTER <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ABSOLUTE_CELLSTATES_NUMBER_RV.csv\")\n",
    "\n",
    "\n",
    "rownames(x) <- x$X\n",
    "x <- x[,-1]\n",
    "l <- strsplit(colnames(x), \"__\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f7345ced",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This steps are needed to make names equal - space and + was converted to .\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\+\", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\" \", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\-\", \"\\\\.\", CELLTYPE_FILTER$cell_states)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "37bcc6df",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.data <- t(as.data.frame(l))\n",
    "rownames(meta.data) <- colnames(x)\n",
    "\n",
    "meta.data <- as.data.frame(meta.data)\n",
    "colnames(meta.data)  <-  c(\"Primary_Genetic_Phenotype\", \"cell_state\", 'Patient', 'X10X_version', 'Sex')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "6531c7f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>H36</th><th scope=col>H14</th><th scope=col>H39</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & H36 & H14 & H39 & H41 & H34 & H35\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| <!--/--> | H36 &lt;int&gt; | H14 &lt;int&gt; | H39 &lt;int&gt; | H41 &lt;int&gt; | H34 &lt;int&gt; | H35 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  H36 H14 H39 H41 H34 H35\n",
       "1 1   0   0   0   0   0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, LMNA in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "580306ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>D1</th><th scope=col>D2</th><th scope=col>D4</th><th scope=col>D5</th><th scope=col>D6</th><th scope=col>D7</th><th scope=col>ED_H20</th><th scope=col>ED_H26</th><th scope=col>ED_H35</th><th scope=col>H49</th><th scope=col>H53</th><th scope=col>H55</th><th scope=col>ED_H15</th><th scope=col>ED_H25</th><th scope=col>ED_H37</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>10</td><td>2</td><td>36</td><td>3</td><td>12</td><td>12</td><td>41</td><td>42</td><td>6</td><td>67</td><td>9</td><td>4</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & D1 & D2 & D4 & D5 & D6 & D7 & ED\\_H20 & ED\\_H26 & ED\\_H35 & H49 & H53 & H55 & ED\\_H15 & ED\\_H25 & ED\\_H37\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 10 & 2 & 36 & 3 & 12 & 12 & 41 & 42 & 6 & 67 & 9 & 4 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 15\n",
       "\n",
       "| <!--/--> | D1 &lt;int&gt; | D2 &lt;int&gt; | D4 &lt;int&gt; | D5 &lt;int&gt; | D6 &lt;int&gt; | D7 &lt;int&gt; | ED_H20 &lt;int&gt; | ED_H26 &lt;int&gt; | ED_H35 &lt;int&gt; | H49 &lt;int&gt; | H53 &lt;int&gt; | H55 &lt;int&gt; | ED_H15 &lt;int&gt; | ED_H25 &lt;int&gt; | ED_H37 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 10 | 2 | 36 | 3 | 12 | 12 | 41 | 42 | 6 | 67 | 9 | 4 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  D1 D2 D4 D5 D6 D7 ED_H20 ED_H26 ED_H35 H49 H53 H55 ED_H15 ED_H25 ED_H37\n",
       "1 10 2  36 3  12 12 41     42     6      67  9   4   0      0      0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, control in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"control\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a45af7fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Primary_Genetic_Phenotype</th><th scope=col>cell_state</th><th scope=col>Patient</th><th scope=col>X10X_version</th><th scope=col>Sex</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H06__V3__f</th><td>LMNA</td><td>AD1.0</td><td>H06</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H24__V3__f</th><td>LMNA</td><td>AD1.0</td><td>H24</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.0__H28__V3__m</th><td>LMNA</td><td>AD1.0</td><td>H28</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.1__H06__V3__f</th><td>LMNA</td><td>AD1.1</td><td>H06</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.1__H24__V3__f</th><td>LMNA</td><td>AD1.1</td><td>H24</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD1.1__H25__V3__m</th><td>LMNA</td><td>AD1.1</td><td>H25</td><td>V3</td><td>m</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Primary\\_Genetic\\_Phenotype & cell\\_state & Patient & X10X\\_version & Sex\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H06\\_\\_V3\\_\\_f & LMNA & AD1.0 & H06 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H24\\_\\_V3\\_\\_f & LMNA & AD1.0 & H24 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD1.0\\_\\_H28\\_\\_V3\\_\\_m & LMNA & AD1.0 & H28 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD1.1\\_\\_H06\\_\\_V3\\_\\_f & LMNA & AD1.1 & H06 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD1.1\\_\\_H24\\_\\_V3\\_\\_f & LMNA & AD1.1 & H24 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD1.1\\_\\_H25\\_\\_V3\\_\\_m & LMNA & AD1.1 & H25 & V3 & m\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | Primary_Genetic_Phenotype &lt;fct&gt; | cell_state &lt;fct&gt; | Patient &lt;fct&gt; | X10X_version &lt;fct&gt; | Sex &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| LMNA__AD1.0__H06__V3__f | LMNA | AD1.0 | H06 | V3 | f |\n",
       "| LMNA__AD1.0__H24__V3__f | LMNA | AD1.0 | H24 | V3 | f |\n",
       "| LMNA__AD1.0__H28__V3__m | LMNA | AD1.0 | H28 | V3 | m |\n",
       "| LMNA__AD1.1__H06__V3__f | LMNA | AD1.1 | H06 | V3 | f |\n",
       "| LMNA__AD1.1__H24__V3__f | LMNA | AD1.1 | H24 | V3 | f |\n",
       "| LMNA__AD1.1__H25__V3__m | LMNA | AD1.1 | H25 | V3 | m |\n",
       "\n"
      ],
      "text/plain": [
       "                        Primary_Genetic_Phenotype cell_state Patient\n",
       "LMNA__AD1.0__H06__V3__f LMNA                      AD1.0      H06    \n",
       "LMNA__AD1.0__H24__V3__f LMNA                      AD1.0      H24    \n",
       "LMNA__AD1.0__H28__V3__m LMNA                      AD1.0      H28    \n",
       "LMNA__AD1.1__H06__V3__f LMNA                      AD1.1      H06    \n",
       "LMNA__AD1.1__H24__V3__f LMNA                      AD1.1      H24    \n",
       "LMNA__AD1.1__H25__V3__m LMNA                      AD1.1      H25    \n",
       "                        X10X_version Sex\n",
       "LMNA__AD1.0__H06__V3__f V3           f  \n",
       "LMNA__AD1.0__H24__V3__f V3           f  \n",
       "LMNA__AD1.0__H28__V3__m V3           m  \n",
       "LMNA__AD1.1__H06__V3__f V3           f  \n",
       "LMNA__AD1.1__H24__V3__f V3           f  \n",
       "LMNA__AD1.1__H25__V3__m V3           m  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meta.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "88a9954b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>LMNA</li><li>PKP2</li><li>PVneg</li><li>RBM20</li><li>TTN</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'control'</li><li>'LMNA'</li><li>'PKP2'</li><li>'PVneg'</li><li>'RBM20'</li><li>'TTN'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item LMNA\n",
       "\\item PKP2\n",
       "\\item PVneg\n",
       "\\item RBM20\n",
       "\\item TTN\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'control'\n",
       "\\item 'LMNA'\n",
       "\\item 'PKP2'\n",
       "\\item 'PVneg'\n",
       "\\item 'RBM20'\n",
       "\\item 'TTN'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. LMNA\n",
       "2. PKP2\n",
       "3. PVneg\n",
       "4. RBM20\n",
       "5. TTN\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'control'\n",
       "2. 'LMNA'\n",
       "3. 'PKP2'\n",
       "4. 'PVneg'\n",
       "5. 'RBM20'\n",
       "6. 'TTN'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] LMNA  PKP2  PVneg RBM20 TTN  \n",
       "Levels: control LMNA PKP2 PVneg RBM20 TTN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GENOTYPES <- unique(meta.data[,\"Primary_Genetic_Phenotype\"][meta.data[,\"Primary_Genetic_Phenotype\"] != \"control\"])\n",
    "GENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "8c7d41a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "CELL_LEVEL <- \"CELLSTATES\"\n",
    "REGION <- \"RV\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "80a140c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########\n",
    "# Convert to bool\n",
    "CELLTYPE_FILTER <- CELLTYPE_FILTER\n",
    "CELLTYPE_FILTER[,2:ncol(CELLTYPE_FILTER)] <- apply(CELLTYPE_FILTER[,-1], 2, function(i) i>=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "db9eba41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>H36</li><li>H14</li><li>H39</li><li>H41</li><li>H34</li><li>H35</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'D1'</li><li>'D2'</li><li>'D4'</li><li>'D5'</li><li>'D6'</li><li>'D7'</li><li>'ED_H15'</li><li>'ED_H20'</li><li>'ED_H25'</li><li>'ED_H26'</li><li>'ED_H35'</li><li>'ED_H37'</li><li>'H01'</li><li>'H02'</li><li>'H03'</li><li>'H06'</li><li>'H07'</li><li>'H08'</li><li>'H09'</li><li>'H10'</li><li>'H11'</li><li>'H14'</li><li>'H15'</li><li>'H19'</li><li>'H20'</li><li>'H22'</li><li>'H24'</li><li>'H25'</li><li>'H27'</li><li>'H28'</li><li>'H29'</li><li>'H31'</li><li>'H33'</li><li>'H34'</li><li>'H35'</li><li>'H36'</li><li>'H39'</li><li>'H41'</li><li>'H49'</li><li>'H53'</li><li>'H55'</li><li>'IC_H04'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item H36\n",
       "\\item H14\n",
       "\\item H39\n",
       "\\item H41\n",
       "\\item H34\n",
       "\\item H35\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'D1'\n",
       "\\item 'D2'\n",
       "\\item 'D4'\n",
       "\\item 'D5'\n",
       "\\item 'D6'\n",
       "\\item 'D7'\n",
       "\\item 'ED\\_H15'\n",
       "\\item 'ED\\_H20'\n",
       "\\item 'ED\\_H25'\n",
       "\\item 'ED\\_H26'\n",
       "\\item 'ED\\_H35'\n",
       "\\item 'ED\\_H37'\n",
       "\\item 'H01'\n",
       "\\item 'H02'\n",
       "\\item 'H03'\n",
       "\\item 'H06'\n",
       "\\item 'H07'\n",
       "\\item 'H08'\n",
       "\\item 'H09'\n",
       "\\item 'H10'\n",
       "\\item 'H11'\n",
       "\\item 'H14'\n",
       "\\item 'H15'\n",
       "\\item 'H19'\n",
       "\\item 'H20'\n",
       "\\item 'H22'\n",
       "\\item 'H24'\n",
       "\\item 'H25'\n",
       "\\item 'H27'\n",
       "\\item 'H28'\n",
       "\\item 'H29'\n",
       "\\item 'H31'\n",
       "\\item 'H33'\n",
       "\\item 'H34'\n",
       "\\item 'H35'\n",
       "\\item 'H36'\n",
       "\\item 'H39'\n",
       "\\item 'H41'\n",
       "\\item 'H49'\n",
       "\\item 'H53'\n",
       "\\item 'H55'\n",
       "\\item 'IC\\_H04'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. H36\n",
       "2. H14\n",
       "3. H39\n",
       "4. H41\n",
       "5. H34\n",
       "6. H35\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'D1'\n",
       "2. 'D2'\n",
       "3. 'D4'\n",
       "4. 'D5'\n",
       "5. 'D6'\n",
       "6. 'D7'\n",
       "7. 'ED_H15'\n",
       "8. 'ED_H20'\n",
       "9. 'ED_H25'\n",
       "10. 'ED_H26'\n",
       "11. 'ED_H35'\n",
       "12. 'ED_H37'\n",
       "13. 'H01'\n",
       "14. 'H02'\n",
       "15. 'H03'\n",
       "16. 'H06'\n",
       "17. 'H07'\n",
       "18. 'H08'\n",
       "19. 'H09'\n",
       "20. 'H10'\n",
       "21. 'H11'\n",
       "22. 'H14'\n",
       "23. 'H15'\n",
       "24. 'H19'\n",
       "25. 'H20'\n",
       "26. 'H22'\n",
       "27. 'H24'\n",
       "28. 'H25'\n",
       "29. 'H27'\n",
       "30. 'H28'\n",
       "31. 'H29'\n",
       "32. 'H31'\n",
       "33. 'H33'\n",
       "34. 'H34'\n",
       "35. 'H35'\n",
       "36. 'H36'\n",
       "37. 'H39'\n",
       "38. 'H41'\n",
       "39. 'H49'\n",
       "40. 'H53'\n",
       "41. 'H55'\n",
       "42. 'IC_H04'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] H36 H14 H39 H41 H34 H35\n",
       "42 Levels: D1 D2 D4 D5 D6 D7 ED_H15 ED_H20 ED_H25 ED_H26 ED_H35 ED_H37 ... IC_H04"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "423daed1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>H36</th><th scope=col>H14</th><th scope=col>H39</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & H36 & H14 & H39 & H41 & H34 & H35\\\\\n",
       "  & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & FALSE & FALSE & FALSE & FALSE & FALSE & FALSE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| <!--/--> | H36 &lt;lgl&gt; | H14 &lt;lgl&gt; | H39 &lt;lgl&gt; | H41 &lt;lgl&gt; | H34 &lt;lgl&gt; | H35 &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | FALSE | FALSE | FALSE | FALSE | FALSE | FALSE |\n",
       "\n"
      ],
      "text/plain": [
       "  H36   H14   H39   H41   H34   H35  \n",
       "1 FALSE FALSE FALSE FALSE FALSE FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# This is the example from above, they will be excluded in downstream analysis\n",
    "# In this case, the comparison will be excluded, as only one sample has enough observations\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"]==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "b296f91b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "###START: LMNA #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n",
      "\n",
      "###START: PKP2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n",
      "\n",
      "###START: PVneg #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n",
      "\n",
      "###START: RBM20 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n",
      "\n",
      "###START: TTN #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: B #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_act #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD4T_naive #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD4T_reg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_cytox #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_em #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te_IFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: CD8T_te #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CD8T_trans #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC2.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC5.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC6.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC7.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC8.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: ILC #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MAIT.like #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_CD16. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MO_VCAN. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_FOLR2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_ISG #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_intermediate #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1hi.MHCII_lo #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_LYVE1lo.MHCII_hi #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: MP_NFkB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_OSM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_TREM2. #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: MP_transitional #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Meso #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NC5 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: NK_CD16hiIFNGhi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD16hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NK_CD56hi #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: PC3 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Plasma #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: SMC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: cDC1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: cDC2 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Lymphoids #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: prolif_Myeloids #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM1.3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM3.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM4 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vCM5 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.0 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.1 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB1.2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB3 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: vFB4 #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD3 #####\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "i <- 1\n",
    "\n",
    "for(GENOTYPE in as.character(GENOTYPES)){\n",
    "    message(\"\\n###START: \", GENOTYPE,\" #####\\n\")\n",
    "    \n",
    "    for(CELL_STATE in as.character(unique(meta.data$cell_state))){\n",
    "        \n",
    "        CONTROL_COLUMN <- paste(\"control_\",CELL_STATE, sep=\"\")\n",
    "        GENOTYPE_COLUMN <- paste(GENOTYPE,\"_\", CELL_STATE, sep=\"\")\n",
    "        CELLTYPE <- as.character(CELLTYPE_STATE[which(CELLTYPE_STATE$cell_states==CELL_STATE),\"cell_type\"])\n",
    "        \n",
    "        \n",
    "        x_sub <- x[,which(meta.data[,\"Primary_Genetic_Phenotype\"] %in% c(\"control\", GENOTYPE))]\n",
    "        meta.data_sub <- meta.data[which(meta.data[,\"Primary_Genetic_Phenotype\"] %in% c(\"control\", GENOTYPE)),]\n",
    "        \n",
    "        x_sub <- x_sub[,which(meta.data_sub$cell_state %in% c(CELL_STATE))]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$cell_state %in% c(CELL_STATE)),]\n",
    "        \n",
    "        ##### remove first line (cell_type column)\n",
    "        CELLTYPE_FILTER_SUB <- CELLTYPE_FILTER\n",
    "        PATIENTS_TOKEEP <- CELLTYPE_FILTER_SUB[which(CELLTYPE_FILTER_SUB$cell_states==CELL_STATE),-1]\n",
    "        PATIENTS_TOKEEP <- colnames(PATIENTS_TOKEEP)[which(as.logical(PATIENTS_TOKEEP))]\n",
    "        #####\n",
    "        \n",
    "        # Subset only for patients which fulfill condition\n",
    "        x_sub <- x_sub[,which(meta.data_sub$Patient %in% PATIENTS_TOKEEP)]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$Patient %in% PATIENTS_TOKEEP),]\n",
    "        meta.data_sub[,\"Primary_Genetic_Phenotype\"] = droplevels(meta.data_sub[,\"Primary_Genetic_Phenotype\"]) ## Drop factor levels\n",
    "   \n",
    "        \n",
    "        # The test is only run, if 2 conditions have at least 2 patients\n",
    "        if (length(unique(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))==2 & \n",
    "            all(table(as.character(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))>=3))\n",
    "           {\n",
    "             # Prepare DGEList object\n",
    "            meta.data_sub[,\"Primary_Genetic_Phenotype\"] <- as.factor(as.character(meta.data_sub[,\"Primary_Genetic_Phenotype\"]))\n",
    "            dge <- DGEList(counts=x_sub, group=meta.data_sub[,\"Primary_Genetic_Phenotype\"])\n",
    "            \n",
    "            # Filter genes based on expression, the qlf Object will be filtered to exclude them from FDR calculation\n",
    "            genes_tofilter_sub_GENOTYPE <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% GENOTYPE &\n",
    "                                                                 gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                 gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "\n",
    "            genes_tofilter_sub_CONTROL <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% 'control' &\n",
    "                                                                gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "            \n",
    "            genes_tofilter_sub <- data.frame(\"X\"=genes_tofilter[,\"X\"],\n",
    "                                             CONTROL_COLUMN=apply(genes_tofilter_sub_CONTROL, 1, mean),\n",
    "                                             GENOTYPE_COLUMN=apply(genes_tofilter_sub_GENOTYPE, 1, mean))\n",
    "            colnames(genes_tofilter_sub) <- c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)\n",
    "            \n",
    "            keep <- which(genes_tofilter_sub[,CONTROL_COLUMN] > 0.0125 |\n",
    "                          genes_tofilter_sub[,GENOTYPE_COLUMN] > 0.0125)\n",
    "            \n",
    "            #keep <- genes_tofilter[,CONTROL_COLUMN]>0.0125 | ##### change & to | ##### \n",
    "            #        genes_tofilter[,GENOTYPE_COLUMN]>0.0125\n",
    "            #dge <- dge[keep, , keep.lib.sizes=FALSE]\n",
    "    \n",
    "            # PP, model matrix, https://www.nature.com/articles/nmeth.4612 (edgeRQLFDetRate)\n",
    "            dge <- calcNormFactors(dge)\n",
    "            cdr <- scale(colMeans(x_sub > 0))\n",
    "            design <- model.matrix(~ cdr + meta.data_sub[,\"Primary_Genetic_Phenotype\"])\n",
    "    \n",
    "            # \n",
    "            dge <- estimateDisp(dge, design = design)\n",
    "            fit <- glmQLFit(dge, design = design)\n",
    "            qlf <- glmQLFTest(fit)\n",
    "    \n",
    "            # For all\n",
    "            tt <- topTags(qlf, n = Inf)\n",
    "            \n",
    "            # Only for \"expressed/detected\" genes\n",
    "            tt_filtered <- topTags(qlf[keep,], n = Inf)\n",
    "            \n",
    "            tt$table[,\"Gene\"] <- as.character(rownames(tt$table))\n",
    "            tt_filtered$table[,\"Gene\"] <- as.character(rownames(tt_filtered$table))\n",
    "            tt_filtered$table[,\"low_expression\"] <- \"F\" #####  Low-expression column ##### \n",
    "            tt_filtered$table[,\"FDR_plot\"] <- -log10(tt_filtered$table$FDR) #####  For plotting Volcano, here we use FDR, not pValue  ##### \n",
    "            \n",
    "            \n",
    "            tt_merged <- merge(tt$table[,c(\"Gene\", \"logFC\", \"logCPM\", 'F', 'PValue')], \n",
    "                               tt_filtered$table[,c(\"Gene\", \"FDR\", \"low_expression\", \"FDR_plot\")], \n",
    "                               by=\"Gene\", all=T)   \n",
    "            \n",
    "            # Replace NA FDRs with NA                   #####  ORDER CHANGED #####  \n",
    "            tt_merged[which(is.na(tt_merged$FDR)),\"FDR\"] <- 1 \n",
    "            \n",
    "            EXPRESSION_MEAN <- genes_tofilter_sub[,c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)]\n",
    "            colnames(EXPRESSION_MEAN) <- c(\"Gene\", \"mean_exp_control\", 'mean_exp_genotype')\n",
    "            tt_merged <- merge(tt_merged, EXPRESSION_MEAN, by=\"Gene\")\n",
    "    \n",
    "            \n",
    "            tt_merged[,\"Region\"] <- REGION\n",
    "            tt_merged[,\"annotation_level\"] <- CELL_LEVEL\n",
    "            tt_merged[,\"cell_state\"] <- CELL_STATE\n",
    "            tt_merged[,\"cell_type\"] <- CELLTYPE\n",
    "            tt_merged[,\"comparison\"] <- paste(\"control_\", GENOTYPE, sep=\"\")\n",
    "            \n",
    "            ###### Suggestion to add those columns\n",
    "            tt_merged[,\"Observations_genotype\"] <- as.numeric(table(meta.data_sub[,\"Primary_Genetic_Phenotype\"])[GENOTYPE])\n",
    "            tt_merged[,\"Observations_reference\"] <- as.numeric(table(meta.data_sub[,\"Primary_Genetic_Phenotype\"])['control'])\n",
    "            \n",
    "            # low expression add T\n",
    "            tt_merged[which(is.na(tt_merged$low_expression)),\"low_expression\"] <- \"T\"\n",
    "            \n",
    "            if(i==1){\n",
    "                 final_df <- tt_merged   \n",
    "            } else {\n",
    "                final_df <- rbind(final_df, tt_merged)\n",
    "            }\n",
    "            \n",
    "            i <- i + 1\n",
    "            message(\"\\n###FINISHED: \", CELL_STATE,\" #####\\n\")\n",
    "        } else {\n",
    "            message(\"\\n###Skip: \", CELL_STATE,\" #####\\n\")\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "f8a75a92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>7982039</th><td>ZXDB  </td><td> 0.2745719</td><td>3.865191</td><td> 0.1260335</td><td>0.750507607</td><td>1.00000000</td><td>T</td><td>       NA</td><td>0.0025263358</td><td>0.006797264</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>7982040</th><td>ZXDC  </td><td> 0.2828352</td><td>6.474848</td><td> 0.8021216</td><td>0.372774272</td><td>0.69410837</td><td>F</td><td>0.1585727</td><td>0.1454636983</td><td>0.231421238</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>7982041</th><td>ZYG11A</td><td> 0.9231232</td><td>3.793061</td><td> 1.4493956</td><td>0.394627470</td><td>1.00000000</td><td>T</td><td>       NA</td><td>0.0001649621</td><td>0.005780245</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>7982042</th><td>ZYG11B</td><td> 0.2472182</td><td>6.315714</td><td> 0.5491753</td><td>0.460523752</td><td>0.75874789</td><td>F</td><td>0.1199025</td><td>0.1072939055</td><td>0.215667737</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>7982043</th><td>ZYX   </td><td>-1.9032982</td><td>5.071323</td><td>11.9903377</td><td>0.000810309</td><td>0.01913434</td><td>F</td><td>1.7181864</td><td>0.0418965648</td><td>0.022348622</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>7982044</th><td>ZZEF1 </td><td> 0.3911716</td><td>6.774139</td><td> 1.7548549</td><td>0.188516571</td><td>0.51901617</td><td>F</td><td>0.2848191</td><td>0.1629268748</td><td>0.319265485</td><td>RV</td><td>CELLSTATES</td><td>vFB4</td><td>FB</td><td>control_TTN</td><td>6</td><td>14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t7982039 & ZXDB   &  0.2745719 & 3.865191 &  0.1260335 & 0.750507607 & 1.00000000 & T &        NA & 0.0025263358 & 0.006797264 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\t7982040 & ZXDC   &  0.2828352 & 6.474848 &  0.8021216 & 0.372774272 & 0.69410837 & F & 0.1585727 & 0.1454636983 & 0.231421238 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\t7982041 & ZYG11A &  0.9231232 & 3.793061 &  1.4493956 & 0.394627470 & 1.00000000 & T &        NA & 0.0001649621 & 0.005780245 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\t7982042 & ZYG11B &  0.2472182 & 6.315714 &  0.5491753 & 0.460523752 & 0.75874789 & F & 0.1199025 & 0.1072939055 & 0.215667737 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\t7982043 & ZYX    & -1.9032982 & 5.071323 & 11.9903377 & 0.000810309 & 0.01913434 & F & 1.7181864 & 0.0418965648 & 0.022348622 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\t7982044 & ZZEF1  &  0.3911716 & 6.774139 &  1.7548549 & 0.188516571 & 0.51901617 & F & 0.2848191 & 0.1629268748 & 0.319265485 & RV & CELLSTATES & vFB4 & FB & control\\_TTN & 6 & 14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;chr&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;chr&gt; | annotation_level &lt;chr&gt; | cell_state &lt;chr&gt; | cell_type &lt;chr&gt; | comparison &lt;chr&gt; | Observations_genotype &lt;dbl&gt; | Observations_reference &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 7982039 | ZXDB   |  0.2745719 | 3.865191 |  0.1260335 | 0.750507607 | 1.00000000 | T |        NA | 0.0025263358 | 0.006797264 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "| 7982040 | ZXDC   |  0.2828352 | 6.474848 |  0.8021216 | 0.372774272 | 0.69410837 | F | 0.1585727 | 0.1454636983 | 0.231421238 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "| 7982041 | ZYG11A |  0.9231232 | 3.793061 |  1.4493956 | 0.394627470 | 1.00000000 | T |        NA | 0.0001649621 | 0.005780245 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "| 7982042 | ZYG11B |  0.2472182 | 6.315714 |  0.5491753 | 0.460523752 | 0.75874789 | F | 0.1199025 | 0.1072939055 | 0.215667737 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "| 7982043 | ZYX    | -1.9032982 | 5.071323 | 11.9903377 | 0.000810309 | 0.01913434 | F | 1.7181864 | 0.0418965648 | 0.022348622 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "| 7982044 | ZZEF1  |  0.3911716 | 6.774139 |  1.7548549 | 0.188516571 | 0.51901617 | F | 0.2848191 | 0.1629268748 | 0.319265485 | RV | CELLSTATES | vFB4 | FB | control_TTN | 6 | 14 |\n",
       "\n"
      ],
      "text/plain": [
       "        Gene   logFC      logCPM   F          PValue      FDR       \n",
       "7982039 ZXDB    0.2745719 3.865191  0.1260335 0.750507607 1.00000000\n",
       "7982040 ZXDC    0.2828352 6.474848  0.8021216 0.372774272 0.69410837\n",
       "7982041 ZYG11A  0.9231232 3.793061  1.4493956 0.394627470 1.00000000\n",
       "7982042 ZYG11B  0.2472182 6.315714  0.5491753 0.460523752 0.75874789\n",
       "7982043 ZYX    -1.9032982 5.071323 11.9903377 0.000810309 0.01913434\n",
       "7982044 ZZEF1   0.3911716 6.774139  1.7548549 0.188516571 0.51901617\n",
       "        low_expression FDR_plot  mean_exp_control mean_exp_genotype Region\n",
       "7982039 T                     NA 0.0025263358     0.006797264       RV    \n",
       "7982040 F              0.1585727 0.1454636983     0.231421238       RV    \n",
       "7982041 T                     NA 0.0001649621     0.005780245       RV    \n",
       "7982042 F              0.1199025 0.1072939055     0.215667737       RV    \n",
       "7982043 F              1.7181864 0.0418965648     0.022348622       RV    \n",
       "7982044 F              0.2848191 0.1629268748     0.319265485       RV    \n",
       "        annotation_level cell_state cell_type comparison  Observations_genotype\n",
       "7982039 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "7982040 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "7982041 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "7982042 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "7982043 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "7982044 CELLSTATES       vFB4       FB        control_TTN 6                    \n",
       "        Observations_reference\n",
       "7982039 14                    \n",
       "7982040 14                    \n",
       "7982041 14                    \n",
       "7982042 14                    \n",
       "7982043 14                    \n",
       "7982044 14                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(final_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "1fa57343",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "                          AD2                      CD4T_act \n",
       "                        67076                        167690 \n",
       "                   CD4T_naive                    CD8T_cytox \n",
       "                       100614                        100614 \n",
       "                      CD8T_em                       CD8T_te \n",
       "                       134152                        100614 \n",
       "                   CD8T_trans                          cDC2 \n",
       "                       167690                        167690 \n",
       "                        EC1.0                         EC2.0 \n",
       "                       167690                        167690 \n",
       "                        EC5.0                         EC6.0 \n",
       "                       167690                        167690 \n",
       "                        EC7.0                         EC8.0 \n",
       "                       167690                        167690 \n",
       "                         Mast                      MO_CD16. \n",
       "                       100614                        100614 \n",
       "                     MO_VCAN.                      MP_FOLR2 \n",
       "                       167690                        134152 \n",
       "                       MP_ISG MP_LYVE1hi.MHCII_intermediate \n",
       "                       167690                        167690 \n",
       "          MP_LYVE1hi.MHCII_lo           MP_LYVE1lo.MHCII_hi \n",
       "                       167690                        167690 \n",
       "                       MP_OSM               MP_transitional \n",
       "                       167690                        167690 \n",
       "                    MP_TREM2.                         NC1.0 \n",
       "                       167690                        167690 \n",
       "                        NC1.1                           NC2 \n",
       "                       167690                        167690 \n",
       "                    NK_CD16hi                     NK_CD56hi \n",
       "                       167690                        167690 \n",
       "                          PC1                           PC2 \n",
       "                       167690                        167690 \n",
       "                          PC3                        SMC1.1 \n",
       "                       167690                        167690 \n",
       "                       SMC1.2                          SMC2 \n",
       "                       167690                        167690 \n",
       "                       vCM1.0                        vCM1.1 \n",
       "                       167690                        167690 \n",
       "                       vCM1.2                        vCM1.3 \n",
       "                       167690                        167690 \n",
       "                         vCM2                        vCM3.0 \n",
       "                       167690                        167690 \n",
       "                       vCM3.1                          vCM4 \n",
       "                       167690                        134152 \n",
       "                         vCM5                        vFB1.0 \n",
       "                       134152                        167690 \n",
       "                       vFB1.1                        vFB1.2 \n",
       "                       167690                        167690 \n",
       "                         vFB2                          vFB3 \n",
       "                       167690                        167690 \n",
       "                         vFB4 \n",
       "                       167690 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(final_df[,'cell_state'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "fe6997ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>AD2</dt><dd>1.17293826354795e-11</dd><dt>CD4T_act</dt><dd>2.31871644705094e-19</dd><dt>CD4T_naive</dt><dd>0.000212690136967371</dd><dt>CD8T_cytox</dt><dd>0.00112620078663161</dd><dt>CD8T_em</dt><dd>0.00174185040819269</dd><dt>CD8T_te</dt><dd>0.000287392866469654</dd><dt>CD8T_trans</dt><dd>1.47768686654592e-19</dd><dt>cDC2</dt><dd>1.64261321826296e-12</dd><dt>EC1.0</dt><dd>7.0782008788313e-14</dd><dt>EC2.0</dt><dd>1.28862495662597e-12</dd><dt>EC5.0</dt><dd>2.43771529210398e-21</dd><dt>EC6.0</dt><dd>5.43757224750818e-15</dd><dt>EC7.0</dt><dd>1.10186961196622e-29</dd><dt>EC8.0</dt><dd>2.63579745578668e-45</dd><dt>Mast</dt><dd>3.56286204766228e-13</dd><dt>MO_CD16.</dt><dd>5.30428479242191e-07</dd><dt>MO_VCAN.</dt><dd>3.99100588020443e-20</dd><dt>MP_FOLR2</dt><dd>4.86380409209085e-08</dd><dt>MP_ISG</dt><dd>3.02008168158959e-22</dd><dt>MP_LYVE1hi.MHCII_intermediate</dt><dd>5.02305290365156e-19</dd><dt>MP_LYVE1hi.MHCII_lo</dt><dd>6.87384351520599e-20</dd><dt>MP_LYVE1lo.MHCII_hi</dt><dd>1.2375640866342e-20</dd><dt>MP_OSM</dt><dd>7.96865207961677e-34</dd><dt>MP_transitional</dt><dd>6.61701507361961e-22</dd><dt>MP_TREM2.</dt><dd>0.00057609980794478</dd><dt>NC1.0</dt><dd>1.58342548349439e-06</dd><dt>NC1.1</dt><dd>1.19503953339598e-13</dd><dt>NC2</dt><dd>4.3866225299049e-09</dd><dt>NK_CD16hi</dt><dd>4.19204441249063e-10</dd><dt>NK_CD56hi</dt><dd>4.90818470736811e-08</dd><dt>PC1</dt><dd>7.98279502384276e-09</dd><dt>PC2</dt><dd>1.11693618113781e-40</dd><dt>PC3</dt><dd>7.45538220732637e-17</dd><dt>SMC1.1</dt><dd>3.26095272054131e-17</dd><dt>SMC1.2</dt><dd>8.12971719725626e-25</dd><dt>SMC2</dt><dd>5.30167229317535e-18</dd><dt>vCM1.0</dt><dd>2.84297939661571e-07</dd><dt>vCM1.1</dt><dd>6.75744751550913e-11</dd><dt>vCM1.2</dt><dd>4.45649006146983e-18</dd><dt>vCM1.3</dt><dd>3.22175875404986e-11</dd><dt>vCM2</dt><dd>1.05477615192838e-05</dd><dt>vCM3.0</dt><dd>6.2615542456183e-07</dd><dt>vCM3.1</dt><dd>9.66240736835193e-08</dd><dt>vCM4</dt><dd>1.94368565306556e-10</dd><dt>vCM5</dt><dd>5.06367945255929e-15</dd><dt>vFB1.0</dt><dd>2.7466084368987e-08</dd><dt>vFB1.1</dt><dd>2.42270850173302e-33</dd><dt>vFB1.2</dt><dd>7.07187976526721e-23</dd><dt>vFB2</dt><dd>9.45577607610829e-11</dd><dt>vFB3</dt><dd>5.44567361247431e-20</dd><dt>vFB4</dt><dd>4.16687248765392e-16</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[AD2] 1.17293826354795e-11\n",
       "\\item[CD4T\\textbackslash{}\\_act] 2.31871644705094e-19\n",
       "\\item[CD4T\\textbackslash{}\\_naive] 0.000212690136967371\n",
       "\\item[CD8T\\textbackslash{}\\_cytox] 0.00112620078663161\n",
       "\\item[CD8T\\textbackslash{}\\_em] 0.00174185040819269\n",
       "\\item[CD8T\\textbackslash{}\\_te] 0.000287392866469654\n",
       "\\item[CD8T\\textbackslash{}\\_trans] 1.47768686654592e-19\n",
       "\\item[cDC2] 1.64261321826296e-12\n",
       "\\item[EC1.0] 7.0782008788313e-14\n",
       "\\item[EC2.0] 1.28862495662597e-12\n",
       "\\item[EC5.0] 2.43771529210398e-21\n",
       "\\item[EC6.0] 5.43757224750818e-15\n",
       "\\item[EC7.0] 1.10186961196622e-29\n",
       "\\item[EC8.0] 2.63579745578668e-45\n",
       "\\item[Mast] 3.56286204766228e-13\n",
       "\\item[MO\\textbackslash{}\\_CD16.] 5.30428479242191e-07\n",
       "\\item[MO\\textbackslash{}\\_VCAN.] 3.99100588020443e-20\n",
       "\\item[MP\\textbackslash{}\\_FOLR2] 4.86380409209085e-08\n",
       "\\item[MP\\textbackslash{}\\_ISG] 3.02008168158959e-22\n",
       "\\item[MP\\textbackslash{}\\_LYVE1hi.MHCII\\textbackslash{}\\_intermediate] 5.02305290365156e-19\n",
       "\\item[MP\\textbackslash{}\\_LYVE1hi.MHCII\\textbackslash{}\\_lo] 6.87384351520599e-20\n",
       "\\item[MP\\textbackslash{}\\_LYVE1lo.MHCII\\textbackslash{}\\_hi] 1.2375640866342e-20\n",
       "\\item[MP\\textbackslash{}\\_OSM] 7.96865207961677e-34\n",
       "\\item[MP\\textbackslash{}\\_transitional] 6.61701507361961e-22\n",
       "\\item[MP\\textbackslash{}\\_TREM2.] 0.00057609980794478\n",
       "\\item[NC1.0] 1.58342548349439e-06\n",
       "\\item[NC1.1] 1.19503953339598e-13\n",
       "\\item[NC2] 4.3866225299049e-09\n",
       "\\item[NK\\textbackslash{}\\_CD16hi] 4.19204441249063e-10\n",
       "\\item[NK\\textbackslash{}\\_CD56hi] 4.90818470736811e-08\n",
       "\\item[PC1] 7.98279502384276e-09\n",
       "\\item[PC2] 1.11693618113781e-40\n",
       "\\item[PC3] 7.45538220732637e-17\n",
       "\\item[SMC1.1] 3.26095272054131e-17\n",
       "\\item[SMC1.2] 8.12971719725626e-25\n",
       "\\item[SMC2] 5.30167229317535e-18\n",
       "\\item[vCM1.0] 2.84297939661571e-07\n",
       "\\item[vCM1.1] 6.75744751550913e-11\n",
       "\\item[vCM1.2] 4.45649006146983e-18\n",
       "\\item[vCM1.3] 3.22175875404986e-11\n",
       "\\item[vCM2] 1.05477615192838e-05\n",
       "\\item[vCM3.0] 6.2615542456183e-07\n",
       "\\item[vCM3.1] 9.66240736835193e-08\n",
       "\\item[vCM4] 1.94368565306556e-10\n",
       "\\item[vCM5] 5.06367945255929e-15\n",
       "\\item[vFB1.0] 2.7466084368987e-08\n",
       "\\item[vFB1.1] 2.42270850173302e-33\n",
       "\\item[vFB1.2] 7.07187976526721e-23\n",
       "\\item[vFB2] 9.45577607610829e-11\n",
       "\\item[vFB3] 5.44567361247431e-20\n",
       "\\item[vFB4] 4.16687248765392e-16\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "AD2\n",
       ":   1.17293826354795e-11CD4T_act\n",
       ":   2.31871644705094e-19CD4T_naive\n",
       ":   0.000212690136967371CD8T_cytox\n",
       ":   0.00112620078663161CD8T_em\n",
       ":   0.00174185040819269CD8T_te\n",
       ":   0.000287392866469654CD8T_trans\n",
       ":   1.47768686654592e-19cDC2\n",
       ":   1.64261321826296e-12EC1.0\n",
       ":   7.0782008788313e-14EC2.0\n",
       ":   1.28862495662597e-12EC5.0\n",
       ":   2.43771529210398e-21EC6.0\n",
       ":   5.43757224750818e-15EC7.0\n",
       ":   1.10186961196622e-29EC8.0\n",
       ":   2.63579745578668e-45Mast\n",
       ":   3.56286204766228e-13MO_CD16.\n",
       ":   5.30428479242191e-07MO_VCAN.\n",
       ":   3.99100588020443e-20MP_FOLR2\n",
       ":   4.86380409209085e-08MP_ISG\n",
       ":   3.02008168158959e-22MP_LYVE1hi.MHCII_intermediate\n",
       ":   5.02305290365156e-19MP_LYVE1hi.MHCII_lo\n",
       ":   6.87384351520599e-20MP_LYVE1lo.MHCII_hi\n",
       ":   1.2375640866342e-20MP_OSM\n",
       ":   7.96865207961677e-34MP_transitional\n",
       ":   6.61701507361961e-22MP_TREM2.\n",
       ":   0.00057609980794478NC1.0\n",
       ":   1.58342548349439e-06NC1.1\n",
       ":   1.19503953339598e-13NC2\n",
       ":   4.3866225299049e-09NK_CD16hi\n",
       ":   4.19204441249063e-10NK_CD56hi\n",
       ":   4.90818470736811e-08PC1\n",
       ":   7.98279502384276e-09PC2\n",
       ":   1.11693618113781e-40PC3\n",
       ":   7.45538220732637e-17SMC1.1\n",
       ":   3.26095272054131e-17SMC1.2\n",
       ":   8.12971719725626e-25SMC2\n",
       ":   5.30167229317535e-18vCM1.0\n",
       ":   2.84297939661571e-07vCM1.1\n",
       ":   6.75744751550913e-11vCM1.2\n",
       ":   4.45649006146983e-18vCM1.3\n",
       ":   3.22175875404986e-11vCM2\n",
       ":   1.05477615192838e-05vCM3.0\n",
       ":   6.2615542456183e-07vCM3.1\n",
       ":   9.66240736835193e-08vCM4\n",
       ":   1.94368565306556e-10vCM5\n",
       ":   5.06367945255929e-15vFB1.0\n",
       ":   2.7466084368987e-08vFB1.1\n",
       ":   2.42270850173302e-33vFB1.2\n",
       ":   7.07187976526721e-23vFB2\n",
       ":   9.45577607610829e-11vFB3\n",
       ":   5.44567361247431e-20vFB4\n",
       ":   4.16687248765392e-16\n",
       "\n"
      ],
      "text/plain": [
       "                          AD2                      CD4T_act \n",
       "                 1.172938e-11                  2.318716e-19 \n",
       "                   CD4T_naive                    CD8T_cytox \n",
       "                 2.126901e-04                  1.126201e-03 \n",
       "                      CD8T_em                       CD8T_te \n",
       "                 1.741850e-03                  2.873929e-04 \n",
       "                   CD8T_trans                          cDC2 \n",
       "                 1.477687e-19                  1.642613e-12 \n",
       "                        EC1.0                         EC2.0 \n",
       "                 7.078201e-14                  1.288625e-12 \n",
       "                        EC5.0                         EC6.0 \n",
       "                 2.437715e-21                  5.437572e-15 \n",
       "                        EC7.0                         EC8.0 \n",
       "                 1.101870e-29                  2.635797e-45 \n",
       "                         Mast                      MO_CD16. \n",
       "                 3.562862e-13                  5.304285e-07 \n",
       "                     MO_VCAN.                      MP_FOLR2 \n",
       "                 3.991006e-20                  4.863804e-08 \n",
       "                       MP_ISG MP_LYVE1hi.MHCII_intermediate \n",
       "                 3.020082e-22                  5.023053e-19 \n",
       "          MP_LYVE1hi.MHCII_lo           MP_LYVE1lo.MHCII_hi \n",
       "                 6.873844e-20                  1.237564e-20 \n",
       "                       MP_OSM               MP_transitional \n",
       "                 7.968652e-34                  6.617015e-22 \n",
       "                    MP_TREM2.                         NC1.0 \n",
       "                 5.760998e-04                  1.583425e-06 \n",
       "                        NC1.1                           NC2 \n",
       "                 1.195040e-13                  4.386623e-09 \n",
       "                    NK_CD16hi                     NK_CD56hi \n",
       "                 4.192044e-10                  4.908185e-08 \n",
       "                          PC1                           PC2 \n",
       "                 7.982795e-09                  1.116936e-40 \n",
       "                          PC3                        SMC1.1 \n",
       "                 7.455382e-17                  3.260953e-17 \n",
       "                       SMC1.2                          SMC2 \n",
       "                 8.129717e-25                  5.301672e-18 \n",
       "                       vCM1.0                        vCM1.1 \n",
       "                 2.842979e-07                  6.757448e-11 \n",
       "                       vCM1.2                        vCM1.3 \n",
       "                 4.456490e-18                  3.221759e-11 \n",
       "                         vCM2                        vCM3.0 \n",
       "                 1.054776e-05                  6.261554e-07 \n",
       "                       vCM3.1                          vCM4 \n",
       "                 9.662407e-08                  1.943686e-10 \n",
       "                         vCM5                        vFB1.0 \n",
       "                 5.063679e-15                  2.746608e-08 \n",
       "                       vFB1.1                        vFB1.2 \n",
       "                 2.422709e-33                  7.071880e-23 \n",
       "                         vFB2                          vFB3 \n",
       "                 9.455776e-11                  5.445674e-20 \n",
       "                         vFB4 \n",
       "                 4.166872e-16 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tapply(final_df[,\"FDR\"], \n",
    "       final_df[,'cell_state'], \n",
    "       min)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7d01adb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(final_df, \"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALLCELLSTATES_ALLGENOTYPES_EDGER_RV_V8.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "2b7fc162",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td> 1.2205588</td><td>4.281900</td><td>0.8834923</td><td>0.3472479</td><td>0.7936247</td><td>F</td><td>0.100384830</td><td>0.016013072</td><td>0.04910714</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td> 1.7543292</td><td>4.103563</td><td>1.6803753</td><td>0.1948759</td><td>0.6913935</td><td>F</td><td>0.160274735</td><td>0.005555556</td><td>0.02395833</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>A1CF    </td><td>-1.6756031</td><td>3.987134</td><td>1.1340045</td><td>0.3021312</td><td>0.7597694</td><td>F</td><td>0.119318197</td><td>0.027777778</td><td>0.00000000</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>A2M     </td><td>-0.1773584</td><td>6.034037</td><td>0.1038335</td><td>0.7472772</td><td>0.9772614</td><td>F</td><td>0.009989273</td><td>0.398457378</td><td>0.29449405</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>A2M-AS1 </td><td>-0.5136542</td><td>4.170491</td><td>0.1565951</td><td>0.6923112</td><td>0.9617922</td><td>F</td><td>0.016918769</td><td>0.027777778</td><td>0.01785714</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>A2ML1   </td><td>-2.0343873</td><td>4.098135</td><td>1.9439397</td><td>0.1632427</td><td>0.6913935</td><td>F</td><td>0.160274735</td><td>0.019394714</td><td>0.00000000</td><td>RV</td><td>CELLSTATES</td><td>AD2</td><td>AD</td><td>control_LMNA</td><td>4</td><td>6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     &  1.2205588 & 4.281900 & 0.8834923 & 0.3472479 & 0.7936247 & F & 0.100384830 & 0.016013072 & 0.04910714 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\t2 & A1BG-AS1 &  1.7543292 & 4.103563 & 1.6803753 & 0.1948759 & 0.6913935 & F & 0.160274735 & 0.005555556 & 0.02395833 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\t3 & A1CF     & -1.6756031 & 3.987134 & 1.1340045 & 0.3021312 & 0.7597694 & F & 0.119318197 & 0.027777778 & 0.00000000 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\t4 & A2M      & -0.1773584 & 6.034037 & 0.1038335 & 0.7472772 & 0.9772614 & F & 0.009989273 & 0.398457378 & 0.29449405 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\t5 & A2M-AS1  & -0.5136542 & 4.170491 & 0.1565951 & 0.6923112 & 0.9617922 & F & 0.016918769 & 0.027777778 & 0.01785714 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\t6 & A2ML1    & -2.0343873 & 4.098135 & 1.9439397 & 0.1632427 & 0.6913935 & F & 0.160274735 & 0.019394714 & 0.00000000 & RV & CELLSTATES & AD2 & AD & control\\_LMNA & 4 & 6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;chr&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;chr&gt; | annotation_level &lt;chr&gt; | cell_state &lt;chr&gt; | cell_type &lt;chr&gt; | comparison &lt;chr&gt; | Observations_genotype &lt;dbl&gt; | Observations_reference &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A1BG     |  1.2205588 | 4.281900 | 0.8834923 | 0.3472479 | 0.7936247 | F | 0.100384830 | 0.016013072 | 0.04910714 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "| 2 | A1BG-AS1 |  1.7543292 | 4.103563 | 1.6803753 | 0.1948759 | 0.6913935 | F | 0.160274735 | 0.005555556 | 0.02395833 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "| 3 | A1CF     | -1.6756031 | 3.987134 | 1.1340045 | 0.3021312 | 0.7597694 | F | 0.119318197 | 0.027777778 | 0.00000000 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "| 4 | A2M      | -0.1773584 | 6.034037 | 0.1038335 | 0.7472772 | 0.9772614 | F | 0.009989273 | 0.398457378 | 0.29449405 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "| 5 | A2M-AS1  | -0.5136542 | 4.170491 | 0.1565951 | 0.6923112 | 0.9617922 | F | 0.016918769 | 0.027777778 | 0.01785714 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "| 6 | A2ML1    | -2.0343873 | 4.098135 | 1.9439397 | 0.1632427 | 0.6913935 | F | 0.160274735 | 0.019394714 | 0.00000000 | RV | CELLSTATES | AD2 | AD | control_LMNA | 4 | 6 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene     logFC      logCPM   F         PValue    FDR       low_expression\n",
       "1 A1BG      1.2205588 4.281900 0.8834923 0.3472479 0.7936247 F             \n",
       "2 A1BG-AS1  1.7543292 4.103563 1.6803753 0.1948759 0.6913935 F             \n",
       "3 A1CF     -1.6756031 3.987134 1.1340045 0.3021312 0.7597694 F             \n",
       "4 A2M      -0.1773584 6.034037 0.1038335 0.7472772 0.9772614 F             \n",
       "5 A2M-AS1  -0.5136542 4.170491 0.1565951 0.6923112 0.9617922 F             \n",
       "6 A2ML1    -2.0343873 4.098135 1.9439397 0.1632427 0.6913935 F             \n",
       "  FDR_plot    mean_exp_control mean_exp_genotype Region annotation_level\n",
       "1 0.100384830 0.016013072      0.04910714        RV     CELLSTATES      \n",
       "2 0.160274735 0.005555556      0.02395833        RV     CELLSTATES      \n",
       "3 0.119318197 0.027777778      0.00000000        RV     CELLSTATES      \n",
       "4 0.009989273 0.398457378      0.29449405        RV     CELLSTATES      \n",
       "5 0.016918769 0.027777778      0.01785714        RV     CELLSTATES      \n",
       "6 0.160274735 0.019394714      0.00000000        RV     CELLSTATES      \n",
       "  cell_state cell_type comparison   Observations_genotype\n",
       "1 AD2        AD        control_LMNA 4                    \n",
       "2 AD2        AD        control_LMNA 4                    \n",
       "3 AD2        AD        control_LMNA 4                    \n",
       "4 AD2        AD        control_LMNA 4                    \n",
       "5 AD2        AD        control_LMNA 4                    \n",
       "6 AD2        AD        control_LMNA 4                    \n",
       "  Observations_reference\n",
       "1 6                     \n",
       "2 6                     \n",
       "3 6                     \n",
       "4 6                     \n",
       "5 6                     \n",
       "6 6                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(final_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e415deb",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e704f35e",
   "metadata": {},
   "source": [
    "# On cell-type level"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8f7c6c4",
   "metadata": {},
   "source": [
    "## For RV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0d3260c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "x=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLTYPE_PSEUDOBULK_RV.csv\")\n",
    "genes_tofilter=read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALL_CELLTYPE_PSEUDOBULK_FILTERING_RV.csv\")\n",
    "colnames(genes_tofilter) <- gsub(\"mutation.negative\", 'PVneg', colnames(genes_tofilter))\n",
    "\n",
    "# Removes patients with only few (<10) nuclei numbers (as they might bias the analysis)\n",
    "CELLTYPE_FILTER <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ABSOLUTE_CELLTYPE_NUMBER_RV.csv\")\n",
    "\n",
    "\n",
    "rownames(x) <- x$X\n",
    "x <- x[,-1]\n",
    "l <- strsplit(colnames(x), \"__\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "b2fa7c4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This steps are needed to make names equal - space and + was converted to .\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\+\", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\" \", \"\\\\.\", CELLTYPE_FILTER$cell_states)\n",
    "#CELLTYPE_FILTER$cell_states <- gsub(\"\\\\-\", \"\\\\.\", CELLTYPE_FILTER$cell_states)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "84b99919",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.data <- t(as.data.frame(l))\n",
    "rownames(meta.data) <- colnames(x)\n",
    "\n",
    "meta.data <- as.data.frame(meta.data)\n",
    "colnames(meta.data)  <-  c(\"Genotype\", \"cell_type\", 'Patient', 'X10X_version', 'Gender')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "adda5674",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'AD'</li><li>'CM'</li><li>'EC'</li><li>'FB'</li><li>'Lymphoid'</li><li>'Mast'</li><li>'Mural'</li><li>'Myeloid'</li><li>'NC'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'AD'\n",
       "\\item 'CM'\n",
       "\\item 'EC'\n",
       "\\item 'FB'\n",
       "\\item 'Lymphoid'\n",
       "\\item 'Mast'\n",
       "\\item 'Mural'\n",
       "\\item 'Myeloid'\n",
       "\\item 'NC'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "\n",
       "**Levels**: 1. 'AD'\n",
       "2. 'CM'\n",
       "3. 'EC'\n",
       "4. 'FB'\n",
       "5. 'Lymphoid'\n",
       "6. 'Mast'\n",
       "7. 'Mural'\n",
       "8. 'Myeloid'\n",
       "9. 'NC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "factor(0)\n",
       "Levels: AD CM EC FB Lymphoid Mast Mural Myeloid NC"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CELLTYPE_FILTER[,1][which(!(CELLTYPE_FILTER[,1] %in% meta.data$cell_type))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "68d473c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# meta.data$cell_state <- gsub(\"CD4+T_act\", \"CD4.T_act\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD4+T_naive\", \"CD4.T_naive\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD4+T_reg\", \"CD4.T_reg\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD8+T_cytox\", \"CD8.T_cytox\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD8+T_em\", \"CD8.T_em\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD8+T_te\", \"CD8.T_te\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD8+T_trans\", \"CD8.T_trans\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"CD16+ Mo\", \"CD16..Mo\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"MAIT-like\", \"MAIT.like\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"prolif Lymphoids\", \"prolif.Lymphoids\", meta.data$cell_state)\n",
    "# meta.data$cell_state <- gsub(\"prolif Myeloids\", \"prolif.Myeloids\", meta.data$cell_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "8ba81ed9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>H14</th><th scope=col>H36</th><th scope=col>H39</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>29</td><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & H14 & H36 & H39 & H41 & H34 & H35\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 29 & 1 & 1 & 1 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| <!--/--> | H14 &lt;int&gt; | H36 &lt;int&gt; | H39 &lt;int&gt; | H41 &lt;int&gt; | H34 &lt;int&gt; | H35 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 29 | 1 | 1 | 1 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  H14 H36 H39 H41 H34 H35\n",
       "1 29  1   1   1   0   0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, LMNA in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data$Genotype==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "e91afe98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>D1</th><th scope=col>D2</th><th scope=col>D4</th><th scope=col>D5</th><th scope=col>D6</th><th scope=col>D7</th><th scope=col>ED_H15</th><th scope=col>ED_H20</th><th scope=col>ED_H25</th><th scope=col>ED_H26</th><th scope=col>ED_H35</th><th scope=col>H49</th><th scope=col>H53</th><th scope=col>H55</th><th scope=col>ED_H37</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>10</td><td>2</td><td>38</td><td>3</td><td>12</td><td>13</td><td>1</td><td>80</td><td>10</td><td>89</td><td>25</td><td>97</td><td>19</td><td>4</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & D1 & D2 & D4 & D5 & D6 & D7 & ED\\_H15 & ED\\_H20 & ED\\_H25 & ED\\_H26 & ED\\_H35 & H49 & H53 & H55 & ED\\_H37\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 10 & 2 & 38 & 3 & 12 & 13 & 1 & 80 & 10 & 89 & 25 & 97 & 19 & 4 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 15\n",
       "\n",
       "| <!--/--> | D1 &lt;int&gt; | D2 &lt;int&gt; | D4 &lt;int&gt; | D5 &lt;int&gt; | D6 &lt;int&gt; | D7 &lt;int&gt; | ED_H15 &lt;int&gt; | ED_H20 &lt;int&gt; | ED_H25 &lt;int&gt; | ED_H26 &lt;int&gt; | ED_H35 &lt;int&gt; | H49 &lt;int&gt; | H53 &lt;int&gt; | H55 &lt;int&gt; | ED_H37 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 10 | 2 | 38 | 3 | 12 | 13 | 1 | 80 | 10 | 89 | 25 | 97 | 19 | 4 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  D1 D2 D4 D5 D6 D7 ED_H15 ED_H20 ED_H25 ED_H26 ED_H35 H49 H53 H55 ED_H37\n",
       "1 10 2  38 3  12 13 1      80     10     89     25     97  19  4   0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Numbers of observations in total for Adipocytes, control in RV\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data$Genotype==\"control\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "4456c750",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Genotype</th><th scope=col>cell_type</th><th scope=col>Patient</th><th scope=col>X10X_version</th><th scope=col>Gender</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LMNA__AD__H06__V3__f</th><td>LMNA</td><td>AD</td><td>H06</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD__H24__V3__f</th><td>LMNA</td><td>AD</td><td>H24</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD__H25__V3__m</th><td>LMNA</td><td>AD</td><td>H25</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD__H27__V3__f</th><td>LMNA</td><td>AD</td><td>H27</td><td>V3</td><td>f</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD__H28__V3__m</th><td>LMNA</td><td>AD</td><td>H28</td><td>V3</td><td>m</td></tr>\n",
       "\t<tr><th scope=row>LMNA__AD__H29__V3__f</th><td>LMNA</td><td>AD</td><td>H29</td><td>V3</td><td>f</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Genotype & cell\\_type & Patient & X10X\\_version & Gender\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tLMNA\\_\\_AD\\_\\_H06\\_\\_V3\\_\\_f & LMNA & AD & H06 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD\\_\\_H24\\_\\_V3\\_\\_f & LMNA & AD & H24 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD\\_\\_H25\\_\\_V3\\_\\_m & LMNA & AD & H25 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD\\_\\_H27\\_\\_V3\\_\\_f & LMNA & AD & H27 & V3 & f\\\\\n",
       "\tLMNA\\_\\_AD\\_\\_H28\\_\\_V3\\_\\_m & LMNA & AD & H28 & V3 & m\\\\\n",
       "\tLMNA\\_\\_AD\\_\\_H29\\_\\_V3\\_\\_f & LMNA & AD & H29 & V3 & f\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | Genotype &lt;fct&gt; | cell_type &lt;fct&gt; | Patient &lt;fct&gt; | X10X_version &lt;fct&gt; | Gender &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| LMNA__AD__H06__V3__f | LMNA | AD | H06 | V3 | f |\n",
       "| LMNA__AD__H24__V3__f | LMNA | AD | H24 | V3 | f |\n",
       "| LMNA__AD__H25__V3__m | LMNA | AD | H25 | V3 | m |\n",
       "| LMNA__AD__H27__V3__f | LMNA | AD | H27 | V3 | f |\n",
       "| LMNA__AD__H28__V3__m | LMNA | AD | H28 | V3 | m |\n",
       "| LMNA__AD__H29__V3__f | LMNA | AD | H29 | V3 | f |\n",
       "\n"
      ],
      "text/plain": [
       "                     Genotype cell_type Patient X10X_version Gender\n",
       "LMNA__AD__H06__V3__f LMNA     AD        H06     V3           f     \n",
       "LMNA__AD__H24__V3__f LMNA     AD        H24     V3           f     \n",
       "LMNA__AD__H25__V3__m LMNA     AD        H25     V3           m     \n",
       "LMNA__AD__H27__V3__f LMNA     AD        H27     V3           f     \n",
       "LMNA__AD__H28__V3__m LMNA     AD        H28     V3           m     \n",
       "LMNA__AD__H29__V3__f LMNA     AD        H29     V3           f     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meta.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "c3d1f732",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>LMNA</li><li>PKP2</li><li>PVneg</li><li>RBM20</li><li>TTN</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'control'</li><li>'LMNA'</li><li>'PKP2'</li><li>'PVneg'</li><li>'RBM20'</li><li>'TTN'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item LMNA\n",
       "\\item PKP2\n",
       "\\item PVneg\n",
       "\\item RBM20\n",
       "\\item TTN\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'control'\n",
       "\\item 'LMNA'\n",
       "\\item 'PKP2'\n",
       "\\item 'PVneg'\n",
       "\\item 'RBM20'\n",
       "\\item 'TTN'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. LMNA\n",
       "2. PKP2\n",
       "3. PVneg\n",
       "4. RBM20\n",
       "5. TTN\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'control'\n",
       "2. 'LMNA'\n",
       "3. 'PKP2'\n",
       "4. 'PVneg'\n",
       "5. 'RBM20'\n",
       "6. 'TTN'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] LMNA  PKP2  PVneg RBM20 TTN  \n",
       "Levels: control LMNA PKP2 PVneg RBM20 TTN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GENOTYPES <- unique(meta.data$Genotype[meta.data$Genotype != \"control\"])\n",
    "GENOTYPES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "531bd0e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "CELL_LEVEL <- \"CELLTYPE\"\n",
    "REGION <- \"RV\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "cb34b2d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########\n",
    "# Convert to bool\n",
    "CELLTYPE_FILTER <- CELLTYPE_FILTER\n",
    "CELLTYPE_FILTER[,2:ncol(CELLTYPE_FILTER)] <- apply(CELLTYPE_FILTER[,-1], 2, function(i) i>=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "d49b8791",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>H14</li><li>H36</li><li>H39</li><li>H41</li><li>H34</li><li>H35</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'D1'</li><li>'D2'</li><li>'D4'</li><li>'D5'</li><li>'D6'</li><li>'D7'</li><li>'ED_H15'</li><li>'ED_H20'</li><li>'ED_H25'</li><li>'ED_H26'</li><li>'ED_H35'</li><li>'ED_H37'</li><li>'H01'</li><li>'H02'</li><li>'H03'</li><li>'H06'</li><li>'H07'</li><li>'H08'</li><li>'H09'</li><li>'H10'</li><li>'H11'</li><li>'H14'</li><li>'H15'</li><li>'H19'</li><li>'H20'</li><li>'H22'</li><li>'H24'</li><li>'H25'</li><li>'H27'</li><li>'H28'</li><li>'H29'</li><li>'H31'</li><li>'H33'</li><li>'H34'</li><li>'H35'</li><li>'H36'</li><li>'H39'</li><li>'H41'</li><li>'H49'</li><li>'H53'</li><li>'H55'</li><li>'IC_H04'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item H14\n",
       "\\item H36\n",
       "\\item H39\n",
       "\\item H41\n",
       "\\item H34\n",
       "\\item H35\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'D1'\n",
       "\\item 'D2'\n",
       "\\item 'D4'\n",
       "\\item 'D5'\n",
       "\\item 'D6'\n",
       "\\item 'D7'\n",
       "\\item 'ED\\_H15'\n",
       "\\item 'ED\\_H20'\n",
       "\\item 'ED\\_H25'\n",
       "\\item 'ED\\_H26'\n",
       "\\item 'ED\\_H35'\n",
       "\\item 'ED\\_H37'\n",
       "\\item 'H01'\n",
       "\\item 'H02'\n",
       "\\item 'H03'\n",
       "\\item 'H06'\n",
       "\\item 'H07'\n",
       "\\item 'H08'\n",
       "\\item 'H09'\n",
       "\\item 'H10'\n",
       "\\item 'H11'\n",
       "\\item 'H14'\n",
       "\\item 'H15'\n",
       "\\item 'H19'\n",
       "\\item 'H20'\n",
       "\\item 'H22'\n",
       "\\item 'H24'\n",
       "\\item 'H25'\n",
       "\\item 'H27'\n",
       "\\item 'H28'\n",
       "\\item 'H29'\n",
       "\\item 'H31'\n",
       "\\item 'H33'\n",
       "\\item 'H34'\n",
       "\\item 'H35'\n",
       "\\item 'H36'\n",
       "\\item 'H39'\n",
       "\\item 'H41'\n",
       "\\item 'H49'\n",
       "\\item 'H53'\n",
       "\\item 'H55'\n",
       "\\item 'IC\\_H04'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. H14\n",
       "2. H36\n",
       "3. H39\n",
       "4. H41\n",
       "5. H34\n",
       "6. H35\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'D1'\n",
       "2. 'D2'\n",
       "3. 'D4'\n",
       "4. 'D5'\n",
       "5. 'D6'\n",
       "6. 'D7'\n",
       "7. 'ED_H15'\n",
       "8. 'ED_H20'\n",
       "9. 'ED_H25'\n",
       "10. 'ED_H26'\n",
       "11. 'ED_H35'\n",
       "12. 'ED_H37'\n",
       "13. 'H01'\n",
       "14. 'H02'\n",
       "15. 'H03'\n",
       "16. 'H06'\n",
       "17. 'H07'\n",
       "18. 'H08'\n",
       "19. 'H09'\n",
       "20. 'H10'\n",
       "21. 'H11'\n",
       "22. 'H14'\n",
       "23. 'H15'\n",
       "24. 'H19'\n",
       "25. 'H20'\n",
       "26. 'H22'\n",
       "27. 'H24'\n",
       "28. 'H25'\n",
       "29. 'H27'\n",
       "30. 'H28'\n",
       "31. 'H29'\n",
       "32. 'H31'\n",
       "33. 'H33'\n",
       "34. 'H34'\n",
       "35. 'H35'\n",
       "36. 'H36'\n",
       "37. 'H39'\n",
       "38. 'H41'\n",
       "39. 'H49'\n",
       "40. 'H53'\n",
       "41. 'H55'\n",
       "42. 'IC_H04'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] H14 H36 H39 H41 H34 H35\n",
       "42 Levels: D1 D2 D4 D5 D6 D7 ED_H15 ED_H20 ED_H25 ED_H26 ED_H35 ED_H37 ... IC_H04"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(meta.data[which(meta.data$Genotype==\"TTN\"),\"Patient\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "1c371935",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>H14</th><th scope=col>H36</th><th scope=col>H39</th><th scope=col>H41</th><th scope=col>H34</th><th scope=col>H35</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TRUE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & H14 & H36 & H39 & H41 & H34 & H35\\\\\n",
       "  & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & TRUE & FALSE & FALSE & FALSE & FALSE & FALSE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| <!--/--> | H14 &lt;lgl&gt; | H36 &lt;lgl&gt; | H39 &lt;lgl&gt; | H41 &lt;lgl&gt; | H34 &lt;lgl&gt; | H35 &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | TRUE | FALSE | FALSE | FALSE | FALSE | FALSE |\n",
       "\n"
      ],
      "text/plain": [
       "  H14  H36   H39   H41   H34   H35  \n",
       "1 TRUE FALSE FALSE FALSE FALSE FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# This is the example from above, they will be excluded in downstream analysis\n",
    "# In this case, the comparison will be excluded, as only one sample has enough observations\n",
    "CELLTYPE_FILTER[1,as.character(unique(meta.data[which(meta.data$Genotype==\"TTN\"),\"Patient\"]))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "28cc3d47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'AD'</li><li>'CM'</li><li>'EC'</li><li>'FB'</li><li>'Lymphoid'</li><li>'Mast'</li><li>'Mural'</li><li>'Myeloid'</li><li>'NC'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'AD'\n",
       "\\item 'CM'\n",
       "\\item 'EC'\n",
       "\\item 'FB'\n",
       "\\item 'Lymphoid'\n",
       "\\item 'Mast'\n",
       "\\item 'Mural'\n",
       "\\item 'Myeloid'\n",
       "\\item 'NC'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'AD'\n",
       "2. 'CM'\n",
       "3. 'EC'\n",
       "4. 'FB'\n",
       "5. 'Lymphoid'\n",
       "6. 'Mast'\n",
       "7. 'Mural'\n",
       "8. 'Myeloid'\n",
       "9. 'NC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"AD\"       \"CM\"       \"EC\"       \"FB\"       \"Lymphoid\" \"Mast\"     \"Mural\"   \n",
       "[8] \"Myeloid\"  \"NC\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.character(unique(meta.data$cell_type))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "43655e9b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "###START: LMNA #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: FB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Lymphoid #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mural #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Myeloid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC #####\n",
      "\n",
      "\n",
      "\n",
      "###START: PKP2 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: FB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Lymphoid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mural #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Myeloid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC #####\n",
      "\n",
      "\n",
      "\n",
      "###START: PVneg #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: FB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Lymphoid #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mural #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Myeloid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC #####\n",
      "\n",
      "\n",
      "\n",
      "###START: RBM20 #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: AD #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: FB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Lymphoid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mural #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Myeloid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC #####\n",
      "\n",
      "\n",
      "\n",
      "###START: TTN #####\n",
      "\n",
      "\n",
      "\n",
      "###Skip: AD #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: CM #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: EC #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: FB #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Lymphoid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mast #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Mural #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: Myeloid #####\n",
      "\n",
      "\n",
      "\n",
      "###FINISHED: NC #####\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "i <- 1\n",
    "\n",
    "for(GENOTYPE in as.character(GENOTYPES)){\n",
    "    message(\"\\n###START: \", GENOTYPE,\" #####\\n\")\n",
    "    \n",
    "    for(CELL_STATE in as.character(unique(meta.data$cell_type))){\n",
    "        \n",
    "        CONTROL_COLUMN <- paste(\"control_\",CELL_STATE, sep=\"\")\n",
    "        GENOTYPE_COLUMN <- paste(GENOTYPE,\"_\", CELL_STATE, sep=\"\")\n",
    "        CELLTYPE <- CELL_STATE #as.character(CELLTYPE_STATE[which(CELLTYPE_STATE$cell_type==CELL_STATE),\"cell_type\"])\n",
    "        \n",
    "        \n",
    "        x_sub <- x[,which(meta.data$Genotype %in% c(\"control\", GENOTYPE))]\n",
    "        meta.data_sub <- meta.data[which(meta.data$Genotype %in% c(\"control\", GENOTYPE)),]\n",
    "        \n",
    "        x_sub <- x_sub[,which(meta.data_sub$cell_type %in% c(CELL_STATE))]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$cell_type %in% c(CELL_STATE)),] \n",
    "        \n",
    "        ##### remove first line (cell_type column)\n",
    "        CELLTYPE_FILTER_SUB <- CELLTYPE_FILTER\n",
    "        PATIENTS_TOKEEP <- CELLTYPE_FILTER_SUB[which(CELLTYPE_FILTER_SUB$cell_type==CELL_STATE),-1]\n",
    "        PATIENTS_TOKEEP <- colnames(PATIENTS_TOKEEP)[which(as.logical(PATIENTS_TOKEEP))]\n",
    "        #####\n",
    "        \n",
    "        # Subset only for patients which fulfill condition\n",
    "        x_sub <- x_sub[,which(meta.data_sub$Patient %in% PATIENTS_TOKEEP)]\n",
    "        meta.data_sub <- meta.data_sub[which(meta.data_sub$Patient %in% PATIENTS_TOKEEP),]\n",
    "        meta.data_sub$Genotype = droplevels(meta.data_sub$Genotype) ## Drop factor levels\n",
    "   \n",
    "        \n",
    "        # The test is only run, if 2 conditions have at least 2 patients\n",
    "        if (length(unique(meta.data_sub$Genotype))==2 & \n",
    "            all(table(as.character(meta.data_sub$Genotype))>=3))\n",
    "           {\n",
    "             # Prepare DGEList object\n",
    "            meta.data_sub$Genotype <- as.factor(as.character(meta.data_sub$Genotype))\n",
    "            dge <- DGEList(counts=x_sub, group=meta.data_sub$Genotype)\n",
    "            \n",
    "            # Filter genes based on expression, the qlf Object will be filtered to exclude them from FDR calculation\n",
    "            genes_tofilter_sub_GENOTYPE <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% GENOTYPE &\n",
    "                                                                 gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                 gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "\n",
    "            genes_tofilter_sub_CONTROL <- genes_tofilter[,which(gsub(\"^(.*)__.*__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% 'control' &\n",
    "                                                                gsub(\"^.*__(.*)__.*$\", \"\\\\1\", colnames(genes_tofilter)) %in% CELL_STATE & \n",
    "                                                                gsub(\"^.*__.*__(.*)$\", \"\\\\1\", colnames(genes_tofilter)) %in% PATIENTS_TOKEEP)]\n",
    "            \n",
    "            genes_tofilter_sub <- data.frame(\"X\"=genes_tofilter[,\"X\"],\n",
    "                                             CONTROL_COLUMN=apply(genes_tofilter_sub_CONTROL, 1, mean),\n",
    "                                             GENOTYPE_COLUMN=apply(genes_tofilter_sub_GENOTYPE, 1, mean))\n",
    "            colnames(genes_tofilter_sub) <- c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)\n",
    "            \n",
    "            keep <- which(genes_tofilter_sub[,CONTROL_COLUMN] > 0.0125 |\n",
    "                          genes_tofilter_sub[,GENOTYPE_COLUMN] > 0.0125)\n",
    "            \n",
    "            #keep <- genes_tofilter[,CONTROL_COLUMN]>0.0125 | ##### change & to | ##### \n",
    "            #        genes_tofilter[,GENOTYPE_COLUMN]>0.0125\n",
    "            #dge <- dge[keep, , keep.lib.sizes=FALSE]\n",
    "    \n",
    "            # PP, model matrix, https://www.nature.com/articles/nmeth.4612 (edgeRQLFDetRate)\n",
    "            dge <- calcNormFactors(dge)\n",
    "            cdr <- scale(colMeans(x_sub > 0))\n",
    "            design <- model.matrix(~ cdr + meta.data_sub$Genotype)\n",
    "    \n",
    "            # \n",
    "            dge <- estimateDisp(dge, design = design)\n",
    "            fit <- glmQLFit(dge, design = design)\n",
    "            qlf <- glmQLFTest(fit)\n",
    "    \n",
    "            # For all\n",
    "            tt <- topTags(qlf, n = Inf)\n",
    "            \n",
    "            # Only for \"expressed/detected\" genes\n",
    "            tt_filtered <- topTags(qlf[keep,], n = Inf)\n",
    "            \n",
    "            tt$table[,\"Gene\"] <- as.character(rownames(tt$table))\n",
    "            tt_filtered$table[,\"Gene\"] <- as.character(rownames(tt_filtered$table))\n",
    "            tt_filtered$table[,\"low_expression\"] <- \"F\" #####  Low-expression column ##### \n",
    "            tt_filtered$table[,\"FDR_plot\"] <- -log10(tt_filtered$table$FDR) #####  For plotting Volcano, here we use FDR, not pValue  ##### \n",
    "            \n",
    "            \n",
    "            tt_merged <- merge(tt$table[,c(\"Gene\", \"logFC\", \"logCPM\", 'F', 'PValue')], \n",
    "                               tt_filtered$table[,c(\"Gene\", \"FDR\", \"low_expression\", \"FDR_plot\")], \n",
    "                               by=\"Gene\", all=T)   \n",
    "            \n",
    "            # Replace NA FDRs with NA                   #####  ORDER CHANGED #####  \n",
    "            tt_merged[which(is.na(tt_merged$FDR)),\"FDR\"] <- 1 \n",
    "            \n",
    "            EXPRESSION_MEAN <- genes_tofilter_sub[,c(\"X\", CONTROL_COLUMN, GENOTYPE_COLUMN)]\n",
    "            colnames(EXPRESSION_MEAN) <- c(\"Gene\", \"mean_exp_control\", 'mean_exp_genotype')\n",
    "            tt_merged <- merge(tt_merged, EXPRESSION_MEAN, by=\"Gene\")\n",
    "    \n",
    "            \n",
    "            tt_merged[,\"Region\"] <- REGION\n",
    "            tt_merged[,\"annotation_level\"] <- CELL_LEVEL\n",
    "            tt_merged[,\"cell_state\"] <- CELL_STATE\n",
    "            tt_merged[,\"cell_type\"] <- CELLTYPE\n",
    "            tt_merged[,\"comparison\"] <- paste(\"control_\", GENOTYPE, sep=\"\")\n",
    "            \n",
    "            ###### Suggestion to add those columns\n",
    "            tt_merged[,\"Observations_genotype\"] <- as.numeric(table(meta.data_sub$Genotype)[GENOTYPE])\n",
    "            tt_merged[,\"Observations_reference\"] <- as.numeric(table(meta.data_sub$Genotype)['control'])\n",
    "            \n",
    "            # low expression add T\n",
    "            tt_merged[which(is.na(tt_merged$low_expression)),\"low_expression\"] <- \"T\"\n",
    "            \n",
    "            if(i==1){\n",
    "                 final_df <- tt_merged   \n",
    "            } else {\n",
    "                final_df <- rbind(final_df, tt_merged)\n",
    "            }\n",
    "            \n",
    "            i <- i + 1\n",
    "            message(\"\\n###FINISHED: \", CELL_STATE,\" #####\\n\")\n",
    "        } else {\n",
    "            message(\"\\n###Skip: \", CELL_STATE,\" #####\\n\")\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "37940b4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td> 0.5928048</td><td>3.613484</td><td>0.4485169</td><td>0.5508307</td><td>0.8964962</td><td>F</td><td>0.04745154</td><td>0.011645189</td><td>0.017549262</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td> 0.7467388</td><td>3.494549</td><td>0.8893044</td><td>0.4586736</td><td>1.0000000</td><td>T</td><td>        NA</td><td>0.003185451</td><td>0.007628843</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>A1CF    </td><td>-0.3983224</td><td>3.472606</td><td>0.1569457</td><td>0.7752702</td><td>1.0000000</td><td>T</td><td>        NA</td><td>0.012154523</td><td>0.003164557</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>A2M     </td><td>-0.4780581</td><td>5.389797</td><td>0.8279467</td><td>0.3683503</td><td>0.8103331</td><td>F</td><td>0.09133640</td><td>0.151950451</td><td>0.113212418</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>A2M-AS1 </td><td>-0.7838415</td><td>3.559792</td><td>0.5894344</td><td>0.4471737</td><td>0.8525847</td><td>F</td><td>0.06926247</td><td>0.013619099</td><td>0.008620690</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>A2ML1   </td><td> 0.3085760</td><td>3.813569</td><td>0.1445228</td><td>0.7058489</td><td>0.9488013</td><td>F</td><td>0.02282472</td><td>0.010749557</td><td>0.017122514</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     &  0.5928048 & 3.613484 & 0.4485169 & 0.5508307 & 0.8964962 & F & 0.04745154 & 0.011645189 & 0.017549262 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\t2 & A1BG-AS1 &  0.7467388 & 3.494549 & 0.8893044 & 0.4586736 & 1.0000000 & T &         NA & 0.003185451 & 0.007628843 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\t3 & A1CF     & -0.3983224 & 3.472606 & 0.1569457 & 0.7752702 & 1.0000000 & T &         NA & 0.012154523 & 0.003164557 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\t4 & A2M      & -0.4780581 & 5.389797 & 0.8279467 & 0.3683503 & 0.8103331 & F & 0.09133640 & 0.151950451 & 0.113212418 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\t5 & A2M-AS1  & -0.7838415 & 3.559792 & 0.5894344 & 0.4471737 & 0.8525847 & F & 0.06926247 & 0.013619099 & 0.008620690 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\t6 & A2ML1    &  0.3085760 & 3.813569 & 0.1445228 & 0.7058489 & 0.9488013 & F & 0.02282472 & 0.010749557 & 0.017122514 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;chr&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;chr&gt; | annotation_level &lt;chr&gt; | cell_state &lt;chr&gt; | cell_type &lt;chr&gt; | comparison &lt;chr&gt; | Observations_genotype &lt;dbl&gt; | Observations_reference &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A1BG     |  0.5928048 | 3.613484 | 0.4485169 | 0.5508307 | 0.8964962 | F | 0.04745154 | 0.011645189 | 0.017549262 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "| 2 | A1BG-AS1 |  0.7467388 | 3.494549 | 0.8893044 | 0.4586736 | 1.0000000 | T |         NA | 0.003185451 | 0.007628843 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "| 3 | A1CF     | -0.3983224 | 3.472606 | 0.1569457 | 0.7752702 | 1.0000000 | T |         NA | 0.012154523 | 0.003164557 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "| 4 | A2M      | -0.4780581 | 5.389797 | 0.8279467 | 0.3683503 | 0.8103331 | F | 0.09133640 | 0.151950451 | 0.113212418 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "| 5 | A2M-AS1  | -0.7838415 | 3.559792 | 0.5894344 | 0.4471737 | 0.8525847 | F | 0.06926247 | 0.013619099 | 0.008620690 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "| 6 | A2ML1    |  0.3085760 | 3.813569 | 0.1445228 | 0.7058489 | 0.9488013 | F | 0.02282472 | 0.010749557 | 0.017122514 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene     logFC      logCPM   F         PValue    FDR       low_expression\n",
       "1 A1BG      0.5928048 3.613484 0.4485169 0.5508307 0.8964962 F             \n",
       "2 A1BG-AS1  0.7467388 3.494549 0.8893044 0.4586736 1.0000000 T             \n",
       "3 A1CF     -0.3983224 3.472606 0.1569457 0.7752702 1.0000000 T             \n",
       "4 A2M      -0.4780581 5.389797 0.8279467 0.3683503 0.8103331 F             \n",
       "5 A2M-AS1  -0.7838415 3.559792 0.5894344 0.4471737 0.8525847 F             \n",
       "6 A2ML1     0.3085760 3.813569 0.1445228 0.7058489 0.9488013 F             \n",
       "  FDR_plot   mean_exp_control mean_exp_genotype Region annotation_level\n",
       "1 0.04745154 0.011645189      0.017549262       RV     CELLTYPE        \n",
       "2         NA 0.003185451      0.007628843       RV     CELLTYPE        \n",
       "3         NA 0.012154523      0.003164557       RV     CELLTYPE        \n",
       "4 0.09133640 0.151950451      0.113212418       RV     CELLTYPE        \n",
       "5 0.06926247 0.013619099      0.008620690       RV     CELLTYPE        \n",
       "6 0.02282472 0.010749557      0.017122514       RV     CELLTYPE        \n",
       "  cell_state cell_type comparison   Observations_genotype\n",
       "1 AD         AD        control_LMNA 4                    \n",
       "2 AD         AD        control_LMNA 4                    \n",
       "3 AD         AD        control_LMNA 4                    \n",
       "4 AD         AD        control_LMNA 4                    \n",
       "5 AD         AD        control_LMNA 4                    \n",
       "6 AD         AD        control_LMNA 4                    \n",
       "  Observations_reference\n",
       "1 10                    \n",
       "2 10                    \n",
       "3 10                    \n",
       "4 10                    \n",
       "5 10                    \n",
       "6 10                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(final_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "3798579e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1408591</th><td>ZXDB  </td><td> 1.12076971</td><td>5.227782</td><td>4.54494520</td><td>0.1431496</td><td>1.0000000</td><td>T</td><td>         NA</td><td>0.00000000</td><td>0.004504505</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408592</th><td>ZXDC  </td><td> 0.41384666</td><td>7.213291</td><td>0.95890384</td><td>0.3274636</td><td>0.8265347</td><td>F</td><td>0.082738903</td><td>0.14185210</td><td>0.240512273</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408593</th><td>ZYG11A</td><td>-1.13629790</td><td>5.304394</td><td>3.36949385</td><td>0.1663576</td><td>0.6800895</td><td>F</td><td>0.167433910</td><td>0.01607527</td><td>0.000000000</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408594</th><td>ZYG11B</td><td> 0.10468329</td><td>6.849860</td><td>0.05170007</td><td>0.8201311</td><td>0.9890962</td><td>F</td><td>0.004761453</td><td>0.11172083</td><td>0.129940881</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408595</th><td>ZYX   </td><td>-1.21363157</td><td>5.608672</td><td>2.25447012</td><td>0.1332296</td><td>0.6313218</td><td>F</td><td>0.199749238</td><td>0.02662245</td><td>0.007905865</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408596</th><td>ZZEF1 </td><td> 0.09761451</td><td>7.200671</td><td>0.05165851</td><td>0.8202021</td><td>0.9890962</td><td>F</td><td>0.004761453</td><td>0.15385156</td><td>0.222772332</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1408591 & ZXDB   &  1.12076971 & 5.227782 & 4.54494520 & 0.1431496 & 1.0000000 & T &          NA & 0.00000000 & 0.004504505 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408592 & ZXDC   &  0.41384666 & 7.213291 & 0.95890384 & 0.3274636 & 0.8265347 & F & 0.082738903 & 0.14185210 & 0.240512273 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408593 & ZYG11A & -1.13629790 & 5.304394 & 3.36949385 & 0.1663576 & 0.6800895 & F & 0.167433910 & 0.01607527 & 0.000000000 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408594 & ZYG11B &  0.10468329 & 6.849860 & 0.05170007 & 0.8201311 & 0.9890962 & F & 0.004761453 & 0.11172083 & 0.129940881 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408595 & ZYX    & -1.21363157 & 5.608672 & 2.25447012 & 0.1332296 & 0.6313218 & F & 0.199749238 & 0.02662245 & 0.007905865 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408596 & ZZEF1  &  0.09761451 & 7.200671 & 0.05165851 & 0.8202021 & 0.9890962 & F & 0.004761453 & 0.15385156 & 0.222772332 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;chr&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;chr&gt; | annotation_level &lt;chr&gt; | cell_state &lt;chr&gt; | cell_type &lt;chr&gt; | comparison &lt;chr&gt; | Observations_genotype &lt;dbl&gt; | Observations_reference &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1408591 | ZXDB   |  1.12076971 | 5.227782 | 4.54494520 | 0.1431496 | 1.0000000 | T |          NA | 0.00000000 | 0.004504505 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408592 | ZXDC   |  0.41384666 | 7.213291 | 0.95890384 | 0.3274636 | 0.8265347 | F | 0.082738903 | 0.14185210 | 0.240512273 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408593 | ZYG11A | -1.13629790 | 5.304394 | 3.36949385 | 0.1663576 | 0.6800895 | F | 0.167433910 | 0.01607527 | 0.000000000 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408594 | ZYG11B |  0.10468329 | 6.849860 | 0.05170007 | 0.8201311 | 0.9890962 | F | 0.004761453 | 0.11172083 | 0.129940881 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408595 | ZYX    | -1.21363157 | 5.608672 | 2.25447012 | 0.1332296 | 0.6313218 | F | 0.199749238 | 0.02662245 | 0.007905865 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408596 | ZZEF1  |  0.09761451 | 7.200671 | 0.05165851 | 0.8202021 | 0.9890962 | F | 0.004761453 | 0.15385156 | 0.222772332 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "\n"
      ],
      "text/plain": [
       "        Gene   logFC       logCPM   F          PValue    FDR      \n",
       "1408591 ZXDB    1.12076971 5.227782 4.54494520 0.1431496 1.0000000\n",
       "1408592 ZXDC    0.41384666 7.213291 0.95890384 0.3274636 0.8265347\n",
       "1408593 ZYG11A -1.13629790 5.304394 3.36949385 0.1663576 0.6800895\n",
       "1408594 ZYG11B  0.10468329 6.849860 0.05170007 0.8201311 0.9890962\n",
       "1408595 ZYX    -1.21363157 5.608672 2.25447012 0.1332296 0.6313218\n",
       "1408596 ZZEF1   0.09761451 7.200671 0.05165851 0.8202021 0.9890962\n",
       "        low_expression FDR_plot    mean_exp_control mean_exp_genotype Region\n",
       "1408591 T                       NA 0.00000000       0.004504505       RV    \n",
       "1408592 F              0.082738903 0.14185210       0.240512273       RV    \n",
       "1408593 F              0.167433910 0.01607527       0.000000000       RV    \n",
       "1408594 F              0.004761453 0.11172083       0.129940881       RV    \n",
       "1408595 F              0.199749238 0.02662245       0.007905865       RV    \n",
       "1408596 F              0.004761453 0.15385156       0.222772332       RV    \n",
       "        annotation_level cell_state cell_type comparison  Observations_genotype\n",
       "1408591 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408592 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408593 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408594 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408595 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408596 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "        Observations_reference\n",
       "1408591 15                    \n",
       "1408592 15                    \n",
       "1408593 15                    \n",
       "1408594 15                    \n",
       "1408595 15                    \n",
       "1408596 15                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(final_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "4fdbdeea",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(final_df, \"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALLCELLTYPES_ALLGENOTYPES_EDGER_RV_V8.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c816f2a8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "7325a4de",
   "metadata": {},
   "outputs": [],
   "source": [
    "x <- read.csv(\"/fast/AG_Huebner/huebner3/ANALYSES/20190926_gp_BadOyenhausen/scanpy/20210302/Global_touse/ALLCELLTYPES_ALLGENOTYPES_EDGER_RV_V8.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "d20fc2de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1408591</th><td>1408591</td><td>ZXDB  </td><td> 1.12076971</td><td>5.227782</td><td>4.54494520</td><td>0.1431496</td><td>1.0000000</td><td> TRUE</td><td>         NA</td><td>0.00000000</td><td>0.004504505</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408592</th><td>1408592</td><td>ZXDC  </td><td> 0.41384666</td><td>7.213291</td><td>0.95890384</td><td>0.3274636</td><td>0.8265347</td><td>FALSE</td><td>0.082738903</td><td>0.14185210</td><td>0.240512273</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408593</th><td>1408593</td><td>ZYG11A</td><td>-1.13629790</td><td>5.304394</td><td>3.36949385</td><td>0.1663576</td><td>0.6800895</td><td>FALSE</td><td>0.167433910</td><td>0.01607527</td><td>0.000000000</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408594</th><td>1408594</td><td>ZYG11B</td><td> 0.10468329</td><td>6.849860</td><td>0.05170007</td><td>0.8201311</td><td>0.9890962</td><td>FALSE</td><td>0.004761453</td><td>0.11172083</td><td>0.129940881</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408595</th><td>1408595</td><td>ZYX   </td><td>-1.21363157</td><td>5.608672</td><td>2.25447012</td><td>0.1332296</td><td>0.6313218</td><td>FALSE</td><td>0.199749238</td><td>0.02662245</td><td>0.007905865</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>1408596</th><td>1408596</td><td>ZZEF1 </td><td> 0.09761451</td><td>7.200671</td><td>0.05165851</td><td>0.8202021</td><td>0.9890962</td><td>FALSE</td><td>0.004761453</td><td>0.15385156</td><td>0.222772332</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & X & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference\\\\\n",
       "  & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <lgl> & <dbl> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1408591 & 1408591 & ZXDB   &  1.12076971 & 5.227782 & 4.54494520 & 0.1431496 & 1.0000000 &  TRUE &          NA & 0.00000000 & 0.004504505 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408592 & 1408592 & ZXDC   &  0.41384666 & 7.213291 & 0.95890384 & 0.3274636 & 0.8265347 & FALSE & 0.082738903 & 0.14185210 & 0.240512273 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408593 & 1408593 & ZYG11A & -1.13629790 & 5.304394 & 3.36949385 & 0.1663576 & 0.6800895 & FALSE & 0.167433910 & 0.01607527 & 0.000000000 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408594 & 1408594 & ZYG11B &  0.10468329 & 6.849860 & 0.05170007 & 0.8201311 & 0.9890962 & FALSE & 0.004761453 & 0.11172083 & 0.129940881 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408595 & 1408595 & ZYX    & -1.21363157 & 5.608672 & 2.25447012 & 0.1332296 & 0.6313218 & FALSE & 0.199749238 & 0.02662245 & 0.007905865 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\t1408596 & 1408596 & ZZEF1  &  0.09761451 & 7.200671 & 0.05165851 & 0.8202021 & 0.9890962 & FALSE & 0.004761453 & 0.15385156 & 0.222772332 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | Gene &lt;fct&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;lgl&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;fct&gt; | annotation_level &lt;fct&gt; | cell_state &lt;fct&gt; | cell_type &lt;fct&gt; | comparison &lt;fct&gt; | Observations_genotype &lt;int&gt; | Observations_reference &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1408591 | 1408591 | ZXDB   |  1.12076971 | 5.227782 | 4.54494520 | 0.1431496 | 1.0000000 |  TRUE |          NA | 0.00000000 | 0.004504505 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408592 | 1408592 | ZXDC   |  0.41384666 | 7.213291 | 0.95890384 | 0.3274636 | 0.8265347 | FALSE | 0.082738903 | 0.14185210 | 0.240512273 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408593 | 1408593 | ZYG11A | -1.13629790 | 5.304394 | 3.36949385 | 0.1663576 | 0.6800895 | FALSE | 0.167433910 | 0.01607527 | 0.000000000 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408594 | 1408594 | ZYG11B |  0.10468329 | 6.849860 | 0.05170007 | 0.8201311 | 0.9890962 | FALSE | 0.004761453 | 0.11172083 | 0.129940881 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408595 | 1408595 | ZYX    | -1.21363157 | 5.608672 | 2.25447012 | 0.1332296 | 0.6313218 | FALSE | 0.199749238 | 0.02662245 | 0.007905865 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "| 1408596 | 1408596 | ZZEF1  |  0.09761451 | 7.200671 | 0.05165851 | 0.8202021 | 0.9890962 | FALSE | 0.004761453 | 0.15385156 | 0.222772332 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 |\n",
       "\n"
      ],
      "text/plain": [
       "        X       Gene   logFC       logCPM   F          PValue    FDR      \n",
       "1408591 1408591 ZXDB    1.12076971 5.227782 4.54494520 0.1431496 1.0000000\n",
       "1408592 1408592 ZXDC    0.41384666 7.213291 0.95890384 0.3274636 0.8265347\n",
       "1408593 1408593 ZYG11A -1.13629790 5.304394 3.36949385 0.1663576 0.6800895\n",
       "1408594 1408594 ZYG11B  0.10468329 6.849860 0.05170007 0.8201311 0.9890962\n",
       "1408595 1408595 ZYX    -1.21363157 5.608672 2.25447012 0.1332296 0.6313218\n",
       "1408596 1408596 ZZEF1   0.09761451 7.200671 0.05165851 0.8202021 0.9890962\n",
       "        low_expression FDR_plot    mean_exp_control mean_exp_genotype Region\n",
       "1408591  TRUE                   NA 0.00000000       0.004504505       RV    \n",
       "1408592 FALSE          0.082738903 0.14185210       0.240512273       RV    \n",
       "1408593 FALSE          0.167433910 0.01607527       0.000000000       RV    \n",
       "1408594 FALSE          0.004761453 0.11172083       0.129940881       RV    \n",
       "1408595 FALSE          0.199749238 0.02662245       0.007905865       RV    \n",
       "1408596 FALSE          0.004761453 0.15385156       0.222772332       RV    \n",
       "        annotation_level cell_state cell_type comparison  Observations_genotype\n",
       "1408591 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408592 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408593 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408594 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408595 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "1408596 CELLTYPE         NC         NC        control_TTN 6                    \n",
       "        Observations_reference\n",
       "1408591 15                    \n",
       "1408592 15                    \n",
       "1408593 15                    \n",
       "1408594 15                    \n",
       "1408595 15                    \n",
       "1408596 15                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "80c93426",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "10977"
      ],
      "text/latex": [
       "10977"
      ],
      "text/markdown": [
       "10977"
      ],
      "text/plain": [
       "[1] 10977"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(which(p.adjust(x[-which(is.na(x$FDR_plot)),\"PValue\"], method = 'BH')<0.05))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "e3474358",
   "metadata": {},
   "outputs": [],
   "source": [
    "x[-which(is.na(x$FDR_plot)),\"FDR_plot_all\"] <- p.adjust(x[-which(is.na(x$FDR_plot)),\n",
    "                                                          \"PValue\"], \n",
    "                                                        method = 'BH')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "29f81ade",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 511188 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>Gene</th><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>low_expression</th><th scope=col>FDR_plot</th><th scope=col>mean_exp_control</th><th scope=col>mean_exp_genotype</th><th scope=col>Region</th><th scope=col>annotation_level</th><th scope=col>cell_state</th><th scope=col>cell_type</th><th scope=col>comparison</th><th scope=col>Observations_genotype</th><th scope=col>Observations_reference</th><th scope=col>FDR_plot_all</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td> 1</td><td>A1BG      </td><td> 0.59280479</td><td>3.613484</td><td>0.44851693</td><td>0.550830654</td><td>0.8964962</td><td>FALSE</td><td>0.047451535</td><td>0.011645189</td><td>0.01754926</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.8767031</td></tr>\n",
       "\t<tr><th scope=row>4</th><td> 4</td><td>A2M       </td><td>-0.47805814</td><td>5.389797</td><td>0.82794671</td><td>0.368350327</td><td>0.8103331</td><td>FALSE</td><td>0.091336403</td><td>0.151950451</td><td>0.11321242</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.7785514</td></tr>\n",
       "\t<tr><th scope=row>5</th><td> 5</td><td>A2M-AS1   </td><td>-0.78384150</td><td>3.559792</td><td>0.58943441</td><td>0.447173742</td><td>0.8525847</td><td>FALSE</td><td>0.069262471</td><td>0.013619099</td><td>0.00862069</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.8266161</td></tr>\n",
       "\t<tr><th scope=row>6</th><td> 6</td><td>A2ML1     </td><td> 0.30857597</td><td>3.813569</td><td>0.14452276</td><td>0.705848923</td><td>0.9488013</td><td>FALSE</td><td>0.022824715</td><td>0.010749557</td><td>0.01712251</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9370670</td></tr>\n",
       "\t<tr><th scope=row>7</th><td> 7</td><td>A2ML1-AS1 </td><td> 0.57931432</td><td>5.658915</td><td>1.74265342</td><td>0.194352982</td><td>0.6787966</td><td>FALSE</td><td>0.168260364</td><td>0.156997997</td><td>0.24860482</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.6329451</td></tr>\n",
       "\t<tr><th scope=row>9</th><td> 9</td><td>A3GALT2   </td><td>-2.43383781</td><td>3.526761</td><td>5.50740195</td><td>0.049297917</td><td>0.4312155</td><td>FALSE</td><td>0.365305621</td><td>0.016359989</td><td>0.00000000</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.3699974</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>10</td><td>A4GALT    </td><td>-1.52364885</td><td>4.982904</td><td>6.05813642</td><td>0.018280381</td><td>0.3050396</td><td>FALSE</td><td>0.515643730</td><td>0.140615087</td><td>0.03783633</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.2375570</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>12</td><td>AAAS      </td><td>-0.46603918</td><td>4.709302</td><td>0.58753257</td><td>0.447902270</td><td>0.8529603</td><td>FALSE</td><td>0.069071167</td><td>0.084879806</td><td>0.05538560</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.8270879</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>13</td><td>AACS      </td><td>-0.98908185</td><td>7.550449</td><td>5.26392754</td><td>0.027132919</td><td>0.3449701</td><td>FALSE</td><td>0.462218499</td><td>1.043062838</td><td>0.52544506</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.2840057</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>19</td><td>AADAT     </td><td> 0.94386331</td><td>4.134502</td><td>1.62164242</td><td>0.262455054</td><td>0.7405559</td><td>FALSE</td><td>0.130442155</td><td>0.018837166</td><td>0.04744887</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.6992791</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>20</td><td>AAED1     </td><td>-0.24401663</td><td>5.161923</td><td>0.26371353</td><td>0.610422448</td><td>0.9190844</td><td>FALSE</td><td>0.036644615</td><td>0.145873508</td><td>0.11187294</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9018107</td></tr>\n",
       "\t<tr><th scope=row>21</th><td>21</td><td>AAGAB     </td><td> 0.45101359</td><td>5.855196</td><td>1.28131223</td><td>0.264436204</td><td>0.7426188</td><td>FALSE</td><td>0.129234031</td><td>0.218194138</td><td>0.28857486</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.7011770</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>22</td><td>AAK1      </td><td>-0.41885189</td><td>7.437454</td><td>1.56756773</td><td>0.217876102</td><td>0.6979025</td><td>FALSE</td><td>0.156205216</td><td>0.807023347</td><td>0.66403594</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.6573869</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>23</td><td>AAMDC     </td><td>-0.05066493</td><td>6.737111</td><td>0.02175286</td><td>0.883491843</td><td>0.9957384</td><td>FALSE</td><td>0.001854735</td><td>0.439556781</td><td>0.46552705</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9862150</td></tr>\n",
       "\t<tr><th scope=row>24</th><td>24</td><td>AAMP      </td><td> 0.88638900</td><td>5.954126</td><td>4.81878864</td><td>0.034051429</td><td>0.3709559</td><td>FALSE</td><td>0.430677673</td><td>0.184870489</td><td>0.38042888</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.3128976</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>26</td><td>AAR2      </td><td>-0.45415937</td><td>4.453402</td><td>0.46191759</td><td>0.500667238</td><td>0.8771219</td><td>FALSE</td><td>0.056940050</td><td>0.076662406</td><td>0.04974044</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.8535110</td></tr>\n",
       "\t<tr><th scope=row>28</th><td>28</td><td>AARS      </td><td> 0.10138538</td><td>5.832970</td><td>0.05575659</td><td>0.814545311</td><td>0.9831224</td><td>FALSE</td><td>0.007392397</td><td>0.210718786</td><td>0.23817032</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9698404</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>29</td><td>AARS2     </td><td>-0.25884368</td><td>3.969703</td><td>0.09998201</td><td>0.838490446</td><td>0.9899243</td><td>FALSE</td><td>0.004397996</td><td>0.025139571</td><td>0.02127892</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9760201</td></tr>\n",
       "\t<tr><th scope=row>31</th><td>31</td><td>AASDH     </td><td>-0.41023512</td><td>6.313453</td><td>1.17929368</td><td>0.284040294</td><td>0.7587664</td><td>FALSE</td><td>0.119891915</td><td>0.344242545</td><td>0.27855545</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.7177845</td></tr>\n",
       "\t<tr><th scope=row>32</th><td>32</td><td>AASDHPPT  </td><td>-0.11776054</td><td>4.854325</td><td>0.04887435</td><td>0.826165142</td><td>0.9863212</td><td>FALSE</td><td>0.005981612</td><td>0.096734620</td><td>0.09087111</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9728590</td></tr>\n",
       "\t<tr><th scope=row>33</th><td>33</td><td>AASS      </td><td> 1.34607718</td><td>8.199859</td><td>8.59974659</td><td>0.005554113</td><td>0.2009727</td><td>FALSE</td><td>0.696863035</td><td>0.789486746</td><td>2.46272870</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.1296984</td></tr>\n",
       "\t<tr><th scope=row>35</th><td>35</td><td>AATF      </td><td> 0.32030802</td><td>5.823508</td><td>0.54991830</td><td>0.462709147</td><td>0.8598546</td><td>FALSE</td><td>0.065574979</td><td>0.174846891</td><td>0.25512331</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.8348744</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>37</td><td>AB015752.1</td><td> 1.19703057</td><td>3.799611</td><td>2.36077590</td><td>0.161080081</td><td>0.6420055</td><td>FALSE</td><td>0.192461253</td><td>0.011167681</td><td>0.04025769</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.5925748</td></tr>\n",
       "\t<tr><th scope=row>39</th><td>39</td><td>ABAT      </td><td> 1.11303692</td><td>5.071607</td><td>4.23657117</td><td>0.046157659</td><td>0.4204372</td><td>FALSE</td><td>0.376298862</td><td>0.061287154</td><td>0.17958940</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.3591232</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>40</td><td>ABCA1     </td><td>-0.10690232</td><td>8.900441</td><td>0.05711192</td><td>0.812347923</td><td>0.9827512</td><td>FALSE</td><td>0.007556420</td><td>2.536833495</td><td>2.15172900</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9692521</td></tr>\n",
       "\t<tr><th scope=row>41</th><td>41</td><td>ABCA10    </td><td>-0.19649533</td><td>8.399351</td><td>0.23646518</td><td>0.629440533</td><td>0.9258420</td><td>FALSE</td><td>0.033463108</td><td>1.807205705</td><td>1.56555575</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9091768</td></tr>\n",
       "\t<tr><th scope=row>42</th><td>42</td><td>ABCA12    </td><td>-0.17838759</td><td>3.677913</td><td>0.02885796</td><td>0.865968860</td><td>0.9937067</td><td>FALSE</td><td>0.002741783</td><td>0.008870902</td><td>0.01308498</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.9824329</td></tr>\n",
       "\t<tr><th scope=row>44</th><td>44</td><td>ABCA2     </td><td>-1.10131876</td><td>4.836746</td><td>4.01166083</td><td>0.052037530</td><td>0.4373436</td><td>FALSE</td><td>0.359177197</td><td>0.095370814</td><td>0.05678431</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.3786983</td></tr>\n",
       "\t<tr><th scope=row>45</th><td>45</td><td>ABCA3     </td><td>-0.79701954</td><td>5.093562</td><td>1.90898289</td><td>0.174791298</td><td>0.6572500</td><td>FALSE</td><td>0.182269410</td><td>0.120983852</td><td>0.06592178</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.6099814</td></tr>\n",
       "\t<tr><th scope=row>47</th><td>47</td><td>ABCA5     </td><td> 0.54234585</td><td>8.193122</td><td>2.45393371</td><td>0.125156200</td><td>0.5883407</td><td>FALSE</td><td>0.230371114</td><td>1.146144429</td><td>1.85268285</td><td>RV</td><td>CELLTYPE</td><td>AD</td><td>AD</td><td>control_LMNA</td><td>4</td><td>10</td><td>0.5397835</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>1408550</th><td>1408550</td><td>ZRANB2     </td><td>-0.41898202</td><td>8.448639</td><td>1.26870387</td><td>0.26001042</td><td>0.7792071</td><td>FALSE</td><td>0.1083470826</td><td>0.495081987</td><td>0.445529033</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6969873</td></tr>\n",
       "\t<tr><th scope=row>1408551</th><td>1408551</td><td>ZRANB2-AS1 </td><td> 0.79287936</td><td>6.177717</td><td>1.84300242</td><td>0.17459996</td><td>0.6923383</td><td>FALSE</td><td>0.1596816366</td><td>0.036655743</td><td>0.088308936</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6097120</td></tr>\n",
       "\t<tr><th scope=row>1408552</th><td>1408552</td><td>ZRANB3     </td><td> 0.31956251</td><td>7.424698</td><td>0.62199530</td><td>0.43030709</td><td>0.8827371</td><td>FALSE</td><td>0.0541686033</td><td>0.168801299</td><td>0.293178722</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8168626</td></tr>\n",
       "\t<tr><th scope=row>1408553</th><td>1408553</td><td>ZRSR2      </td><td>-0.45465320</td><td>6.059947</td><td>0.51802073</td><td>0.47168749</td><td>0.8998568</td><td>FALSE</td><td>0.0458266101</td><td>0.054437668</td><td>0.038249592</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8393780</td></tr>\n",
       "\t<tr><th scope=row>1408556</th><td>1408556</td><td>ZSCAN12    </td><td>-0.74060434</td><td>5.633310</td><td>0.93613950</td><td>0.33327303</td><td>0.8303767</td><td>FALSE</td><td>0.0807248385</td><td>0.022631743</td><td>0.024572532</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7551393</td></tr>\n",
       "\t<tr><th scope=row>1408558</th><td>1408558</td><td>ZSCAN16-AS1</td><td> 0.62339750</td><td>6.207718</td><td>1.21660302</td><td>0.27002878</td><td>0.7862802</td><td>FALSE</td><td>0.1044226827</td><td>0.049068949</td><td>0.077743672</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7055624</td></tr>\n",
       "\t<tr><th scope=row>1408561</th><td>1408561</td><td>ZSCAN20    </td><td> 1.06303363</td><td>5.379492</td><td>2.06137799</td><td>0.17638007</td><td>0.6950039</td><td>FALSE</td><td>0.1580127864</td><td>0.006820119</td><td>0.013613779</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6117026</td></tr>\n",
       "\t<tr><th scope=row>1408562</th><td>1408562</td><td>ZSCAN21    </td><td>-1.00452090</td><td>5.752995</td><td>1.79168088</td><td>0.18072200</td><td>0.7001433</td><td>FALSE</td><td>0.1548130924</td><td>0.032959138</td><td>0.029816296</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6172248</td></tr>\n",
       "\t<tr><th scope=row>1408564</th><td>1408564</td><td>ZSCAN23    </td><td> 0.16046129</td><td>5.895627</td><td>0.05770854</td><td>0.81015513</td><td>0.9877482</td><td>FALSE</td><td>0.0053537579</td><td>0.029201649</td><td>0.061522312</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9686215</td></tr>\n",
       "\t<tr><th scope=row>1408565</th><td>1408565</td><td>ZSCAN25    </td><td>-1.54206513</td><td>5.631264</td><td>3.47778069</td><td>0.06219859</td><td>0.4786725</td><td>FALSE</td><td>0.3199615603</td><td>0.024957639</td><td>0.008236628</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.4087370</td></tr>\n",
       "\t<tr><th scope=row>1408568</th><td>1408568</td><td>ZSCAN30    </td><td> 0.05440543</td><td>6.379628</td><td>0.01055081</td><td>0.91818752</td><td>0.9996648</td><td>FALSE</td><td>0.0001455942</td><td>0.065791183</td><td>0.077565676</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9924597</td></tr>\n",
       "\t<tr><th scope=row>1408569</th><td>1408569</td><td>ZSCAN31    </td><td>-1.06271914</td><td>5.604548</td><td>1.64146815</td><td>0.34161908</td><td>0.8348459</td><td>FALSE</td><td>0.0783936885</td><td>0.020349848</td><td>0.008370015</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7605515</td></tr>\n",
       "\t<tr><th scope=row>1408570</th><td>1408570</td><td>ZSCAN32    </td><td> 0.81596942</td><td>5.540384</td><td>1.17296830</td><td>0.39077779</td><td>0.8642478</td><td>FALSE</td><td>0.0633617307</td><td>0.007653265</td><td>0.024787618</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7931896</td></tr>\n",
       "\t<tr><th scope=row>1408572</th><td>1408572</td><td>ZSCAN5A    </td><td>-0.67410217</td><td>6.681754</td><td>1.76446333</td><td>0.18406927</td><td>0.7041076</td><td>FALSE</td><td>0.1523609408</td><td>0.104571220</td><td>0.077512146</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6211056</td></tr>\n",
       "\t<tr><th scope=row>1408575</th><td>1408575</td><td>ZSCAN9     </td><td>-0.60970015</td><td>5.636025</td><td>0.63332216</td><td>0.42613965</td><td>0.8823821</td><td>FALSE</td><td>0.0543432973</td><td>0.017384014</td><td>0.018118284</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8145620</td></tr>\n",
       "\t<tr><th scope=row>1408578</th><td>1408578</td><td>ZSWIM3     </td><td> 0.56558939</td><td>5.754830</td><td>0.65249464</td><td>0.41922251</td><td>0.8802542</td><td>FALSE</td><td>0.0553918746</td><td>0.022269408</td><td>0.034635120</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8104435</td></tr>\n",
       "\t<tr><th scope=row>1408579</th><td>1408579</td><td>ZSWIM4     </td><td>-0.18332237</td><td>6.111745</td><td>0.08476571</td><td>0.77094014</td><td>0.9796231</td><td>FALSE</td><td>0.0089409995</td><td>0.052123460</td><td>0.042345600</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9574083</td></tr>\n",
       "\t<tr><th scope=row>1408580</th><td>1408580</td><td>ZSWIM5     </td><td> 0.09104139</td><td>6.300519</td><td>0.02733914</td><td>0.86867209</td><td>0.9958883</td><td>FALSE</td><td>0.0017893879</td><td>0.055440952</td><td>0.068932999</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9829380</td></tr>\n",
       "\t<tr><th scope=row>1408581</th><td>1408581</td><td>ZSWIM6     </td><td>-0.38265922</td><td>9.598415</td><td>1.18186247</td><td>0.27697759</td><td>0.7907358</td><td>FALSE</td><td>0.1019685740</td><td>1.367779867</td><td>1.119595615</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7116676</td></tr>\n",
       "\t<tr><th scope=row>1408582</th><td>1408582</td><td>ZSWIM7     </td><td> 0.65144509</td><td>6.723159</td><td>1.96087040</td><td>0.16142077</td><td>0.6746079</td><td>FALSE</td><td>0.1709485821</td><td>0.091570986</td><td>0.145823402</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.5931049</td></tr>\n",
       "\t<tr><th scope=row>1408583</th><td>1408583</td><td>ZSWIM8     </td><td> 0.59437810</td><td>6.493243</td><td>1.32810503</td><td>0.24914307</td><td>0.7687747</td><td>FALSE</td><td>0.1142009435</td><td>0.064281509</td><td>0.114732082</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6870315</td></tr>\n",
       "\t<tr><th scope=row>1408585</th><td>1408585</td><td>ZSWIM9     </td><td> 0.52355515</td><td>5.807250</td><td>0.60875500</td><td>0.43525726</td><td>0.8836248</td><td>FALSE</td><td>0.0537321259</td><td>0.031769727</td><td>0.036879675</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8195799</td></tr>\n",
       "\t<tr><th scope=row>1408586</th><td>1408586</td><td>ZUP1       </td><td>-1.16323146</td><td>5.843864</td><td>2.50731095</td><td>0.11331967</td><td>0.5977405</td><td>FALSE</td><td>0.2234872920</td><td>0.033584395</td><td>0.022733513</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.5204889</td></tr>\n",
       "\t<tr><th scope=row>1408587</th><td>1408587</td><td>ZW10       </td><td> 0.36458751</td><td>5.817240</td><td>0.27125688</td><td>0.60249008</td><td>0.9407317</td><td>FALSE</td><td>0.0265342062</td><td>0.019933530</td><td>0.053036620</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.8984469</td></tr>\n",
       "\t<tr><th scope=row>1408588</th><td>1408588</td><td>ZWILCH     </td><td> 1.10724111</td><td>5.399117</td><td>2.24174458</td><td>0.25572211</td><td>0.7762917</td><td>FALSE</td><td>0.1099750694</td><td>0.003225806</td><td>0.013946664</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.6931081</td></tr>\n",
       "\t<tr><th scope=row>1408592</th><td>1408592</td><td>ZXDC       </td><td> 0.41384666</td><td>7.213291</td><td>0.95890384</td><td>0.32746360</td><td>0.8265347</td><td>FALSE</td><td>0.0827389028</td><td>0.141852096</td><td>0.240512273</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.7509211</td></tr>\n",
       "\t<tr><th scope=row>1408593</th><td>1408593</td><td>ZYG11A     </td><td>-1.13629790</td><td>5.304394</td><td>3.36949385</td><td>0.16635761</td><td>0.6800895</td><td>FALSE</td><td>0.1674339095</td><td>0.016075269</td><td>0.000000000</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.5995422</td></tr>\n",
       "\t<tr><th scope=row>1408594</th><td>1408594</td><td>ZYG11B     </td><td> 0.10468329</td><td>6.849860</td><td>0.05170007</td><td>0.82013107</td><td>0.9890962</td><td>FALSE</td><td>0.0047614528</td><td>0.111720833</td><td>0.129940881</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9713227</td></tr>\n",
       "\t<tr><th scope=row>1408595</th><td>1408595</td><td>ZYX        </td><td>-1.21363157</td><td>5.608672</td><td>2.25447012</td><td>0.13322957</td><td>0.6313218</td><td>FALSE</td><td>0.1997492378</td><td>0.026622454</td><td>0.007905865</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.5531024</td></tr>\n",
       "\t<tr><th scope=row>1408596</th><td>1408596</td><td>ZZEF1      </td><td> 0.09761451</td><td>7.200671</td><td>0.05165851</td><td>0.82020214</td><td>0.9890962</td><td>FALSE</td><td>0.0047614528</td><td>0.153851562</td><td>0.222772332</td><td>RV</td><td>CELLTYPE</td><td>NC</td><td>NC</td><td>control_TTN</td><td>6</td><td>15</td><td>0.9713227</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 511188 × 19\n",
       "\\begin{tabular}{r|lllllllllllllllllll}\n",
       "  & X & Gene & logFC & logCPM & F & PValue & FDR & low\\_expression & FDR\\_plot & mean\\_exp\\_control & mean\\_exp\\_genotype & Region & annotation\\_level & cell\\_state & cell\\_type & comparison & Observations\\_genotype & Observations\\_reference & FDR\\_plot\\_all\\\\\n",
       "  & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <lgl> & <dbl> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 &  1 & A1BG       &  0.59280479 & 3.613484 & 0.44851693 & 0.550830654 & 0.8964962 & FALSE & 0.047451535 & 0.011645189 & 0.01754926 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.8767031\\\\\n",
       "\t4 &  4 & A2M        & -0.47805814 & 5.389797 & 0.82794671 & 0.368350327 & 0.8103331 & FALSE & 0.091336403 & 0.151950451 & 0.11321242 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.7785514\\\\\n",
       "\t5 &  5 & A2M-AS1    & -0.78384150 & 3.559792 & 0.58943441 & 0.447173742 & 0.8525847 & FALSE & 0.069262471 & 0.013619099 & 0.00862069 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.8266161\\\\\n",
       "\t6 &  6 & A2ML1      &  0.30857597 & 3.813569 & 0.14452276 & 0.705848923 & 0.9488013 & FALSE & 0.022824715 & 0.010749557 & 0.01712251 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9370670\\\\\n",
       "\t7 &  7 & A2ML1-AS1  &  0.57931432 & 5.658915 & 1.74265342 & 0.194352982 & 0.6787966 & FALSE & 0.168260364 & 0.156997997 & 0.24860482 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.6329451\\\\\n",
       "\t9 &  9 & A3GALT2    & -2.43383781 & 3.526761 & 5.50740195 & 0.049297917 & 0.4312155 & FALSE & 0.365305621 & 0.016359989 & 0.00000000 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.3699974\\\\\n",
       "\t10 & 10 & A4GALT     & -1.52364885 & 4.982904 & 6.05813642 & 0.018280381 & 0.3050396 & FALSE & 0.515643730 & 0.140615087 & 0.03783633 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.2375570\\\\\n",
       "\t12 & 12 & AAAS       & -0.46603918 & 4.709302 & 0.58753257 & 0.447902270 & 0.8529603 & FALSE & 0.069071167 & 0.084879806 & 0.05538560 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.8270879\\\\\n",
       "\t13 & 13 & AACS       & -0.98908185 & 7.550449 & 5.26392754 & 0.027132919 & 0.3449701 & FALSE & 0.462218499 & 1.043062838 & 0.52544506 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.2840057\\\\\n",
       "\t19 & 19 & AADAT      &  0.94386331 & 4.134502 & 1.62164242 & 0.262455054 & 0.7405559 & FALSE & 0.130442155 & 0.018837166 & 0.04744887 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.6992791\\\\\n",
       "\t20 & 20 & AAED1      & -0.24401663 & 5.161923 & 0.26371353 & 0.610422448 & 0.9190844 & FALSE & 0.036644615 & 0.145873508 & 0.11187294 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9018107\\\\\n",
       "\t21 & 21 & AAGAB      &  0.45101359 & 5.855196 & 1.28131223 & 0.264436204 & 0.7426188 & FALSE & 0.129234031 & 0.218194138 & 0.28857486 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.7011770\\\\\n",
       "\t22 & 22 & AAK1       & -0.41885189 & 7.437454 & 1.56756773 & 0.217876102 & 0.6979025 & FALSE & 0.156205216 & 0.807023347 & 0.66403594 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.6573869\\\\\n",
       "\t23 & 23 & AAMDC      & -0.05066493 & 6.737111 & 0.02175286 & 0.883491843 & 0.9957384 & FALSE & 0.001854735 & 0.439556781 & 0.46552705 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9862150\\\\\n",
       "\t24 & 24 & AAMP       &  0.88638900 & 5.954126 & 4.81878864 & 0.034051429 & 0.3709559 & FALSE & 0.430677673 & 0.184870489 & 0.38042888 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.3128976\\\\\n",
       "\t26 & 26 & AAR2       & -0.45415937 & 4.453402 & 0.46191759 & 0.500667238 & 0.8771219 & FALSE & 0.056940050 & 0.076662406 & 0.04974044 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.8535110\\\\\n",
       "\t28 & 28 & AARS       &  0.10138538 & 5.832970 & 0.05575659 & 0.814545311 & 0.9831224 & FALSE & 0.007392397 & 0.210718786 & 0.23817032 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9698404\\\\\n",
       "\t29 & 29 & AARS2      & -0.25884368 & 3.969703 & 0.09998201 & 0.838490446 & 0.9899243 & FALSE & 0.004397996 & 0.025139571 & 0.02127892 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9760201\\\\\n",
       "\t31 & 31 & AASDH      & -0.41023512 & 6.313453 & 1.17929368 & 0.284040294 & 0.7587664 & FALSE & 0.119891915 & 0.344242545 & 0.27855545 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.7177845\\\\\n",
       "\t32 & 32 & AASDHPPT   & -0.11776054 & 4.854325 & 0.04887435 & 0.826165142 & 0.9863212 & FALSE & 0.005981612 & 0.096734620 & 0.09087111 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9728590\\\\\n",
       "\t33 & 33 & AASS       &  1.34607718 & 8.199859 & 8.59974659 & 0.005554113 & 0.2009727 & FALSE & 0.696863035 & 0.789486746 & 2.46272870 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.1296984\\\\\n",
       "\t35 & 35 & AATF       &  0.32030802 & 5.823508 & 0.54991830 & 0.462709147 & 0.8598546 & FALSE & 0.065574979 & 0.174846891 & 0.25512331 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.8348744\\\\\n",
       "\t37 & 37 & AB015752.1 &  1.19703057 & 3.799611 & 2.36077590 & 0.161080081 & 0.6420055 & FALSE & 0.192461253 & 0.011167681 & 0.04025769 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.5925748\\\\\n",
       "\t39 & 39 & ABAT       &  1.11303692 & 5.071607 & 4.23657117 & 0.046157659 & 0.4204372 & FALSE & 0.376298862 & 0.061287154 & 0.17958940 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.3591232\\\\\n",
       "\t40 & 40 & ABCA1      & -0.10690232 & 8.900441 & 0.05711192 & 0.812347923 & 0.9827512 & FALSE & 0.007556420 & 2.536833495 & 2.15172900 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9692521\\\\\n",
       "\t41 & 41 & ABCA10     & -0.19649533 & 8.399351 & 0.23646518 & 0.629440533 & 0.9258420 & FALSE & 0.033463108 & 1.807205705 & 1.56555575 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9091768\\\\\n",
       "\t42 & 42 & ABCA12     & -0.17838759 & 3.677913 & 0.02885796 & 0.865968860 & 0.9937067 & FALSE & 0.002741783 & 0.008870902 & 0.01308498 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.9824329\\\\\n",
       "\t44 & 44 & ABCA2      & -1.10131876 & 4.836746 & 4.01166083 & 0.052037530 & 0.4373436 & FALSE & 0.359177197 & 0.095370814 & 0.05678431 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.3786983\\\\\n",
       "\t45 & 45 & ABCA3      & -0.79701954 & 5.093562 & 1.90898289 & 0.174791298 & 0.6572500 & FALSE & 0.182269410 & 0.120983852 & 0.06592178 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.6099814\\\\\n",
       "\t47 & 47 & ABCA5      &  0.54234585 & 8.193122 & 2.45393371 & 0.125156200 & 0.5883407 & FALSE & 0.230371114 & 1.146144429 & 1.85268285 & RV & CELLTYPE & AD & AD & control\\_LMNA & 4 & 10 & 0.5397835\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t1408550 & 1408550 & ZRANB2      & -0.41898202 & 8.448639 & 1.26870387 & 0.26001042 & 0.7792071 & FALSE & 0.1083470826 & 0.495081987 & 0.445529033 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6969873\\\\\n",
       "\t1408551 & 1408551 & ZRANB2-AS1  &  0.79287936 & 6.177717 & 1.84300242 & 0.17459996 & 0.6923383 & FALSE & 0.1596816366 & 0.036655743 & 0.088308936 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6097120\\\\\n",
       "\t1408552 & 1408552 & ZRANB3      &  0.31956251 & 7.424698 & 0.62199530 & 0.43030709 & 0.8827371 & FALSE & 0.0541686033 & 0.168801299 & 0.293178722 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8168626\\\\\n",
       "\t1408553 & 1408553 & ZRSR2       & -0.45465320 & 6.059947 & 0.51802073 & 0.47168749 & 0.8998568 & FALSE & 0.0458266101 & 0.054437668 & 0.038249592 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8393780\\\\\n",
       "\t1408556 & 1408556 & ZSCAN12     & -0.74060434 & 5.633310 & 0.93613950 & 0.33327303 & 0.8303767 & FALSE & 0.0807248385 & 0.022631743 & 0.024572532 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7551393\\\\\n",
       "\t1408558 & 1408558 & ZSCAN16-AS1 &  0.62339750 & 6.207718 & 1.21660302 & 0.27002878 & 0.7862802 & FALSE & 0.1044226827 & 0.049068949 & 0.077743672 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7055624\\\\\n",
       "\t1408561 & 1408561 & ZSCAN20     &  1.06303363 & 5.379492 & 2.06137799 & 0.17638007 & 0.6950039 & FALSE & 0.1580127864 & 0.006820119 & 0.013613779 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6117026\\\\\n",
       "\t1408562 & 1408562 & ZSCAN21     & -1.00452090 & 5.752995 & 1.79168088 & 0.18072200 & 0.7001433 & FALSE & 0.1548130924 & 0.032959138 & 0.029816296 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6172248\\\\\n",
       "\t1408564 & 1408564 & ZSCAN23     &  0.16046129 & 5.895627 & 0.05770854 & 0.81015513 & 0.9877482 & FALSE & 0.0053537579 & 0.029201649 & 0.061522312 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9686215\\\\\n",
       "\t1408565 & 1408565 & ZSCAN25     & -1.54206513 & 5.631264 & 3.47778069 & 0.06219859 & 0.4786725 & FALSE & 0.3199615603 & 0.024957639 & 0.008236628 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.4087370\\\\\n",
       "\t1408568 & 1408568 & ZSCAN30     &  0.05440543 & 6.379628 & 0.01055081 & 0.91818752 & 0.9996648 & FALSE & 0.0001455942 & 0.065791183 & 0.077565676 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9924597\\\\\n",
       "\t1408569 & 1408569 & ZSCAN31     & -1.06271914 & 5.604548 & 1.64146815 & 0.34161908 & 0.8348459 & FALSE & 0.0783936885 & 0.020349848 & 0.008370015 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7605515\\\\\n",
       "\t1408570 & 1408570 & ZSCAN32     &  0.81596942 & 5.540384 & 1.17296830 & 0.39077779 & 0.8642478 & FALSE & 0.0633617307 & 0.007653265 & 0.024787618 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7931896\\\\\n",
       "\t1408572 & 1408572 & ZSCAN5A     & -0.67410217 & 6.681754 & 1.76446333 & 0.18406927 & 0.7041076 & FALSE & 0.1523609408 & 0.104571220 & 0.077512146 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6211056\\\\\n",
       "\t1408575 & 1408575 & ZSCAN9      & -0.60970015 & 5.636025 & 0.63332216 & 0.42613965 & 0.8823821 & FALSE & 0.0543432973 & 0.017384014 & 0.018118284 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8145620\\\\\n",
       "\t1408578 & 1408578 & ZSWIM3      &  0.56558939 & 5.754830 & 0.65249464 & 0.41922251 & 0.8802542 & FALSE & 0.0553918746 & 0.022269408 & 0.034635120 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8104435\\\\\n",
       "\t1408579 & 1408579 & ZSWIM4      & -0.18332237 & 6.111745 & 0.08476571 & 0.77094014 & 0.9796231 & FALSE & 0.0089409995 & 0.052123460 & 0.042345600 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9574083\\\\\n",
       "\t1408580 & 1408580 & ZSWIM5      &  0.09104139 & 6.300519 & 0.02733914 & 0.86867209 & 0.9958883 & FALSE & 0.0017893879 & 0.055440952 & 0.068932999 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9829380\\\\\n",
       "\t1408581 & 1408581 & ZSWIM6      & -0.38265922 & 9.598415 & 1.18186247 & 0.27697759 & 0.7907358 & FALSE & 0.1019685740 & 1.367779867 & 1.119595615 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7116676\\\\\n",
       "\t1408582 & 1408582 & ZSWIM7      &  0.65144509 & 6.723159 & 1.96087040 & 0.16142077 & 0.6746079 & FALSE & 0.1709485821 & 0.091570986 & 0.145823402 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.5931049\\\\\n",
       "\t1408583 & 1408583 & ZSWIM8      &  0.59437810 & 6.493243 & 1.32810503 & 0.24914307 & 0.7687747 & FALSE & 0.1142009435 & 0.064281509 & 0.114732082 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6870315\\\\\n",
       "\t1408585 & 1408585 & ZSWIM9      &  0.52355515 & 5.807250 & 0.60875500 & 0.43525726 & 0.8836248 & FALSE & 0.0537321259 & 0.031769727 & 0.036879675 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8195799\\\\\n",
       "\t1408586 & 1408586 & ZUP1        & -1.16323146 & 5.843864 & 2.50731095 & 0.11331967 & 0.5977405 & FALSE & 0.2234872920 & 0.033584395 & 0.022733513 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.5204889\\\\\n",
       "\t1408587 & 1408587 & ZW10        &  0.36458751 & 5.817240 & 0.27125688 & 0.60249008 & 0.9407317 & FALSE & 0.0265342062 & 0.019933530 & 0.053036620 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.8984469\\\\\n",
       "\t1408588 & 1408588 & ZWILCH      &  1.10724111 & 5.399117 & 2.24174458 & 0.25572211 & 0.7762917 & FALSE & 0.1099750694 & 0.003225806 & 0.013946664 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.6931081\\\\\n",
       "\t1408592 & 1408592 & ZXDC        &  0.41384666 & 7.213291 & 0.95890384 & 0.32746360 & 0.8265347 & FALSE & 0.0827389028 & 0.141852096 & 0.240512273 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.7509211\\\\\n",
       "\t1408593 & 1408593 & ZYG11A      & -1.13629790 & 5.304394 & 3.36949385 & 0.16635761 & 0.6800895 & FALSE & 0.1674339095 & 0.016075269 & 0.000000000 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.5995422\\\\\n",
       "\t1408594 & 1408594 & ZYG11B      &  0.10468329 & 6.849860 & 0.05170007 & 0.82013107 & 0.9890962 & FALSE & 0.0047614528 & 0.111720833 & 0.129940881 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9713227\\\\\n",
       "\t1408595 & 1408595 & ZYX         & -1.21363157 & 5.608672 & 2.25447012 & 0.13322957 & 0.6313218 & FALSE & 0.1997492378 & 0.026622454 & 0.007905865 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.5531024\\\\\n",
       "\t1408596 & 1408596 & ZZEF1       &  0.09761451 & 7.200671 & 0.05165851 & 0.82020214 & 0.9890962 & FALSE & 0.0047614528 & 0.153851562 & 0.222772332 & RV & CELLTYPE & NC & NC & control\\_TTN & 6 & 15 & 0.9713227\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 511188 × 19\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | Gene &lt;fct&gt; | logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | low_expression &lt;lgl&gt; | FDR_plot &lt;dbl&gt; | mean_exp_control &lt;dbl&gt; | mean_exp_genotype &lt;dbl&gt; | Region &lt;fct&gt; | annotation_level &lt;fct&gt; | cell_state &lt;fct&gt; | cell_type &lt;fct&gt; | comparison &lt;fct&gt; | Observations_genotype &lt;int&gt; | Observations_reference &lt;int&gt; | FDR_plot_all &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 |  1 | A1BG       |  0.59280479 | 3.613484 | 0.44851693 | 0.550830654 | 0.8964962 | FALSE | 0.047451535 | 0.011645189 | 0.01754926 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.8767031 |\n",
       "| 4 |  4 | A2M        | -0.47805814 | 5.389797 | 0.82794671 | 0.368350327 | 0.8103331 | FALSE | 0.091336403 | 0.151950451 | 0.11321242 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.7785514 |\n",
       "| 5 |  5 | A2M-AS1    | -0.78384150 | 3.559792 | 0.58943441 | 0.447173742 | 0.8525847 | FALSE | 0.069262471 | 0.013619099 | 0.00862069 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.8266161 |\n",
       "| 6 |  6 | A2ML1      |  0.30857597 | 3.813569 | 0.14452276 | 0.705848923 | 0.9488013 | FALSE | 0.022824715 | 0.010749557 | 0.01712251 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9370670 |\n",
       "| 7 |  7 | A2ML1-AS1  |  0.57931432 | 5.658915 | 1.74265342 | 0.194352982 | 0.6787966 | FALSE | 0.168260364 | 0.156997997 | 0.24860482 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.6329451 |\n",
       "| 9 |  9 | A3GALT2    | -2.43383781 | 3.526761 | 5.50740195 | 0.049297917 | 0.4312155 | FALSE | 0.365305621 | 0.016359989 | 0.00000000 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.3699974 |\n",
       "| 10 | 10 | A4GALT     | -1.52364885 | 4.982904 | 6.05813642 | 0.018280381 | 0.3050396 | FALSE | 0.515643730 | 0.140615087 | 0.03783633 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.2375570 |\n",
       "| 12 | 12 | AAAS       | -0.46603918 | 4.709302 | 0.58753257 | 0.447902270 | 0.8529603 | FALSE | 0.069071167 | 0.084879806 | 0.05538560 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.8270879 |\n",
       "| 13 | 13 | AACS       | -0.98908185 | 7.550449 | 5.26392754 | 0.027132919 | 0.3449701 | FALSE | 0.462218499 | 1.043062838 | 0.52544506 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.2840057 |\n",
       "| 19 | 19 | AADAT      |  0.94386331 | 4.134502 | 1.62164242 | 0.262455054 | 0.7405559 | FALSE | 0.130442155 | 0.018837166 | 0.04744887 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.6992791 |\n",
       "| 20 | 20 | AAED1      | -0.24401663 | 5.161923 | 0.26371353 | 0.610422448 | 0.9190844 | FALSE | 0.036644615 | 0.145873508 | 0.11187294 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9018107 |\n",
       "| 21 | 21 | AAGAB      |  0.45101359 | 5.855196 | 1.28131223 | 0.264436204 | 0.7426188 | FALSE | 0.129234031 | 0.218194138 | 0.28857486 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.7011770 |\n",
       "| 22 | 22 | AAK1       | -0.41885189 | 7.437454 | 1.56756773 | 0.217876102 | 0.6979025 | FALSE | 0.156205216 | 0.807023347 | 0.66403594 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.6573869 |\n",
       "| 23 | 23 | AAMDC      | -0.05066493 | 6.737111 | 0.02175286 | 0.883491843 | 0.9957384 | FALSE | 0.001854735 | 0.439556781 | 0.46552705 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9862150 |\n",
       "| 24 | 24 | AAMP       |  0.88638900 | 5.954126 | 4.81878864 | 0.034051429 | 0.3709559 | FALSE | 0.430677673 | 0.184870489 | 0.38042888 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.3128976 |\n",
       "| 26 | 26 | AAR2       | -0.45415937 | 4.453402 | 0.46191759 | 0.500667238 | 0.8771219 | FALSE | 0.056940050 | 0.076662406 | 0.04974044 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.8535110 |\n",
       "| 28 | 28 | AARS       |  0.10138538 | 5.832970 | 0.05575659 | 0.814545311 | 0.9831224 | FALSE | 0.007392397 | 0.210718786 | 0.23817032 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9698404 |\n",
       "| 29 | 29 | AARS2      | -0.25884368 | 3.969703 | 0.09998201 | 0.838490446 | 0.9899243 | FALSE | 0.004397996 | 0.025139571 | 0.02127892 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9760201 |\n",
       "| 31 | 31 | AASDH      | -0.41023512 | 6.313453 | 1.17929368 | 0.284040294 | 0.7587664 | FALSE | 0.119891915 | 0.344242545 | 0.27855545 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.7177845 |\n",
       "| 32 | 32 | AASDHPPT   | -0.11776054 | 4.854325 | 0.04887435 | 0.826165142 | 0.9863212 | FALSE | 0.005981612 | 0.096734620 | 0.09087111 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9728590 |\n",
       "| 33 | 33 | AASS       |  1.34607718 | 8.199859 | 8.59974659 | 0.005554113 | 0.2009727 | FALSE | 0.696863035 | 0.789486746 | 2.46272870 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.1296984 |\n",
       "| 35 | 35 | AATF       |  0.32030802 | 5.823508 | 0.54991830 | 0.462709147 | 0.8598546 | FALSE | 0.065574979 | 0.174846891 | 0.25512331 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.8348744 |\n",
       "| 37 | 37 | AB015752.1 |  1.19703057 | 3.799611 | 2.36077590 | 0.161080081 | 0.6420055 | FALSE | 0.192461253 | 0.011167681 | 0.04025769 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.5925748 |\n",
       "| 39 | 39 | ABAT       |  1.11303692 | 5.071607 | 4.23657117 | 0.046157659 | 0.4204372 | FALSE | 0.376298862 | 0.061287154 | 0.17958940 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.3591232 |\n",
       "| 40 | 40 | ABCA1      | -0.10690232 | 8.900441 | 0.05711192 | 0.812347923 | 0.9827512 | FALSE | 0.007556420 | 2.536833495 | 2.15172900 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9692521 |\n",
       "| 41 | 41 | ABCA10     | -0.19649533 | 8.399351 | 0.23646518 | 0.629440533 | 0.9258420 | FALSE | 0.033463108 | 1.807205705 | 1.56555575 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9091768 |\n",
       "| 42 | 42 | ABCA12     | -0.17838759 | 3.677913 | 0.02885796 | 0.865968860 | 0.9937067 | FALSE | 0.002741783 | 0.008870902 | 0.01308498 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.9824329 |\n",
       "| 44 | 44 | ABCA2      | -1.10131876 | 4.836746 | 4.01166083 | 0.052037530 | 0.4373436 | FALSE | 0.359177197 | 0.095370814 | 0.05678431 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.3786983 |\n",
       "| 45 | 45 | ABCA3      | -0.79701954 | 5.093562 | 1.90898289 | 0.174791298 | 0.6572500 | FALSE | 0.182269410 | 0.120983852 | 0.06592178 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.6099814 |\n",
       "| 47 | 47 | ABCA5      |  0.54234585 | 8.193122 | 2.45393371 | 0.125156200 | 0.5883407 | FALSE | 0.230371114 | 1.146144429 | 1.85268285 | RV | CELLTYPE | AD | AD | control_LMNA | 4 | 10 | 0.5397835 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 1408550 | 1408550 | ZRANB2      | -0.41898202 | 8.448639 | 1.26870387 | 0.26001042 | 0.7792071 | FALSE | 0.1083470826 | 0.495081987 | 0.445529033 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6969873 |\n",
       "| 1408551 | 1408551 | ZRANB2-AS1  |  0.79287936 | 6.177717 | 1.84300242 | 0.17459996 | 0.6923383 | FALSE | 0.1596816366 | 0.036655743 | 0.088308936 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6097120 |\n",
       "| 1408552 | 1408552 | ZRANB3      |  0.31956251 | 7.424698 | 0.62199530 | 0.43030709 | 0.8827371 | FALSE | 0.0541686033 | 0.168801299 | 0.293178722 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8168626 |\n",
       "| 1408553 | 1408553 | ZRSR2       | -0.45465320 | 6.059947 | 0.51802073 | 0.47168749 | 0.8998568 | FALSE | 0.0458266101 | 0.054437668 | 0.038249592 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8393780 |\n",
       "| 1408556 | 1408556 | ZSCAN12     | -0.74060434 | 5.633310 | 0.93613950 | 0.33327303 | 0.8303767 | FALSE | 0.0807248385 | 0.022631743 | 0.024572532 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7551393 |\n",
       "| 1408558 | 1408558 | ZSCAN16-AS1 |  0.62339750 | 6.207718 | 1.21660302 | 0.27002878 | 0.7862802 | FALSE | 0.1044226827 | 0.049068949 | 0.077743672 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7055624 |\n",
       "| 1408561 | 1408561 | ZSCAN20     |  1.06303363 | 5.379492 | 2.06137799 | 0.17638007 | 0.6950039 | FALSE | 0.1580127864 | 0.006820119 | 0.013613779 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6117026 |\n",
       "| 1408562 | 1408562 | ZSCAN21     | -1.00452090 | 5.752995 | 1.79168088 | 0.18072200 | 0.7001433 | FALSE | 0.1548130924 | 0.032959138 | 0.029816296 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6172248 |\n",
       "| 1408564 | 1408564 | ZSCAN23     |  0.16046129 | 5.895627 | 0.05770854 | 0.81015513 | 0.9877482 | FALSE | 0.0053537579 | 0.029201649 | 0.061522312 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9686215 |\n",
       "| 1408565 | 1408565 | ZSCAN25     | -1.54206513 | 5.631264 | 3.47778069 | 0.06219859 | 0.4786725 | FALSE | 0.3199615603 | 0.024957639 | 0.008236628 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.4087370 |\n",
       "| 1408568 | 1408568 | ZSCAN30     |  0.05440543 | 6.379628 | 0.01055081 | 0.91818752 | 0.9996648 | FALSE | 0.0001455942 | 0.065791183 | 0.077565676 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9924597 |\n",
       "| 1408569 | 1408569 | ZSCAN31     | -1.06271914 | 5.604548 | 1.64146815 | 0.34161908 | 0.8348459 | FALSE | 0.0783936885 | 0.020349848 | 0.008370015 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7605515 |\n",
       "| 1408570 | 1408570 | ZSCAN32     |  0.81596942 | 5.540384 | 1.17296830 | 0.39077779 | 0.8642478 | FALSE | 0.0633617307 | 0.007653265 | 0.024787618 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7931896 |\n",
       "| 1408572 | 1408572 | ZSCAN5A     | -0.67410217 | 6.681754 | 1.76446333 | 0.18406927 | 0.7041076 | FALSE | 0.1523609408 | 0.104571220 | 0.077512146 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6211056 |\n",
       "| 1408575 | 1408575 | ZSCAN9      | -0.60970015 | 5.636025 | 0.63332216 | 0.42613965 | 0.8823821 | FALSE | 0.0543432973 | 0.017384014 | 0.018118284 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8145620 |\n",
       "| 1408578 | 1408578 | ZSWIM3      |  0.56558939 | 5.754830 | 0.65249464 | 0.41922251 | 0.8802542 | FALSE | 0.0553918746 | 0.022269408 | 0.034635120 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8104435 |\n",
       "| 1408579 | 1408579 | ZSWIM4      | -0.18332237 | 6.111745 | 0.08476571 | 0.77094014 | 0.9796231 | FALSE | 0.0089409995 | 0.052123460 | 0.042345600 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9574083 |\n",
       "| 1408580 | 1408580 | ZSWIM5      |  0.09104139 | 6.300519 | 0.02733914 | 0.86867209 | 0.9958883 | FALSE | 0.0017893879 | 0.055440952 | 0.068932999 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9829380 |\n",
       "| 1408581 | 1408581 | ZSWIM6      | -0.38265922 | 9.598415 | 1.18186247 | 0.27697759 | 0.7907358 | FALSE | 0.1019685740 | 1.367779867 | 1.119595615 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7116676 |\n",
       "| 1408582 | 1408582 | ZSWIM7      |  0.65144509 | 6.723159 | 1.96087040 | 0.16142077 | 0.6746079 | FALSE | 0.1709485821 | 0.091570986 | 0.145823402 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.5931049 |\n",
       "| 1408583 | 1408583 | ZSWIM8      |  0.59437810 | 6.493243 | 1.32810503 | 0.24914307 | 0.7687747 | FALSE | 0.1142009435 | 0.064281509 | 0.114732082 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6870315 |\n",
       "| 1408585 | 1408585 | ZSWIM9      |  0.52355515 | 5.807250 | 0.60875500 | 0.43525726 | 0.8836248 | FALSE | 0.0537321259 | 0.031769727 | 0.036879675 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8195799 |\n",
       "| 1408586 | 1408586 | ZUP1        | -1.16323146 | 5.843864 | 2.50731095 | 0.11331967 | 0.5977405 | FALSE | 0.2234872920 | 0.033584395 | 0.022733513 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.5204889 |\n",
       "| 1408587 | 1408587 | ZW10        |  0.36458751 | 5.817240 | 0.27125688 | 0.60249008 | 0.9407317 | FALSE | 0.0265342062 | 0.019933530 | 0.053036620 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.8984469 |\n",
       "| 1408588 | 1408588 | ZWILCH      |  1.10724111 | 5.399117 | 2.24174458 | 0.25572211 | 0.7762917 | FALSE | 0.1099750694 | 0.003225806 | 0.013946664 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.6931081 |\n",
       "| 1408592 | 1408592 | ZXDC        |  0.41384666 | 7.213291 | 0.95890384 | 0.32746360 | 0.8265347 | FALSE | 0.0827389028 | 0.141852096 | 0.240512273 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.7509211 |\n",
       "| 1408593 | 1408593 | ZYG11A      | -1.13629790 | 5.304394 | 3.36949385 | 0.16635761 | 0.6800895 | FALSE | 0.1674339095 | 0.016075269 | 0.000000000 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.5995422 |\n",
       "| 1408594 | 1408594 | ZYG11B      |  0.10468329 | 6.849860 | 0.05170007 | 0.82013107 | 0.9890962 | FALSE | 0.0047614528 | 0.111720833 | 0.129940881 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9713227 |\n",
       "| 1408595 | 1408595 | ZYX         | -1.21363157 | 5.608672 | 2.25447012 | 0.13322957 | 0.6313218 | FALSE | 0.1997492378 | 0.026622454 | 0.007905865 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.5531024 |\n",
       "| 1408596 | 1408596 | ZZEF1       |  0.09761451 | 7.200671 | 0.05165851 | 0.82020214 | 0.9890962 | FALSE | 0.0047614528 | 0.153851562 | 0.222772332 | RV | CELLTYPE | NC | NC | control_TTN | 6 | 15 | 0.9713227 |\n",
       "\n"
      ],
      "text/plain": [
       "        X       Gene        logFC       logCPM   F          PValue     \n",
       "1        1      A1BG         0.59280479 3.613484 0.44851693 0.550830654\n",
       "4        4      A2M         -0.47805814 5.389797 0.82794671 0.368350327\n",
       "5        5      A2M-AS1     -0.78384150 3.559792 0.58943441 0.447173742\n",
       "6        6      A2ML1        0.30857597 3.813569 0.14452276 0.705848923\n",
       "7        7      A2ML1-AS1    0.57931432 5.658915 1.74265342 0.194352982\n",
       "9        9      A3GALT2     -2.43383781 3.526761 5.50740195 0.049297917\n",
       "10      10      A4GALT      -1.52364885 4.982904 6.05813642 0.018280381\n",
       "12      12      AAAS        -0.46603918 4.709302 0.58753257 0.447902270\n",
       "13      13      AACS        -0.98908185 7.550449 5.26392754 0.027132919\n",
       "19      19      AADAT        0.94386331 4.134502 1.62164242 0.262455054\n",
       "20      20      AAED1       -0.24401663 5.161923 0.26371353 0.610422448\n",
       "21      21      AAGAB        0.45101359 5.855196 1.28131223 0.264436204\n",
       "22      22      AAK1        -0.41885189 7.437454 1.56756773 0.217876102\n",
       "23      23      AAMDC       -0.05066493 6.737111 0.02175286 0.883491843\n",
       "24      24      AAMP         0.88638900 5.954126 4.81878864 0.034051429\n",
       "26      26      AAR2        -0.45415937 4.453402 0.46191759 0.500667238\n",
       "28      28      AARS         0.10138538 5.832970 0.05575659 0.814545311\n",
       "29      29      AARS2       -0.25884368 3.969703 0.09998201 0.838490446\n",
       "31      31      AASDH       -0.41023512 6.313453 1.17929368 0.284040294\n",
       "32      32      AASDHPPT    -0.11776054 4.854325 0.04887435 0.826165142\n",
       "33      33      AASS         1.34607718 8.199859 8.59974659 0.005554113\n",
       "35      35      AATF         0.32030802 5.823508 0.54991830 0.462709147\n",
       "37      37      AB015752.1   1.19703057 3.799611 2.36077590 0.161080081\n",
       "39      39      ABAT         1.11303692 5.071607 4.23657117 0.046157659\n",
       "40      40      ABCA1       -0.10690232 8.900441 0.05711192 0.812347923\n",
       "41      41      ABCA10      -0.19649533 8.399351 0.23646518 0.629440533\n",
       "42      42      ABCA12      -0.17838759 3.677913 0.02885796 0.865968860\n",
       "44      44      ABCA2       -1.10131876 4.836746 4.01166083 0.052037530\n",
       "45      45      ABCA3       -0.79701954 5.093562 1.90898289 0.174791298\n",
       "47      47      ABCA5        0.54234585 8.193122 2.45393371 0.125156200\n",
       "⋮       ⋮       ⋮           ⋮           ⋮        ⋮          ⋮          \n",
       "1408550 1408550 ZRANB2      -0.41898202 8.448639 1.26870387 0.26001042 \n",
       "1408551 1408551 ZRANB2-AS1   0.79287936 6.177717 1.84300242 0.17459996 \n",
       "1408552 1408552 ZRANB3       0.31956251 7.424698 0.62199530 0.43030709 \n",
       "1408553 1408553 ZRSR2       -0.45465320 6.059947 0.51802073 0.47168749 \n",
       "1408556 1408556 ZSCAN12     -0.74060434 5.633310 0.93613950 0.33327303 \n",
       "1408558 1408558 ZSCAN16-AS1  0.62339750 6.207718 1.21660302 0.27002878 \n",
       "1408561 1408561 ZSCAN20      1.06303363 5.379492 2.06137799 0.17638007 \n",
       "1408562 1408562 ZSCAN21     -1.00452090 5.752995 1.79168088 0.18072200 \n",
       "1408564 1408564 ZSCAN23      0.16046129 5.895627 0.05770854 0.81015513 \n",
       "1408565 1408565 ZSCAN25     -1.54206513 5.631264 3.47778069 0.06219859 \n",
       "1408568 1408568 ZSCAN30      0.05440543 6.379628 0.01055081 0.91818752 \n",
       "1408569 1408569 ZSCAN31     -1.06271914 5.604548 1.64146815 0.34161908 \n",
       "1408570 1408570 ZSCAN32      0.81596942 5.540384 1.17296830 0.39077779 \n",
       "1408572 1408572 ZSCAN5A     -0.67410217 6.681754 1.76446333 0.18406927 \n",
       "1408575 1408575 ZSCAN9      -0.60970015 5.636025 0.63332216 0.42613965 \n",
       "1408578 1408578 ZSWIM3       0.56558939 5.754830 0.65249464 0.41922251 \n",
       "1408579 1408579 ZSWIM4      -0.18332237 6.111745 0.08476571 0.77094014 \n",
       "1408580 1408580 ZSWIM5       0.09104139 6.300519 0.02733914 0.86867209 \n",
       "1408581 1408581 ZSWIM6      -0.38265922 9.598415 1.18186247 0.27697759 \n",
       "1408582 1408582 ZSWIM7       0.65144509 6.723159 1.96087040 0.16142077 \n",
       "1408583 1408583 ZSWIM8       0.59437810 6.493243 1.32810503 0.24914307 \n",
       "1408585 1408585 ZSWIM9       0.52355515 5.807250 0.60875500 0.43525726 \n",
       "1408586 1408586 ZUP1        -1.16323146 5.843864 2.50731095 0.11331967 \n",
       "1408587 1408587 ZW10         0.36458751 5.817240 0.27125688 0.60249008 \n",
       "1408588 1408588 ZWILCH       1.10724111 5.399117 2.24174458 0.25572211 \n",
       "1408592 1408592 ZXDC         0.41384666 7.213291 0.95890384 0.32746360 \n",
       "1408593 1408593 ZYG11A      -1.13629790 5.304394 3.36949385 0.16635761 \n",
       "1408594 1408594 ZYG11B       0.10468329 6.849860 0.05170007 0.82013107 \n",
       "1408595 1408595 ZYX         -1.21363157 5.608672 2.25447012 0.13322957 \n",
       "1408596 1408596 ZZEF1        0.09761451 7.200671 0.05165851 0.82020214 \n",
       "        FDR       low_expression FDR_plot     mean_exp_control\n",
       "1       0.8964962 FALSE          0.047451535  0.011645189     \n",
       "4       0.8103331 FALSE          0.091336403  0.151950451     \n",
       "5       0.8525847 FALSE          0.069262471  0.013619099     \n",
       "6       0.9488013 FALSE          0.022824715  0.010749557     \n",
       "7       0.6787966 FALSE          0.168260364  0.156997997     \n",
       "9       0.4312155 FALSE          0.365305621  0.016359989     \n",
       "10      0.3050396 FALSE          0.515643730  0.140615087     \n",
       "12      0.8529603 FALSE          0.069071167  0.084879806     \n",
       "13      0.3449701 FALSE          0.462218499  1.043062838     \n",
       "19      0.7405559 FALSE          0.130442155  0.018837166     \n",
       "20      0.9190844 FALSE          0.036644615  0.145873508     \n",
       "21      0.7426188 FALSE          0.129234031  0.218194138     \n",
       "22      0.6979025 FALSE          0.156205216  0.807023347     \n",
       "23      0.9957384 FALSE          0.001854735  0.439556781     \n",
       "24      0.3709559 FALSE          0.430677673  0.184870489     \n",
       "26      0.8771219 FALSE          0.056940050  0.076662406     \n",
       "28      0.9831224 FALSE          0.007392397  0.210718786     \n",
       "29      0.9899243 FALSE          0.004397996  0.025139571     \n",
       "31      0.7587664 FALSE          0.119891915  0.344242545     \n",
       "32      0.9863212 FALSE          0.005981612  0.096734620     \n",
       "33      0.2009727 FALSE          0.696863035  0.789486746     \n",
       "35      0.8598546 FALSE          0.065574979  0.174846891     \n",
       "37      0.6420055 FALSE          0.192461253  0.011167681     \n",
       "39      0.4204372 FALSE          0.376298862  0.061287154     \n",
       "40      0.9827512 FALSE          0.007556420  2.536833495     \n",
       "41      0.9258420 FALSE          0.033463108  1.807205705     \n",
       "42      0.9937067 FALSE          0.002741783  0.008870902     \n",
       "44      0.4373436 FALSE          0.359177197  0.095370814     \n",
       "45      0.6572500 FALSE          0.182269410  0.120983852     \n",
       "47      0.5883407 FALSE          0.230371114  1.146144429     \n",
       "⋮       ⋮         ⋮              ⋮            ⋮               \n",
       "1408550 0.7792071 FALSE          0.1083470826 0.495081987     \n",
       "1408551 0.6923383 FALSE          0.1596816366 0.036655743     \n",
       "1408552 0.8827371 FALSE          0.0541686033 0.168801299     \n",
       "1408553 0.8998568 FALSE          0.0458266101 0.054437668     \n",
       "1408556 0.8303767 FALSE          0.0807248385 0.022631743     \n",
       "1408558 0.7862802 FALSE          0.1044226827 0.049068949     \n",
       "1408561 0.6950039 FALSE          0.1580127864 0.006820119     \n",
       "1408562 0.7001433 FALSE          0.1548130924 0.032959138     \n",
       "1408564 0.9877482 FALSE          0.0053537579 0.029201649     \n",
       "1408565 0.4786725 FALSE          0.3199615603 0.024957639     \n",
       "1408568 0.9996648 FALSE          0.0001455942 0.065791183     \n",
       "1408569 0.8348459 FALSE          0.0783936885 0.020349848     \n",
       "1408570 0.8642478 FALSE          0.0633617307 0.007653265     \n",
       "1408572 0.7041076 FALSE          0.1523609408 0.104571220     \n",
       "1408575 0.8823821 FALSE          0.0543432973 0.017384014     \n",
       "1408578 0.8802542 FALSE          0.0553918746 0.022269408     \n",
       "1408579 0.9796231 FALSE          0.0089409995 0.052123460     \n",
       "1408580 0.9958883 FALSE          0.0017893879 0.055440952     \n",
       "1408581 0.7907358 FALSE          0.1019685740 1.367779867     \n",
       "1408582 0.6746079 FALSE          0.1709485821 0.091570986     \n",
       "1408583 0.7687747 FALSE          0.1142009435 0.064281509     \n",
       "1408585 0.8836248 FALSE          0.0537321259 0.031769727     \n",
       "1408586 0.5977405 FALSE          0.2234872920 0.033584395     \n",
       "1408587 0.9407317 FALSE          0.0265342062 0.019933530     \n",
       "1408588 0.7762917 FALSE          0.1099750694 0.003225806     \n",
       "1408592 0.8265347 FALSE          0.0827389028 0.141852096     \n",
       "1408593 0.6800895 FALSE          0.1674339095 0.016075269     \n",
       "1408594 0.9890962 FALSE          0.0047614528 0.111720833     \n",
       "1408595 0.6313218 FALSE          0.1997492378 0.026622454     \n",
       "1408596 0.9890962 FALSE          0.0047614528 0.153851562     \n",
       "        mean_exp_genotype Region annotation_level cell_state cell_type\n",
       "1       0.01754926        RV     CELLTYPE         AD         AD       \n",
       "4       0.11321242        RV     CELLTYPE         AD         AD       \n",
       "5       0.00862069        RV     CELLTYPE         AD         AD       \n",
       "6       0.01712251        RV     CELLTYPE         AD         AD       \n",
       "7       0.24860482        RV     CELLTYPE         AD         AD       \n",
       "9       0.00000000        RV     CELLTYPE         AD         AD       \n",
       "10      0.03783633        RV     CELLTYPE         AD         AD       \n",
       "12      0.05538560        RV     CELLTYPE         AD         AD       \n",
       "13      0.52544506        RV     CELLTYPE         AD         AD       \n",
       "19      0.04744887        RV     CELLTYPE         AD         AD       \n",
       "20      0.11187294        RV     CELLTYPE         AD         AD       \n",
       "21      0.28857486        RV     CELLTYPE         AD         AD       \n",
       "22      0.66403594        RV     CELLTYPE         AD         AD       \n",
       "23      0.46552705        RV     CELLTYPE         AD         AD       \n",
       "24      0.38042888        RV     CELLTYPE         AD         AD       \n",
       "26      0.04974044        RV     CELLTYPE         AD         AD       \n",
       "28      0.23817032        RV     CELLTYPE         AD         AD       \n",
       "29      0.02127892        RV     CELLTYPE         AD         AD       \n",
       "31      0.27855545        RV     CELLTYPE         AD         AD       \n",
       "32      0.09087111        RV     CELLTYPE         AD         AD       \n",
       "33      2.46272870        RV     CELLTYPE         AD         AD       \n",
       "35      0.25512331        RV     CELLTYPE         AD         AD       \n",
       "37      0.04025769        RV     CELLTYPE         AD         AD       \n",
       "39      0.17958940        RV     CELLTYPE         AD         AD       \n",
       "40      2.15172900        RV     CELLTYPE         AD         AD       \n",
       "41      1.56555575        RV     CELLTYPE         AD         AD       \n",
       "42      0.01308498        RV     CELLTYPE         AD         AD       \n",
       "44      0.05678431        RV     CELLTYPE         AD         AD       \n",
       "45      0.06592178        RV     CELLTYPE         AD         AD       \n",
       "47      1.85268285        RV     CELLTYPE         AD         AD       \n",
       "⋮       ⋮                 ⋮      ⋮                ⋮          ⋮        \n",
       "1408550 0.445529033       RV     CELLTYPE         NC         NC       \n",
       "1408551 0.088308936       RV     CELLTYPE         NC         NC       \n",
       "1408552 0.293178722       RV     CELLTYPE         NC         NC       \n",
       "1408553 0.038249592       RV     CELLTYPE         NC         NC       \n",
       "1408556 0.024572532       RV     CELLTYPE         NC         NC       \n",
       "1408558 0.077743672       RV     CELLTYPE         NC         NC       \n",
       "1408561 0.013613779       RV     CELLTYPE         NC         NC       \n",
       "1408562 0.029816296       RV     CELLTYPE         NC         NC       \n",
       "1408564 0.061522312       RV     CELLTYPE         NC         NC       \n",
       "1408565 0.008236628       RV     CELLTYPE         NC         NC       \n",
       "1408568 0.077565676       RV     CELLTYPE         NC         NC       \n",
       "1408569 0.008370015       RV     CELLTYPE         NC         NC       \n",
       "1408570 0.024787618       RV     CELLTYPE         NC         NC       \n",
       "1408572 0.077512146       RV     CELLTYPE         NC         NC       \n",
       "1408575 0.018118284       RV     CELLTYPE         NC         NC       \n",
       "1408578 0.034635120       RV     CELLTYPE         NC         NC       \n",
       "1408579 0.042345600       RV     CELLTYPE         NC         NC       \n",
       "1408580 0.068932999       RV     CELLTYPE         NC         NC       \n",
       "1408581 1.119595615       RV     CELLTYPE         NC         NC       \n",
       "1408582 0.145823402       RV     CELLTYPE         NC         NC       \n",
       "1408583 0.114732082       RV     CELLTYPE         NC         NC       \n",
       "1408585 0.036879675       RV     CELLTYPE         NC         NC       \n",
       "1408586 0.022733513       RV     CELLTYPE         NC         NC       \n",
       "1408587 0.053036620       RV     CELLTYPE         NC         NC       \n",
       "1408588 0.013946664       RV     CELLTYPE         NC         NC       \n",
       "1408592 0.240512273       RV     CELLTYPE         NC         NC       \n",
       "1408593 0.000000000       RV     CELLTYPE         NC         NC       \n",
       "1408594 0.129940881       RV     CELLTYPE         NC         NC       \n",
       "1408595 0.007905865       RV     CELLTYPE         NC         NC       \n",
       "1408596 0.222772332       RV     CELLTYPE         NC         NC       \n",
       "        comparison   Observations_genotype Observations_reference FDR_plot_all\n",
       "1       control_LMNA 4                     10                     0.8767031   \n",
       "4       control_LMNA 4                     10                     0.7785514   \n",
       "5       control_LMNA 4                     10                     0.8266161   \n",
       "6       control_LMNA 4                     10                     0.9370670   \n",
       "7       control_LMNA 4                     10                     0.6329451   \n",
       "9       control_LMNA 4                     10                     0.3699974   \n",
       "10      control_LMNA 4                     10                     0.2375570   \n",
       "12      control_LMNA 4                     10                     0.8270879   \n",
       "13      control_LMNA 4                     10                     0.2840057   \n",
       "19      control_LMNA 4                     10                     0.6992791   \n",
       "20      control_LMNA 4                     10                     0.9018107   \n",
       "21      control_LMNA 4                     10                     0.7011770   \n",
       "22      control_LMNA 4                     10                     0.6573869   \n",
       "23      control_LMNA 4                     10                     0.9862150   \n",
       "24      control_LMNA 4                     10                     0.3128976   \n",
       "26      control_LMNA 4                     10                     0.8535110   \n",
       "28      control_LMNA 4                     10                     0.9698404   \n",
       "29      control_LMNA 4                     10                     0.9760201   \n",
       "31      control_LMNA 4                     10                     0.7177845   \n",
       "32      control_LMNA 4                     10                     0.9728590   \n",
       "33      control_LMNA 4                     10                     0.1296984   \n",
       "35      control_LMNA 4                     10                     0.8348744   \n",
       "37      control_LMNA 4                     10                     0.5925748   \n",
       "39      control_LMNA 4                     10                     0.3591232   \n",
       "40      control_LMNA 4                     10                     0.9692521   \n",
       "41      control_LMNA 4                     10                     0.9091768   \n",
       "42      control_LMNA 4                     10                     0.9824329   \n",
       "44      control_LMNA 4                     10                     0.3786983   \n",
       "45      control_LMNA 4                     10                     0.6099814   \n",
       "47      control_LMNA 4                     10                     0.5397835   \n",
       "⋮       ⋮            ⋮                     ⋮                      ⋮           \n",
       "1408550 control_TTN  6                     15                     0.6969873   \n",
       "1408551 control_TTN  6                     15                     0.6097120   \n",
       "1408552 control_TTN  6                     15                     0.8168626   \n",
       "1408553 control_TTN  6                     15                     0.8393780   \n",
       "1408556 control_TTN  6                     15                     0.7551393   \n",
       "1408558 control_TTN  6                     15                     0.7055624   \n",
       "1408561 control_TTN  6                     15                     0.6117026   \n",
       "1408562 control_TTN  6                     15                     0.6172248   \n",
       "1408564 control_TTN  6                     15                     0.9686215   \n",
       "1408565 control_TTN  6                     15                     0.4087370   \n",
       "1408568 control_TTN  6                     15                     0.9924597   \n",
       "1408569 control_TTN  6                     15                     0.7605515   \n",
       "1408570 control_TTN  6                     15                     0.7931896   \n",
       "1408572 control_TTN  6                     15                     0.6211056   \n",
       "1408575 control_TTN  6                     15                     0.8145620   \n",
       "1408578 control_TTN  6                     15                     0.8104435   \n",
       "1408579 control_TTN  6                     15                     0.9574083   \n",
       "1408580 control_TTN  6                     15                     0.9829380   \n",
       "1408581 control_TTN  6                     15                     0.7116676   \n",
       "1408582 control_TTN  6                     15                     0.5931049   \n",
       "1408583 control_TTN  6                     15                     0.6870315   \n",
       "1408585 control_TTN  6                     15                     0.8195799   \n",
       "1408586 control_TTN  6                     15                     0.5204889   \n",
       "1408587 control_TTN  6                     15                     0.8984469   \n",
       "1408588 control_TTN  6                     15                     0.6931081   \n",
       "1408592 control_TTN  6                     15                     0.7509211   \n",
       "1408593 control_TTN  6                     15                     0.5995422   \n",
       "1408594 control_TTN  6                     15                     0.9713227   \n",
       "1408595 control_TTN  6                     15                     0.5531024   \n",
       "1408596 control_TTN  6                     15                     0.9713227   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x[which(x$FDR<1),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a394d6b3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
